// @canonical/react-components@3.11.1/dist/esm/components/Code/Code.js
import _pt from "prop-types";
var _excluded = ["children", "className", "inline", "copyable", "numbered"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React from "react";
import CodeSnippet, { CodeSnippetBlockAppearance } from "../CodeSnippet";
import { IS_DEV } from "../../utils";
/**
 * @deprecated Code component is deprecated. Use CodeSnippet component or inline `<code>` instead.
 */
var Code = _ref => {
  var {
      children,
      className,
      inline,
      copyable,
      numbered
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  if (IS_DEV) {
    console.warn("Code component is deprecated. Use CodeSnippet component or inline `<code>` instead.");
  }
  if (inline) {
    return /*#__PURE__*/React.createElement("code", _extends({
      className: className
    }, props), children);
  } else {
    var appearance = null;
    if (numbered) {
      appearance = CodeSnippetBlockAppearance.NUMBERED;
    } else if (copyable) {
      appearance = CodeSnippetBlockAppearance.LINUX_PROMPT;
    }
    return /*#__PURE__*/React.createElement(CodeSnippet, {
      blocks: [{
        appearance,
        code: children
      }]
    });
  }
};
Code.propTypes = {
  children: _pt.string.isRequired,
  inline: _pt.bool,
  copyable: _pt.bool,
  numbered: _pt.bool
};
export default Code;
