// @canonical/react-components@4.5.4/dist/components/MultiSelect/MultiSelect.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithoutSorting = exports.WithDisabledItems = exports.SearchExample = exports.HelpText = exports.FormikCallbacksAndEmptyState = exports.EmptyStateNode = exports.EmptyMessage = exports.Disabled = exports.CondensedExample = void 0;
var _react = _interopRequireWildcard(require("react"));
var _formik = require("formik");
var _index = require("../../index");
var _MultiSelect = require("./MultiSelect");
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
const Template = props => {
  const [selectedItems, setSelectedItems] = (0, _react.useState)(props.selectedItems || []);
  return /*#__PURE__*/_react.default.createElement(_MultiSelect.MultiSelect, _extends({}, props, {
    selectedItems: selectedItems,
    onItemsUpdate: setSelectedItems
  }));
};
const meta = {
  component: _MultiSelect.MultiSelect,
  render: Template,
  tags: ["autodocs"],
  parameters: {}
};
var _default = exports.default = meta;
const groupedItems = [{
  label: "Almond",
  value: "almond",
  group: "Nuts"
}, {
  label: "Cashew",
  value: "cashew",
  group: "Nuts"
}, {
  label: "Mango",
  value: "mango",
  group: "Fruit"
}, {
  label: "Peach",
  value: "peach",
  group: "Fruit"
}];
const CondensedExample = exports.CondensedExample = {
  args: {
    items: [...Array.from({
      length: 26
    }, (_, i) => ({
      label: "".concat(String.fromCharCode(i + 65)),
      value: "".concat(String.fromCharCode(i + 65))
    })), ...Array.from({
      length: 26
    }, (_, i) => ({
      label: "Item ".concat(i + 1),
      value: i + 1
    }))],
    selectedItems: [{
      label: "A",
      value: "A"
    }, {
      label: "Item 2",
      value: 2
    }],
    variant: "condensed"
  }
};
const SearchExample = exports.SearchExample = {
  args: _objectSpread(_objectSpread({}, CondensedExample.args), {}, {
    variant: "search",
    items: [...CondensedExample.args.items.map((item, i) => _objectSpread(_objectSpread({}, item), {}, {
      group: i % 2 === 0 ? "Group 1" : "Group 2"
    }))],
    id: "search-id"
  })
};
const WithDisabledItems = exports.WithDisabledItems = {
  args: _objectSpread(_objectSpread({}, CondensedExample.args), {}, {
    disabledItems: [{
      label: "Item 1",
      value: 1
    }, {
      label: "Item 2",
      value: 2
    }]
  })
};
const WithoutSorting = exports.WithoutSorting = {
  args: {
    items: [{
      label: "Sunday",
      value: 0
    }, {
      label: "Monday",
      value: 1
    }, {
      label: "Tuesday",
      value: 2
    }, {
      label: "Wednesday",
      value: 3
    }, {
      label: "Thursday",
      value: 4
    }, {
      label: "Friday",
      value: 5
    }, {
      label: "Saturday",
      value: 6
    }],
    selectedItems: [{
      label: "Monday",
      value: 1
    }, {
      label: "Wednesday",
      value: 3
    }, {
      label: "Friday",
      value: 5
    }],
    variant: "condensed",
    isSortedAlphabetically: false,
    hasSelectedItemsFirst: false,
    id: "external-id"
  }
};
const Disabled = exports.Disabled = {
  args: _objectSpread(_objectSpread({}, CondensedExample.args), {}, {
    disabled: true
  })
};
const HelpText = exports.HelpText = {
  args: _objectSpread(_objectSpread({}, CondensedExample.args), {}, {
    help: /*#__PURE__*/_react.default.createElement("span", null, "This is a help text, that should appear underneath the component.")
  })
};
const FormikCallbacksAndEmptyStateTemplate = () => {
  const [selectedItems, setSelectedItems] = (0, _react.useState)([]);
  const [events, setEvents] = (0, _react.useState)([]);
  const addEvent = eventName => {
    setEvents(previousEvents => [eventName, ...previousEvents].slice(0, 6));
  };
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      maxWidth: "28rem"
    }
  }, /*#__PURE__*/_react.default.createElement(_formik.Formik, {
    initialValues: {
      ingredients: ""
    },
    onSubmit: () => {}
  }, /*#__PURE__*/_react.default.createElement(_index.FormikField, {
    name: "ingredients",
    component: _MultiSelect.MultiSelect,
    label: "Ingredients",
    variant: "search",
    placeholder: "Search ingredients",
    items: groupedItems,
    selectedItems: selectedItems,
    onItemsUpdate: setSelectedItems,
    onSearchChange: value => {
      addEvent("onSearchChange(\"".concat(value, "\")"));
    },
    onOpen: () => addEvent("onOpen()"),
    onClose: () => addEvent("onClose()"),
    emptyMessage: "No ingredients found"
  })), /*#__PURE__*/_react.default.createElement("p", {
    style: {
      marginBottom: "0.5rem"
    }
  }, "Callback log:"), /*#__PURE__*/_react.default.createElement("ul", {
    style: {
      margin: 0,
      paddingLeft: "1.25rem"
    }
  }, events.map((event, index) => /*#__PURE__*/_react.default.createElement("li", {
    key: "".concat(event, "-").concat(index)
  }, event))));
};
const FormikCallbacksAndEmptyState = exports.FormikCallbacksAndEmptyState = {
  render: FormikCallbacksAndEmptyStateTemplate
};
const EmptyMessage = exports.EmptyMessage = {
  args: {
    variant: "search",
    items: groupedItems,
    emptyMessage: "No matching ingredients.",
    placeholder: "Try typing kiwi"
  }
};
const EmptyStateNode = exports.EmptyStateNode = {
  args: {
    variant: "search",
    items: groupedItems,
    placeholder: "Try typing kiwi",
    emptyState: /*#__PURE__*/_react.default.createElement("div", {
      className: "u-align--center",
      style: {
        padding: "0.75rem 1rem"
      }
    }, /*#__PURE__*/_react.default.createElement("strong", null, "No ingredient matches."), /*#__PURE__*/_react.default.createElement("p", {
      style: {
        margin: "0.25rem 0 0"
      }
    }, "Use a broader term."))
  }
};
