// @canonical/react-components@3.11.1/dist/components/ApplicationLayout/AppNavigationBar/AppNavigationBar.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.States = exports.Default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _Application = _interopRequireDefault(require("../Application"));
var _AppMain = _interopRequireDefault(require("../AppMain"));
var _Panel = _interopRequireDefault(require("../../Panel"));
var _Strip = _interopRequireDefault(require("../../Strip"));
var _AppNavigationBar = _interopRequireDefault(require("./AppNavigationBar"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const meta = {
  component: _AppNavigationBar.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
/**
 * In most common cases an `AppNavigationBar` should contain a `<Panel>`.
 *
 * `AppNavigationBar` should be a direct child of an `<Application>` or when using `ApplicationLayout`
 * it will already be included.
 *
 * The toggle state should control the collapsed state of `AppNavigation`.
 */
const Default = exports.Default = {
  args: {
    children: "AppNavigation"
  }
};
const States = exports.States = {
  render: args => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [collapsed, setCollapsed] = (0, _react.useState)(true);
    return /*#__PURE__*/_react.default.createElement(_Application.default, null, /*#__PURE__*/_react.default.createElement(_AppNavigationBar.default, args, /*#__PURE__*/_react.default.createElement(_Panel.default, {
      dark: true,
      logo: {
        icon: "https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg",
        name: "https://assets.ubuntu.com/v1/2e04d794-logo-jaas.svg",
        nameAlt: "JAAS",
        href: "/"
      },
      toggle: {
        label: "Menu",
        onClick: () => setCollapsed(!collapsed)
      }
    })), /*#__PURE__*/_react.default.createElement(_AppMain.default, null, /*#__PURE__*/_react.default.createElement(_Strip.default, null, /*#__PURE__*/_react.default.createElement("p", null, "Reduce your window size to mobile to see the navigation bar."), /*#__PURE__*/_react.default.createElement("p", null, "Pass the collapsed state to the main menu:", " ", /*#__PURE__*/_react.default.createElement("code", null, "<AppNavigation collapsed={".concat(collapsed, "} />"))))));
  }
};
