// @canonical/react-components@3.11.1/dist/components/ConfirmationButton/ConfirmationButton.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithPreModalOpenHook = exports.IconOnly = exports.IconAndLabel = exports.Default = exports.BaseAppearance = void 0;
var _react = _interopRequireWildcard(require("react"));
var _ConfirmationButton = _interopRequireDefault(require("./ConfirmationButton"));
var _Icon = _interopRequireWildcard(require("../Icon"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const doNothing = () => {};
const meta = {
  component: _ConfirmationButton.default,
  tags: ["autodocs"]
};
const Default = exports.Default = {
  render: () => /*#__PURE__*/_react.default.createElement(_ConfirmationButton.default, {
    confirmationModalProps: {
      title: "Confirm delete",
      confirmButtonLabel: "Delete",
      onConfirm: doNothing,
      children: /*#__PURE__*/_react.default.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/_react.default.createElement("br", null), "You cannot undo this action.")
    }
  }, "Delete"),
  name: "Default"
};
var _default = exports.default = meta;
const IconAndLabel = exports.IconAndLabel = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [isLoading, setLoading] = (0, _react.useState)(false);
    const brieflyLoad = () => {
      setLoading(true);
      setTimeout(() => setLoading(false), 2000);
    };
    return /*#__PURE__*/_react.default.createElement(_ConfirmationButton.default, {
      confirmationModalProps: {
        title: "Confirm delete",
        confirmButtonLabel: "Delete",
        onConfirm: brieflyLoad,
        children: /*#__PURE__*/_react.default.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/_react.default.createElement("br", null), "You cannot undo this action.")
      },
      className: "has-icon",
      disabled: isLoading,
      loading: isLoading,
      shiftClickEnabled: true,
      showShiftClickHint: true
    }, /*#__PURE__*/_react.default.createElement(_Icon.default, {
      name: _Icon.ICONS.delete
    }), /*#__PURE__*/_react.default.createElement("span", null, "Delete"));
  },
  name: "Icon and label"
};
const IconOnly = exports.IconOnly = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [isLoading, setLoading] = (0, _react.useState)(false);
    const brieflyLoad = () => {
      setLoading(true);
      setTimeout(() => setLoading(false), 2000);
    };
    return /*#__PURE__*/_react.default.createElement(_ConfirmationButton.default, {
      confirmationModalProps: {
        title: "Confirm delete",
        confirmButtonLabel: "Delete",
        onConfirm: brieflyLoad,
        children: /*#__PURE__*/_react.default.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/_react.default.createElement("br", null), "You cannot undo this action.")
      },
      className: "has-icon",
      disabled: isLoading,
      loading: isLoading,
      shiftClickEnabled: true,
      showShiftClickHint: true
    }, /*#__PURE__*/_react.default.createElement(_Icon.default, {
      name: _Icon.ICONS.delete
    }));
  },
  name: "Icon only"
};
const BaseAppearance = exports.BaseAppearance = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [isLoading, setLoading] = (0, _react.useState)(false);
    const brieflyLoad = () => {
      setLoading(true);
      setTimeout(() => setLoading(false), 2000);
    };
    return /*#__PURE__*/_react.default.createElement(_ConfirmationButton.default, {
      appearance: "base",
      confirmationModalProps: {
        title: "Confirm delete",
        confirmButtonLabel: "Delete",
        onConfirm: brieflyLoad,
        children: /*#__PURE__*/_react.default.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/_react.default.createElement("br", null), "You cannot undo this action.")
      },
      className: "has-icon",
      disabled: isLoading,
      loading: isLoading,
      shiftClickEnabled: true,
      showShiftClickHint: true
    }, /*#__PURE__*/_react.default.createElement(_Icon.default, {
      name: _Icon.ICONS.delete
    }));
  },
  name: "Base appearance"
};
const WithPreModalOpenHook = exports.WithPreModalOpenHook = {
  render: () => {
    const dontOpenModal = () => false;
    const openModal = () => true;
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_ConfirmationButton.default, {
      preModalOpenHook: dontOpenModal,
      confirmationModalProps: {
        title: "Confirm delete",
        confirmButtonLabel: "Delete",
        children: /*#__PURE__*/_react.default.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/_react.default.createElement("br", null), "You cannot undo this action.")
      }
    }, "Don't open modal"), /*#__PURE__*/_react.default.createElement(_ConfirmationButton.default, {
      preModalOpenHook: openModal,
      confirmationModalProps: {
        title: "Confirm delete",
        confirmButtonLabel: "Delete",
        children: /*#__PURE__*/_react.default.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/_react.default.createElement("br", null), "You cannot undo this action.")
      }
    }, "Open modal"));
  },
  name: "With preModalOpenHook handler"
};
