// @canonical/react-components@3.11.1/dist/components/EmptyState/EmptyState.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithContent = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _EmptyState = _interopRequireDefault(require("./EmptyState"));
var _Icon = _interopRequireWildcard(require("../Icon"));
var _Button = _interopRequireDefault(require("../Button"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const boxImage = /*#__PURE__*/_react.default.createElement("img", {
  src: "https://assets.ubuntu.com/v1/c17e0d92-container.svg",
  style: {
    maxWidth: "10rem",
    maxHeight: "10rem"
  },
  alt: "empty state"
});
const doNothing = () => {};
const meta = {
  component: _EmptyState.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: () => /*#__PURE__*/_react.default.createElement(_EmptyState.default, {
    image: boxImage,
    title: "This application doesn't have any configuration parameters"
  }),
  name: "Default"
};
const WithContent = exports.WithContent = {
  render: () => /*#__PURE__*/_react.default.createElement(_EmptyState.default, {
    image: boxImage,
    title: "No instances found"
  }, /*#__PURE__*/_react.default.createElement("p", null, "There are no instances in this project. Spin up your first instance!"), /*#__PURE__*/_react.default.createElement("p", null, /*#__PURE__*/_react.default.createElement("a", {
    href: "#todo",
    target: "_blank"
  }, "How to create instances", /*#__PURE__*/_react.default.createElement(_Icon.default, {
    name: _Icon.ICONS.externalLink
  }))), /*#__PURE__*/_react.default.createElement(_Button.default, {
    appearance: "positive",
    onClick: doNothing
  }, "Create instance")),
  name: "With content"
};
