// @canonical/react-components@3.11.1/dist/components/PrefixedIpInput/PrefixedIpInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _PrefixedInput = _interopRequireDefault(require("../PrefixedInput"));
var _utils = require("./utils");
const _excluded = ["cidr", "help", "onIpChange", "ip", "name"];
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
const PrefixedIpInput = _ref => {
  let {
      cidr,
      help,
      onIpChange,
      ip,
      name
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  const [networkAddress] = cidr.split("/");
  const isIPV4 = (0, _utils.isIPv4)(networkAddress);
  const [immutable, editable] = (0, _utils.getImmutableAndEditable)(cidr);
  const inputValue = isIPV4 ? ip.split(".").slice(immutable.split(".").length).join(".") : ip.replace(immutable, "");
  const getIPv4MaxLength = () => {
    const immutableOctetsLength = immutable.split(".").length;
    const lengths = [15, 11, 7, 3]; // Corresponding to 0-3 immutable octets
    return lengths[immutableOctetsLength];
  };
  const maxLength = isIPV4 ? getIPv4MaxLength() : editable.length;
  const placeholder = props.disabled ? "" : editable;
  const setIp = editableValue => {
    const fullIp = editableValue ? isIPV4 ? "".concat(immutable, ".").concat(editableValue) : "".concat(immutable).concat(editableValue) : "";
    onIpChange(fullIp);
  };
  const handlePaste = e => {
    e.preventDefault();
    const pastedText = e.clipboardData.getData("text");
    if (isIPV4) {
      const octets = pastedText.split(".");
      const trimmed = octets.slice(0 - editable.split(".").length);
      const ip = trimmed.join(".");
      setIp(ip);
    } else {
      const ip = pastedText.replace(immutable, "");
      setIp(ip);
    }
  };
  return /*#__PURE__*/_react.default.createElement(_PrefixedInput.default, _extends({
    help: help ? help : /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, " ", isIPV4 ? /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, " ", "The available range in this subnet is", " ", /*#__PURE__*/_react.default.createElement("code", null, immutable, ".", editable, " ")) : /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, " ", "The available IPV6 address range is", " ", /*#__PURE__*/_react.default.createElement("code", null, immutable, editable, " ")), "."),
    immutableText: isIPV4 ? "".concat(immutable, ".") : immutable,
    maxLength: maxLength,
    name: name,
    onPaste: handlePaste,
    value: inputValue,
    onChange: e => {
      setIp(e.target.value);
    },
    placeholder: placeholder
  }, props));
};
var _default = exports.default = PrefixedIpInput;
