// @canonical/react-components@3.11.1/dist/components/Badge/Badge.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.UndefinedLargeNumber = exports.RoundedLargeNumber = exports.Default = exports.ColourCoding = void 0;
var _react = _interopRequireDefault(require("react"));
var _Badge = _interopRequireWildcard(require("./Badge"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const meta = {
  component: _Badge.default,
  tags: ["autodocs"],
  argTypes: {
    badgeType: {
      control: {
        options: _Badge.BadgeType,
        type: "select"
      }
    }
  },
  args: {
    badgeType: _Badge.BadgeType.UNDEFINED_LARGE_NUMBER
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default",
  args: {
    value: 9
  }
};
const ColourCoding = exports.ColourCoding = {
  render: () => /*#__PURE__*/_react.default.createElement(_Badge.default, {
    value: 9,
    isNegative: true
  }),
  name: "Colour Coding"
};

/**
 * When the amount of items is only relevant for small values, this variant should be used. When the value grows further than 999, the badge should show "999+".
 */
const UndefinedLargeNumber = exports.UndefinedLargeNumber = {
  render: () => /*#__PURE__*/_react.default.createElement(_Badge.default, {
    value: 1000,
    badgeType: _Badge.BadgeType.UNDEFINED_LARGE_NUMBER
  }),
  name: "Undefined Large Number"
};

/**
 * When the value of the badge represents a type of information where differences between large amounts of items are relevant, it should include a decimal unit prefix and round the value.
 */
const RoundedLargeNumber = exports.RoundedLargeNumber = {
  render: () => /*#__PURE__*/_react.default.createElement(_Badge.default, {
    value: 1234,
    badgeType: _Badge.BadgeType.ROUNDED_LARGE_NUMBER
  }),
  name: "Rounded Large Number"
};
