// @canonical/react-components@3.11.1/dist/components/Modal/Modal.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Focus = exports.Default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _Button = _interopRequireDefault(require("../Button"));
var _Modal = _interopRequireDefault(require("./Modal"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const Template = args => {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      height: "100vh",
      position: "relative"
    }
  }, /*#__PURE__*/_react.default.createElement(_Modal.default, args));
};
const meta = {
  component: _Modal.default,
  render: Template,
  tags: ["autodocs"],
  argTypes: {
    buttonRow: {
      control: {
        disable: true
      }
    },
    children: {
      control: {
        disable: true
      }
    },
    closeOnOutsideClick: {
      control: "boolean"
    }
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: _ref => {
    let {
      closeOnOutsideClick
    } = _ref;
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [modalOpen, setModalOpen] = (0, _react.useState)(true);
    const closeHandler = () => setModalOpen(false);
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
      onClick: () => setModalOpen(true)
    }, "Open modal"), modalOpen ? /*#__PURE__*/_react.default.createElement(_Modal.default, {
      close: closeHandler,
      title: "Confirm delete",
      closeOnOutsideClick: closeOnOutsideClick,
      buttonRow: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
        className: "u-no-margin--bottom",
        onClick: closeHandler
      }, "Cancel"), /*#__PURE__*/_react.default.createElement("button", {
        className: "p-button--negative u-no-margin--bottom"
      }, "Delete"))
    }, /*#__PURE__*/_react.default.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/_react.default.createElement("br", null), "You cannot undo this action.")) : null);
  },
  name: "Default"
};
const Focus = exports.Focus = {
  render: _ref2 => {
    let {
      closeOnOutsideClick
    } = _ref2;
    /* eslint-disable react-hooks/rules-of-hooks */
    const [modalOpen, setModalOpen] = (0, _react.useState)(true);
    const buttonRef = (0, _react.useRef)(null);
    /* eslint-enable react-hooks/rules-of-hooks */

    const closeHandler = () => setModalOpen(false);
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
      onClick: () => setModalOpen(true)
    }, "Open modal"), modalOpen ? /*#__PURE__*/_react.default.createElement(_Modal.default, {
      close: closeHandler,
      title: "Confirm delete",
      closeOnOutsideClick: closeOnOutsideClick,
      buttonRow: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
        className: "u-no-margin--bottom",
        onClick: closeHandler
      }, "Cancel"), /*#__PURE__*/_react.default.createElement("button", {
        className: "p-button--negative u-no-margin--bottom"
      }, "Delete")),
      focusRef: buttonRef
    }, /*#__PURE__*/_react.default.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/_react.default.createElement("br", null), "You cannot undo this action."), /*#__PURE__*/_react.default.createElement("p", null, /*#__PURE__*/_react.default.createElement(_Button.default, {
      appearance: "link",
      ref: buttonRef
    }, "More information"))) : null);
  },
  name: "Focus"
};
