// @canonical/react-components@3.8.2/dist/components/Card/Card.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Overlay = exports.Highlighted = exports.Header = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _ = _interopRequireDefault(require("./"));
var _Col = _interopRequireDefault(require("../Col"));
var _Row = _interopRequireDefault(require("../Row"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const meta = {
  component: _.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  args: {
    title: "We'd love to have you join us as a partner.",
    children: "If you are an independent software vendor or bundle author, it's easy to apply. You can find out more below."
  }
};
const Header = exports.Header = {
  args: {
    title: "Raspberry Pi2 and Pi3",
    thumbnail: "https://assets.ubuntu.com/v1/dca2e4c4-raspberry-logo.png",
    children: "For fun, for education and for profit, the RPi makes device development personal and entertaining. With support for both the Pi2 and the new Pi3, Ubuntu Core supports the world’s most beloved board."
  }
};
const Highlighted = exports.Highlighted = {
  args: {
    ...Default.args,
    highlighted: true
  }
};
const Overlay = exports.Overlay = {
  args: {
    title: "Web browsing",
    overlay: true,
    children: "Renowned for speed and security, Ubuntu and Firefox make browsing\nthe web a pleasure again. Ubuntu also includes Chrome, Opera and\nother browsers that can be installed from the Ubuntu Software\nCentre."
  },
  render: args => /*#__PURE__*/_react.default.createElement("section", {
    className: "p-strip--image is-light",
    style: {
      backgroundImage: "url('https://assets.ubuntu.com/v1/0a98afcd-screenshot_desktop.jpg')"
    }
  }, /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
    size: 6,
    emptyLarge: 7
  }, /*#__PURE__*/_react.default.createElement(_.default, args))))
};
