// @canonical/react-components@3.11.1/dist/esm/components/Panel/Panel.js
var _excluded = ["icon", "iconAlt", "name", "nameAlt", "component"],
  _excluded2 = ["forwardRef", "children", "className", "contentClassName", "controlsClassName", "controls", "dark", "header", "logo", "stickyHeader", "title", "titleClassName", "titleComponent", "titleId", "toggle", "wrapContent"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React from "react";
import classNames from "classnames";
import { isReactNode } from "../../utils";
var generateLogo = logo => {
  if (isReactNode(logo)) {
    return logo;
  }
  var {
      icon,
      iconAlt,
      name,
      nameAlt,
      component: Component = "a"
    } = logo,
    props = _objectWithoutProperties(logo, _excluded);
  return /*#__PURE__*/React.createElement(Component, _extends({
    className: "p-panel__logo"
  }, props), /*#__PURE__*/React.createElement("img", {
    className: "p-panel__logo-icon",
    src: icon,
    alt: iconAlt,
    width: "24",
    height: "24"
  }), /*#__PURE__*/React.createElement("img", {
    className: "p-panel__logo-name is-fading-when-collapsed",
    src: name,
    alt: nameAlt,
    height: "16"
  }));
};

/**
 * This is a [React](https://reactjs.org/) component for panels in the
 * [Vanilla](https://vanillaframework.io/docs/) layouts.
 *
 * The Panel component can be used in many areas of the application layout. It
 * can be the child element of `AppAside`, `AppMain`, `AppNavigation`, `AppNavigationBar`
 * and `AppStatus`.
 */
var Panel = _ref => {
  var {
      forwardRef,
      children,
      className,
      contentClassName,
      controlsClassName,
      controls,
      dark,
      header,
      logo,
      stickyHeader,
      title,
      titleClassName,
      titleComponent: TitleComponent = "h4",
      titleId,
      toggle,
      wrapContent = true
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded2);
  return /*#__PURE__*/React.createElement("div", _extends({}, props, {
    className: classNames("p-panel", className, {
      "is-dark": dark
    }),
    ref: forwardRef
  }), logo || title || controls || toggle ? /*#__PURE__*/React.createElement("div", {
    className: classNames("p-panel__header", {
      "is-sticky": stickyHeader
    })
  }, logo ? generateLogo(logo) : /*#__PURE__*/React.createElement(TitleComponent, {
    className: classNames("p-panel__title", titleClassName),
    id: titleId
  }, title), /*#__PURE__*/React.createElement("div", {
    className: classNames("p-panel__controls", controlsClassName)
  }, toggle ? /*#__PURE__*/React.createElement("span", {
    role: "button",
    tabIndex: 0,
    className: "p-panel__toggle",
    onClick: () => toggle.onClick(),
    onKeyDown: () => toggle.onClick()
  }, toggle.label) : null, controls)) : header, children && wrapContent ? /*#__PURE__*/React.createElement("div", {
    className: classNames("p-panel__content", contentClassName)
  }, children) : children);
};
export default Panel;
