// @canonical/react-components@3.11.1/dist/components/ColumnSelector/ColumnSelector.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _ColumnSelector = _interopRequireDefault(require("./ColumnSelector"));
var _MainTable = _interopRequireDefault(require("../MainTable"));
var _ColumnSelectorStoriesHelper = require("./ColumnSelectorStoriesHelper");
var _columnSelectorHelper = require("./columnSelectorHelper");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const meta = {
  component: _ColumnSelector.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [userHidden, setUserHidden] = (0, _react.useState)([_ColumnSelectorStoriesHelper.FILESYSTEM, _ColumnSelectorStoriesHelper.SNAPSHOTS, _ColumnSelectorStoriesHelper.CLUSTER_MEMBER]);
    const sizeHidden = [_ColumnSelectorStoriesHelper.DESCRIPTION];
    const hiddenColumns = userHidden.concat(sizeHidden);
    const ROWS = (0, _ColumnSelectorStoriesHelper.getRows)();
    const setUserHiddenAndSaveInLocalStorage = columns => {
      setUserHidden(columns);
    };
    const displayedRows = (0, _columnSelectorHelper.visibleRowColumns)(ROWS, hiddenColumns);
    const displayedHeaders = (0, _columnSelectorHelper.visibleHeaderColumns)(_ColumnSelectorStoriesHelper.COLUMN_HEADERS.map(t => {
      return {
        content: t,
        sortKey: t
      };
    }), hiddenColumns);
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_ColumnSelector.default, {
      columns: _ColumnSelectorStoriesHelper.USER_HIDEABLE_COLUMNS,
      userHidden: userHidden,
      sizeHidden: sizeHidden,
      setUserHidden: setUserHiddenAndSaveInLocalStorage
    }), /*#__PURE__*/_react.default.createElement(_MainTable.default, {
      rows: displayedRows,
      headers: displayedHeaders
    }));
  },
  name: "Default"
};
