// @canonical/react-components@4.5.4/dist/esm/components/MultiSelect/MultiSelect.stories.js
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
import React from "react";
import { useState } from "react";
import { Formik } from "formik";
import { FormikField } from "../../index";
import { MultiSelect } from "./MultiSelect";
var Template = props => {
  var [selectedItems, setSelectedItems] = useState(props.selectedItems || []);
  return /*#__PURE__*/React.createElement(MultiSelect, _extends({}, props, {
    selectedItems: selectedItems,
    onItemsUpdate: setSelectedItems
  }));
};
var meta = {
  component: MultiSelect,
  render: Template,
  tags: ["autodocs"],
  parameters: {}
};
export default meta;
var groupedItems = [{
  label: "Almond",
  value: "almond",
  group: "Nuts"
}, {
  label: "Cashew",
  value: "cashew",
  group: "Nuts"
}, {
  label: "Mango",
  value: "mango",
  group: "Fruit"
}, {
  label: "Peach",
  value: "peach",
  group: "Fruit"
}];
export var CondensedExample = {
  args: {
    items: [...Array.from({
      length: 26
    }, (_, i) => ({
      label: "".concat(String.fromCharCode(i + 65)),
      value: "".concat(String.fromCharCode(i + 65))
    })), ...Array.from({
      length: 26
    }, (_, i) => ({
      label: "Item ".concat(i + 1),
      value: i + 1
    }))],
    selectedItems: [{
      label: "A",
      value: "A"
    }, {
      label: "Item 2",
      value: 2
    }],
    variant: "condensed"
  }
};
export var SearchExample = {
  args: _objectSpread(_objectSpread({}, CondensedExample.args), {}, {
    variant: "search",
    items: [...CondensedExample.args.items.map((item, i) => _objectSpread(_objectSpread({}, item), {}, {
      group: i % 2 === 0 ? "Group 1" : "Group 2"
    }))],
    id: "search-id"
  })
};
export var WithDisabledItems = {
  args: _objectSpread(_objectSpread({}, CondensedExample.args), {}, {
    disabledItems: [{
      label: "Item 1",
      value: 1
    }, {
      label: "Item 2",
      value: 2
    }]
  })
};
export var WithoutSorting = {
  args: {
    items: [{
      label: "Sunday",
      value: 0
    }, {
      label: "Monday",
      value: 1
    }, {
      label: "Tuesday",
      value: 2
    }, {
      label: "Wednesday",
      value: 3
    }, {
      label: "Thursday",
      value: 4
    }, {
      label: "Friday",
      value: 5
    }, {
      label: "Saturday",
      value: 6
    }],
    selectedItems: [{
      label: "Monday",
      value: 1
    }, {
      label: "Wednesday",
      value: 3
    }, {
      label: "Friday",
      value: 5
    }],
    variant: "condensed",
    isSortedAlphabetically: false,
    hasSelectedItemsFirst: false,
    id: "external-id"
  }
};
export var Disabled = {
  args: _objectSpread(_objectSpread({}, CondensedExample.args), {}, {
    disabled: true
  })
};
export var HelpText = {
  args: _objectSpread(_objectSpread({}, CondensedExample.args), {}, {
    help: /*#__PURE__*/React.createElement("span", null, "This is a help text, that should appear underneath the component.")
  })
};
var FormikCallbacksAndEmptyStateTemplate = () => {
  var [selectedItems, setSelectedItems] = useState([]);
  var [events, setEvents] = useState([]);
  var addEvent = eventName => {
    setEvents(previousEvents => [eventName, ...previousEvents].slice(0, 6));
  };
  return /*#__PURE__*/React.createElement("div", {
    style: {
      maxWidth: "28rem"
    }
  }, /*#__PURE__*/React.createElement(Formik, {
    initialValues: {
      ingredients: ""
    },
    onSubmit: () => {}
  }, /*#__PURE__*/React.createElement(FormikField, {
    name: "ingredients",
    component: MultiSelect,
    label: "Ingredients",
    variant: "search",
    placeholder: "Search ingredients",
    items: groupedItems,
    selectedItems: selectedItems,
    onItemsUpdate: setSelectedItems,
    onSearchChange: value => {
      addEvent("onSearchChange(\"".concat(value, "\")"));
    },
    onOpen: () => addEvent("onOpen()"),
    onClose: () => addEvent("onClose()"),
    emptyMessage: "No ingredients found"
  })), /*#__PURE__*/React.createElement("p", {
    style: {
      marginBottom: "0.5rem"
    }
  }, "Callback log:"), /*#__PURE__*/React.createElement("ul", {
    style: {
      margin: 0,
      paddingLeft: "1.25rem"
    }
  }, events.map((event, index) => /*#__PURE__*/React.createElement("li", {
    key: "".concat(event, "-").concat(index)
  }, event))));
};
export var FormikCallbacksAndEmptyState = {
  render: FormikCallbacksAndEmptyStateTemplate
};
export var EmptyMessage = {
  args: {
    variant: "search",
    items: groupedItems,
    emptyMessage: "No matching ingredients.",
    placeholder: "Try typing kiwi"
  }
};
export var EmptyStateNode = {
  args: {
    variant: "search",
    items: groupedItems,
    placeholder: "Try typing kiwi",
    emptyState: /*#__PURE__*/React.createElement("div", {
      className: "u-align--center",
      style: {
        padding: "0.75rem 1rem"
      }
    }, /*#__PURE__*/React.createElement("strong", null, "No ingredient matches."), /*#__PURE__*/React.createElement("p", {
      style: {
        margin: "0.25rem 0 0"
      }
    }, "Use a broader term."))
  }
};
