// @canonical/react-components@3.11.1/dist/components/ConfirmationModal/ConfirmationModal.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Positive = exports.NoTitle = exports.Extra = exports.Default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _ConfirmationModal = _interopRequireDefault(require("./ConfirmationModal"));
var _Input = _interopRequireDefault(require("../Input"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const doNothing = () => {};
const meta = {
  component: _ConfirmationModal.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [modalOpen, setModalOpen] = (0, _react.useState)(false);
    const closeHandler = () => setModalOpen(false);
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
      onClick: () => setModalOpen(true)
    }, "Open confirmation modal"), modalOpen ? /*#__PURE__*/_react.default.createElement(_ConfirmationModal.default, {
      title: "Confirm delete",
      confirmButtonLabel: "Delete",
      onConfirm: doNothing,
      close: closeHandler
    }, /*#__PURE__*/_react.default.createElement("p", null, "This will permanently delete the user \"Simon\".", /*#__PURE__*/_react.default.createElement("br", null), "You cannot undo this action.")) : null);
  },
  name: "Default"
};

/**
 * The title is optional, and the content is fully customisable.
 */
const NoTitle = exports.NoTitle = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [modalOpen, setModalOpen] = (0, _react.useState)(false);
    const closeHandler = () => setModalOpen(false);
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
      onClick: () => setModalOpen(true)
    }, "Run"), modalOpen ? /*#__PURE__*/_react.default.createElement(_ConfirmationModal.default, {
      close: closeHandler,
      confirmButtonAppearance: "positive",
      confirmButtonLabel: "Confirm",
      onConfirm: doNothing
    }, /*#__PURE__*/_react.default.createElement("h4", {
      style: {
        margin: "1rem 0"
      }
    }, "Run list-backups?"), /*#__PURE__*/_react.default.createElement("div", {
      style: {
        width: "480px",
        marginBottom: "1rem"
      }
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "u-text--muted"
    }, "UNIT COUNT"), /*#__PURE__*/_react.default.createElement("div", null, "1")), /*#__PURE__*/_react.default.createElement("div", {
      style: {
        marginBottom: "1.5rem"
      }
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "u-text--muted"
    }, "UNIT NAME"), /*#__PURE__*/_react.default.createElement("div", null, "mysql/0"))) : null);
  },
  name: "No title"
};

/**
 * The appearance of the confirm button can be any of the `Button`'s appearance values.
 */
const Positive = exports.Positive = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [modalOpen, setModalOpen] = (0, _react.useState)(false);
    const closeHandler = () => setModalOpen(false);
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
      onClick: () => setModalOpen(true)
    }, "Apply changes"), modalOpen ? /*#__PURE__*/_react.default.createElement(_ConfirmationModal.default, {
      close: closeHandler,
      confirmButtonAppearance: "positive",
      confirmButtonLabel: "Yes, apply changes",
      onConfirm: doNothing
    }, /*#__PURE__*/_react.default.createElement("h4", {
      style: {
        margin: "1rem 0"
      }
    }, "Are you sure you wish to apply these changes?"), /*#__PURE__*/_react.default.createElement("p", null, "You have edited the following values to the mysql configuration:"), /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("h5", {
      className: "u-no-margin--bottom"
    }, "cluster-name"), /*#__PURE__*/_react.default.createElement("pre", {
      className: "u-no-padding",
      style: {
        backgroundColor: "transparent"
      }
    }, "test")), /*#__PURE__*/_react.default.createElement("div", {
      className: "p-text--small",
      style: {
        width: "480px"
      }
    }, "You can revert back to the applications default settings by clicking the \u201CReset all values\u201D button; or reset each edited field by clicking \u201CUse default\u201D.")) : null);
  },
  name: "Positive"
};

/**
 * Extra elements can be added to the button row.
 */
const Extra = exports.Extra = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [modalOpen, setModalOpen] = (0, _react.useState)(false);
    const closeHandler = () => setModalOpen(false);
    return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("button", {
      onClick: () => setModalOpen(true)
    }, "Stop"), modalOpen ? /*#__PURE__*/_react.default.createElement(_ConfirmationModal.default, {
      confirmExtra: /*#__PURE__*/_react.default.createElement("span", {
        className: "u-float-left"
      }, /*#__PURE__*/_react.default.createElement(_Input.default, {
        type: "checkbox",
        id: "extraCheckbox",
        label: "Force stop",
        tabIndex: -1
      })),
      title: "Confirm stop",
      confirmButtonLabel: "Stop",
      onConfirm: doNothing,
      close: closeHandler
    }, /*#__PURE__*/_react.default.createElement("p", null, "This will stop instance", /*#__PURE__*/_react.default.createElement("b", null, "whimsical-mouflon"), ".")) : null);
  },
  name: "Extra"
};
