// @canonical/react-components@3.11.1/dist/components/StatusLabel/StatusLabel.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Positive = exports.Negative = exports.Information = exports.Default = exports.Caution = void 0;
var _react = _interopRequireDefault(require("react"));
var _StatusLabel = _interopRequireWildcard(require("./StatusLabel"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const meta = {
  component: _StatusLabel.default,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: {
        type: "text"
      }
    },
    appearance: {
      control: {
        options: _StatusLabel.StatusLabelAppearance,
        type: "select"
      }
    }
  },
  args: {
    children: "Label",
    appearance: _StatusLabel.StatusLabelAppearance.POSITIVE
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: () => /*#__PURE__*/_react.default.createElement(_StatusLabel.default, null, "Default"),
  name: "Default"
};
const Positive = exports.Positive = {
  render: () => /*#__PURE__*/_react.default.createElement(_StatusLabel.default, {
    appearance: "positive"
  }, "Positive"),
  name: "Positive"
};
const Caution = exports.Caution = {
  render: () => /*#__PURE__*/_react.default.createElement(_StatusLabel.default, {
    appearance: "caution"
  }, "Caution"),
  name: "Caution"
};
const Negative = exports.Negative = {
  render: () => /*#__PURE__*/_react.default.createElement(_StatusLabel.default, {
    appearance: "negative"
  }, "Negative"),
  name: "Negative"
};
const Information = exports.Information = {
  render: () => /*#__PURE__*/_react.default.createElement(_StatusLabel.default, {
    appearance: "information"
  }, "Information"),
  name: "Information"
};
