// @canonical/react-components@3.11.1/dist/components/Tooltip/Tooltip.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.TargetingElements = exports.FollowMouse = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _Button = _interopRequireDefault(require("../Button"));
var _Tooltip = _interopRequireWildcard(require("./Tooltip"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const Template = args => {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: "flex",
      justifyContent: "center",
      padding: "3rem"
    }
  }, /*#__PURE__*/_react.default.createElement(_Tooltip.default, args, /*#__PURE__*/_react.default.createElement(_Button.default, {
    className: "u-no-margin--bottom"
  }, "Hover over me!")));
};
const meta = {
  component: _Tooltip.default,
  render: Template,
  tags: ["autodocs"],
  argTypes: {
    position: {
      control: {
        type: "radio"
      },
      options: Object.values(_Tooltip.position)
    }
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: Template.bind({}),
  name: "Default",
  args: {
    message: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, "Tooltip", " ", /*#__PURE__*/_react.default.createElement("a", {
      href: "#todo",
      target: "_blank"
    }, "Link1"))
  }
};
const FollowMouse = exports.FollowMouse = {
  render: Template.bind({}),
  name: "Follow mouse",
  args: {
    followMouse: true,
    message: "Tooltip message to display.",
    position: "top-center"
  }
};

/**
 * The tooltip elements are displayed inside a [Portal](https://reactjs.org/docs/portals.html) so that the tooltip is not constrained by its current position in the DOM.
 * Sometimes however, you need to target the tooltip elements. This can be achieved by passing class names to the tooltip elements. You can also pass classes to the elements that trigger and position the tooltip.
 */
const TargetingElements = exports.TargetingElements = {
  render: Template.bind({}),
  name: "Targeting elements",
  args: {
    className: "custom-wrapping-class",
    positionElementClassName: "custom-position-class",
    tooltipClassName: "custom-tooltip-class",
    message: "Tooltip message to display."
  }
};
