// @canonical/react-components@3.11.1/dist/esm/components/PrefixedIpInput/PrefixedIpInput.stories.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import React, { useState } from "react";
import PrefixedIpInput from "./PrefixedIpInput";
var PrefixedIpInputWrapper = args => {
  var [ip, setIp] = useState(args.ip);
  return /*#__PURE__*/React.createElement(PrefixedIpInput, _extends({}, args, {
    ip: ip,
    onIpChange: newIp => {
      var _args$onIpChange;
      setIp(newIp);
      (_args$onIpChange = args.onIpChange) === null || _args$onIpChange === void 0 || _args$onIpChange.call(args, newIp);
    }
  }));
};
var meta = {
  component: PrefixedIpInput,
  tags: ["autodocs"],
  argTypes: {
    ip: {
      control: "text"
    },
    cidr: {
      control: "text"
    },
    label: {
      control: "text"
    },
    name: {
      control: "text"
    },
    error: {
      control: "text"
    },
    help: {
      control: "text"
    },
    disabled: {
      control: "boolean"
    },
    required: {
      control: "boolean"
    }
  },
  render: args => /*#__PURE__*/React.createElement(PrefixedIpInputWrapper, args)
};
export default meta;
export var IPv4Default = {
  name: "IPv4 default",
  args: {
    cidr: "192.168.1.0/24",
    ip: "",
    name: "ip-address",
    label: "IP Address",
    onIpChange: ip => console.log("IP changed:", ip)
  }
};
export var IPv4WithValue = {
  name: "IPv4 with value",
  args: {
    cidr: "192.168.1.0/24",
    ip: "192.168.1.100",
    name: "ip-address",
    label: "IP Address",
    onIpChange: ip => console.log("IP changed:", ip)
  }
};
export var IPv4WithError = {
  name: "IPv4 with error",
  args: {
    cidr: "192.168.1.0/24",
    ip: "192.168.1.256",
    name: "ip-address",
    label: "IP Address",
    error: "Invalid IP address",
    onIpChange: ip => console.log("IP changed:", ip)
  }
};
export var IPv4Disabled = {
  name: "IPv4 disabled",
  args: {
    cidr: "192.168.1.0/24",
    ip: "192.168.1.50",
    name: "ip-address",
    label: "IP Address",
    disabled: true,
    onIpChange: ip => console.log("IP changed:", ip)
  }
};
export var IPv6Default = {
  name: "IPv6 default",
  args: {
    cidr: "2001:db8::/32",
    ip: "",
    name: "ipv6-address",
    label: "IPv6 Address",
    onIpChange: ip => console.log("IP changed:", ip)
  }
};
export var IPv6WithValue = {
  name: "IPv6 with value",
  args: {
    cidr: "2001:db8::/32",
    ip: "2001:db8::1",
    name: "ipv6-address",
    label: "IPv6 Address",
    onIpChange: ip => console.log("IP changed:", ip)
  }
};
export var WithCustomHelp = {
  args: {
    cidr: "10.0.0.0/16",
    ip: "",
    name: "ip-address",
    label: "IP Address",
    help: "Enter a custom IP address for this device",
    onIpChange: ip => console.log("IP changed:", ip)
  }
};
export var Required = {
  args: {
    cidr: "192.168.0.0/24",
    ip: "",
    name: "ip-address",
    label: "IP Address",
    required: true,
    onIpChange: ip => console.log("IP changed:", ip)
  }
};
