// @canonical/react-components@3.11.1/dist/esm/components/Select/Select.js
var _excluded = ["label", "value"],
  _excluded2 = ["caution", "className", "error", "help", "helpClassName", "id", "label", "labelClassName", "onChange", "options", "required", "stacked", "success", "takeFocus", "wrapperClassName"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import classNames from "classnames";
import React, { useEffect, useId, useRef } from "react";
import Field from "../Field";

/**
 * The props for the Select component.
 */

var generateOptions = options => options === null || options === void 0 ? void 0 : options.map(_ref => {
  var {
      label,
      value
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/React.createElement("option", _extends({
    value: value,
    key: "".concat(value) || label
  }, props), label);
});

/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Select](https://docs.vanillaframework.io/base/forms/#select).
 *
 * Use the Select component to create a drop-down list.
 */
var Select = _ref2 => {
  var {
      caution,
      className,
      error,
      help,
      helpClassName,
      id,
      label,
      labelClassName,
      onChange: _onChange,
      options,
      required,
      stacked,
      success,
      takeFocus,
      wrapperClassName
    } = _ref2,
    selectProps = _objectWithoutProperties(_ref2, _excluded2);
  var selectRef = useRef(null);
  var validationId = useId();
  var defaultSelectId = useId();
  var selectId = id || defaultSelectId;
  var helpId = useId();
  var hasError = !!error;
  useEffect(() => {
    if (takeFocus) {
      selectRef.current.focus();
    }
  }, [takeFocus]);
  return /*#__PURE__*/React.createElement(Field, {
    caution: caution,
    className: wrapperClassName,
    error: error,
    forId: selectId,
    help: help,
    helpClassName: helpClassName,
    helpId: helpId,
    isSelect: true,
    label: label,
    labelClassName: labelClassName,
    required: required,
    stacked: stacked,
    success: success,
    validationId: validationId
  }, /*#__PURE__*/React.createElement("select", _extends({
    "aria-describedby": [help ? helpId : null, success ? validationId : null].filter(Boolean).join(" "),
    "aria-errormessage": hasError ? validationId : null,
    "aria-invalid": hasError,
    className: classNames("p-form-validation__input", className),
    id: selectId,
    onChange: evt => _onChange && _onChange(evt),
    ref: selectRef,
    required: required
  }, selectProps), generateOptions(options)));
};
export default Select;
