// @canonical/react-components@3.11.1/dist/esm/components/Accordion/Accordion.js
var _excluded = ["key"],
  _excluded2 = ["className", "expanded", "externallyControlled", "onExpandedChange", "sections", "titleElement"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import classNames from "classnames";
import React, { useState } from "react";
import AccordionSection from "./AccordionSection";
var generateSections = (sections, setExpanded, expanded, titleElement) => sections.map((_ref, i) => {
  var {
      key
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/React.createElement(AccordionSection, _extends({
    expanded: expanded,
    key: key || i,
    sectionKey: key === null || key === void 0 ? void 0 : key.toString(),
    setExpanded: setExpanded,
    titleElement: titleElement
  }, props));
});

/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Accordion](https://docs.vanillaframework.io/patterns/accordion/).
 *
 * The sidebar accordion, used in listing pages or as navigation, can hold multiple navigation items or to be used as a filter for content.
 */
var Accordion = _ref2 => {
  var {
      className,
      expanded,
      externallyControlled,
      onExpandedChange,
      sections,
      titleElement
    } = _ref2,
    asideProps = _objectWithoutProperties(_ref2, _excluded2);
  var [expandedSection, setExpandedSection] = useState(expanded);
  var setExpanded = (id, title) => {
    setExpandedSection(id);
    onExpandedChange && onExpandedChange(id, title);
  };
  return /*#__PURE__*/React.createElement("aside", _extends({
    className: classNames(className, "p-accordion")
  }, asideProps, {
    role: "tablist",
    "aria-multiselectable": "true"
  }), /*#__PURE__*/React.createElement("ul", {
    className: "p-accordion__list"
  }, generateSections(sections, setExpanded, externallyControlled ? expanded : expandedSection, titleElement)));
};
export default Accordion;
