// @canonical/landscape-openapi@0.0.60/generated/sdk.gen.ts
// This file is auto-generated by @hey-api/openapi-ts

import type { Client, Options as Options2, TDataShape } from "./client";
import { client } from "./client.gen";
import {
  batchGetMirrorsResponseTransformer,
  createMirrorResponseTransformer,
  createPublicationResponseTransformer,
  getMirrorResponseTransformer,
  getPublicationResponseTransformer,
  listMirrorsResponseTransformer,
  listPublicationsResponseTransformer,
  updateMirrorResponseTransformer,
  updatePublicationResponseTransformer,
} from "./transformers.gen";
import type {
  BatchGetLocalsData,
  BatchGetLocalsErrors,
  BatchGetLocalsResponses,
  BatchGetMirrorsData,
  BatchGetMirrorsErrors,
  BatchGetMirrorsResponses,
  BatchGetPublicationTargetsData,
  BatchGetPublicationTargetsErrors,
  BatchGetPublicationTargetsResponses,
  CreateLocalData,
  CreateLocalErrors,
  CreateLocalResponses,
  CreateMirrorData,
  CreateMirrorErrors,
  CreateMirrorResponses,
  CreatePublicationData,
  CreatePublicationErrors,
  CreatePublicationResponses,
  CreatePublicationTargetData,
  CreatePublicationTargetErrors,
  CreatePublicationTargetResponses,
  DeleteLocalData,
  DeleteLocalErrors,
  DeleteLocalResponses,
  DeleteMirrorData,
  DeleteMirrorErrors,
  DeleteMirrorResponses,
  DeletePublicationData,
  DeletePublicationErrors,
  DeletePublicationResponses,
  DeletePublicationTargetData,
  DeletePublicationTargetErrors,
  DeletePublicationTargetResponses,
  GetLocalData,
  GetLocalErrors,
  GetLocalResponses,
  GetMirrorData,
  GetMirrorErrors,
  GetMirrorResponses,
  GetOperationData,
  GetOperationErrors,
  GetOperationResponses,
  GetPublicationData,
  GetPublicationErrors,
  GetPublicationResponses,
  GetPublicationTargetData,
  GetPublicationTargetErrors,
  GetPublicationTargetResponses,
  ImportLocalPackagesData,
  ImportLocalPackagesErrors,
  ImportLocalPackagesResponses,
  ListLocalPackagesData,
  ListLocalPackagesErrors,
  ListLocalPackagesResponses,
  ListLocalsData,
  ListLocalsErrors,
  ListLocalsResponses,
  ListMirrorPackagesData,
  ListMirrorPackagesErrors,
  ListMirrorPackagesResponses,
  ListMirrorsData,
  ListMirrorsErrors,
  ListMirrorsResponses,
  ListPublicationsData,
  ListPublicationsErrors,
  ListPublicationsResponses,
  ListPublicationTargetsData,
  ListPublicationTargetsErrors,
  ListPublicationTargetsResponses,
  PublishPublicationData,
  PublishPublicationErrors,
  PublishPublicationResponses,
  SyncMirrorData,
  SyncMirrorErrors,
  SyncMirrorResponses,
  UpdateLocalData,
  UpdateLocalErrors,
  UpdateLocalResponses,
  UpdateMirrorData,
  UpdateMirrorErrors,
  UpdateMirrorResponses,
  UpdatePublicationData,
  UpdatePublicationErrors,
  UpdatePublicationResponses,
  UpdatePublicationTargetData,
  UpdatePublicationTargetErrors,
  UpdatePublicationTargetResponses,
} from "./types.gen";

export type Options<
  TData extends TDataShape = TDataShape,
  ThrowOnError extends boolean = boolean,
  TResponse = unknown,
> = Options2<TData, ThrowOnError, TResponse> & {
  /**
   * You can provide a client instance returned by `createClient()` instead of
   * individual options. This might be also useful if you want to implement a
   * custom client.
   */
  client?: Client;
  /**
   * You can pass arbitrary values through the `meta` object. This can be
   * used to access values that aren't defined as part of the SDK function.
   */
  meta?: Record<string, unknown>;
};

/**
 * ListLocals returns a list of Locals.
 */
export const listLocals = <ThrowOnError extends boolean = false>(
  options?: Options<ListLocalsData, ThrowOnError>,
) =>
  (options?.client ?? client).get<
    ListLocalsResponses,
    ListLocalsErrors,
    ThrowOnError
  >({ url: "/v1beta1/locals", ...options });

/**
 * CreateLocal creates a new Local repository.
 */
export const createLocal = <ThrowOnError extends boolean = false>(
  options: Options<CreateLocalData, ThrowOnError>,
) =>
  (options.client ?? client).post<
    CreateLocalResponses,
    CreateLocalErrors,
    ThrowOnError
  >({
    url: "/v1beta1/locals",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * BatchGetLocals retrieves a list of locals based on their name.
 */
export const batchGetLocals = <ThrowOnError extends boolean = false>(
  options: Options<BatchGetLocalsData, ThrowOnError>,
) =>
  (options.client ?? client).post<
    BatchGetLocalsResponses,
    BatchGetLocalsErrors,
    ThrowOnError
  >({
    url: "/v1beta1/locals:batchGet",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * ListMirrors returns a list of Mirrors.
 */
export const listMirrors = <ThrowOnError extends boolean = false>(
  options?: Options<ListMirrorsData, ThrowOnError>,
) =>
  (options?.client ?? client).get<
    ListMirrorsResponses,
    ListMirrorsErrors,
    ThrowOnError
  >({
    responseTransformer: listMirrorsResponseTransformer,
    url: "/v1beta1/mirrors",
    ...options,
  });

/**
 * CreateMirror creates a new Mirror.
 * It validates the upstream but does not initiate synchronization.
 */
export const createMirror = <ThrowOnError extends boolean = false>(
  options: Options<CreateMirrorData, ThrowOnError>,
) =>
  (options.client ?? client).post<
    CreateMirrorResponses,
    CreateMirrorErrors,
    ThrowOnError
  >({
    responseTransformer: createMirrorResponseTransformer,
    url: "/v1beta1/mirrors",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * BatchGetMirrors retrieves a list of mirrors based on their name.
 */
export const batchGetMirrors = <ThrowOnError extends boolean = false>(
  options: Options<BatchGetMirrorsData, ThrowOnError>,
) =>
  (options.client ?? client).post<
    BatchGetMirrorsResponses,
    BatchGetMirrorsErrors,
    ThrowOnError
  >({
    responseTransformer: batchGetMirrorsResponseTransformer,
    url: "/v1beta1/mirrors:batchGet",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * ListPublicationTargets lists all PublicationTargets.
 */
export const listPublicationTargets = <ThrowOnError extends boolean = false>(
  options?: Options<ListPublicationTargetsData, ThrowOnError>,
) =>
  (options?.client ?? client).get<
    ListPublicationTargetsResponses,
    ListPublicationTargetsErrors,
    ThrowOnError
  >({ url: "/v1beta1/publicationTargets", ...options });

/**
 * CreatePublicationTarget creates a new PublicationTarget.
 */
export const createPublicationTarget = <ThrowOnError extends boolean = false>(
  options: Options<CreatePublicationTargetData, ThrowOnError>,
) =>
  (options.client ?? client).post<
    CreatePublicationTargetResponses,
    CreatePublicationTargetErrors,
    ThrowOnError
  >({
    url: "/v1beta1/publicationTargets",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * BatchGetPublicationTargets retrieves a list of publication targets based on their name.
 */
export const batchGetPublicationTargets = <
  ThrowOnError extends boolean = false,
>(
  options: Options<BatchGetPublicationTargetsData, ThrowOnError>,
) =>
  (options.client ?? client).post<
    BatchGetPublicationTargetsResponses,
    BatchGetPublicationTargetsErrors,
    ThrowOnError
  >({
    url: "/v1beta1/publicationTargets:batchGet",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * ListPublications returns a list of Publications.
 */
export const listPublications = <ThrowOnError extends boolean = false>(
  options?: Options<ListPublicationsData, ThrowOnError>,
) =>
  (options?.client ?? client).get<
    ListPublicationsResponses,
    ListPublicationsErrors,
    ThrowOnError
  >({
    responseTransformer: listPublicationsResponseTransformer,
    url: "/v1beta1/publications",
    ...options,
  });

/**
 * CreatePublication creates a new Publication.
 */
export const createPublication = <ThrowOnError extends boolean = false>(
  options: Options<CreatePublicationData, ThrowOnError>,
) =>
  (options.client ?? client).post<
    CreatePublicationResponses,
    CreatePublicationErrors,
    ThrowOnError
  >({
    responseTransformer: createPublicationResponseTransformer,
    url: "/v1beta1/publications",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * UpdateLocal updates the details of a Local.
 */
export const updateLocal = <ThrowOnError extends boolean = false>(
  options: Options<UpdateLocalData, ThrowOnError>,
) =>
  (options.client ?? client).patch<
    UpdateLocalResponses,
    UpdateLocalErrors,
    ThrowOnError
  >({
    url: "/v1beta1/{local.name}",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * UpdateMirror updates the details of a Mirror.
 */
export const updateMirror = <ThrowOnError extends boolean = false>(
  options: Options<UpdateMirrorData, ThrowOnError>,
) =>
  (options.client ?? client).patch<
    UpdateMirrorResponses,
    UpdateMirrorErrors,
    ThrowOnError
  >({
    responseTransformer: updateMirrorResponseTransformer,
    url: "/v1beta1/{mirror.name}",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * DeleteMirror deletes a Mirror.
 */
export const deleteMirror = <ThrowOnError extends boolean = false>(
  options: Options<DeleteMirrorData, ThrowOnError>,
) =>
  (options.client ?? client).delete<
    DeleteMirrorResponses,
    DeleteMirrorErrors,
    ThrowOnError
  >({ url: "/v1beta1/{name_1}", ...options });

/**
 * GetMirror retrieves a Mirror by its resource name.
 */
export const getMirror = <ThrowOnError extends boolean = false>(
  options: Options<GetMirrorData, ThrowOnError>,
) =>
  (options.client ?? client).get<
    GetMirrorResponses,
    GetMirrorErrors,
    ThrowOnError
  >({
    responseTransformer: getMirrorResponseTransformer,
    url: "/v1beta1/{name_1}",
    ...options,
  });

/**
 * DeletePublication deletes a Publication.
 * Note: This operation may trigger a background Task to clean up published artifacts.
 */
export const deletePublication = <ThrowOnError extends boolean = false>(
  options: Options<DeletePublicationData, ThrowOnError>,
) =>
  (options.client ?? client).delete<
    DeletePublicationResponses,
    DeletePublicationErrors,
    ThrowOnError
  >({ url: "/v1beta1/{name_2}", ...options });

/**
 * Gets the latest state of a long-running operation.  Clients can use this
 * method to poll the operation result at intervals as recommended by the API
 * service.
 */
export const getOperation = <ThrowOnError extends boolean = false>(
  options: Options<GetOperationData, ThrowOnError>,
) =>
  (options.client ?? client).get<
    GetOperationResponses,
    GetOperationErrors,
    ThrowOnError
  >({ url: "/v1beta1/{name_2}", ...options });

/**
 * DeletePublicationTarget deletes a PublicationTarget.
 */
export const deletePublicationTarget = <ThrowOnError extends boolean = false>(
  options: Options<DeletePublicationTargetData, ThrowOnError>,
) =>
  (options.client ?? client).delete<
    DeletePublicationTargetResponses,
    DeletePublicationTargetErrors,
    ThrowOnError
  >({ url: "/v1beta1/{name_3}", ...options });

/**
 * GetPublication retrieves a Publication by its resource name.
 */
export const getPublication = <ThrowOnError extends boolean = false>(
  options: Options<GetPublicationData, ThrowOnError>,
) =>
  (options.client ?? client).get<
    GetPublicationResponses,
    GetPublicationErrors,
    ThrowOnError
  >({
    responseTransformer: getPublicationResponseTransformer,
    url: "/v1beta1/{name_3}",
    ...options,
  });

/**
 * GetPublicationTarget gets details of a single PublicationTarget.
 */
export const getPublicationTarget = <ThrowOnError extends boolean = false>(
  options: Options<GetPublicationTargetData, ThrowOnError>,
) =>
  (options.client ?? client).get<
    GetPublicationTargetResponses,
    GetPublicationTargetErrors,
    ThrowOnError
  >({ url: "/v1beta1/{name_4}", ...options });

/**
 * DeleteLocal deletes a Local.
 */
export const deleteLocal = <ThrowOnError extends boolean = false>(
  options: Options<DeleteLocalData, ThrowOnError>,
) =>
  (options.client ?? client).delete<
    DeleteLocalResponses,
    DeleteLocalErrors,
    ThrowOnError
  >({ url: "/v1beta1/{name}", ...options });

/**
 * GetLocal retrieves a Local by its resource name.
 */
export const getLocal = <ThrowOnError extends boolean = false>(
  options: Options<GetLocalData, ThrowOnError>,
) =>
  (options.client ?? client).get<
    GetLocalResponses,
    GetLocalErrors,
    ThrowOnError
  >({ url: "/v1beta1/{name}", ...options });

/**
 * ImportLocalPackages imports packages from a URL pointing to a package list or index
 * into the Local. This is an async operation; poll the returned Operation for status.
 */
export const importLocalPackages = <ThrowOnError extends boolean = false>(
  options: Options<ImportLocalPackagesData, ThrowOnError>,
) =>
  (options.client ?? client).post<
    ImportLocalPackagesResponses,
    ImportLocalPackagesErrors,
    ThrowOnError
  >({
    url: "/v1beta1/{name}:importPackages",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * PublishPublication triggers the actual publication process.
 * This is a custom method that returns a background Task.
 */
export const publishPublication = <ThrowOnError extends boolean = false>(
  options: Options<PublishPublicationData, ThrowOnError>,
) =>
  (options.client ?? client).post<
    PublishPublicationResponses,
    PublishPublicationErrors,
    ThrowOnError
  >({
    url: "/v1beta1/{name}:publish",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * SyncMirror synchronizes a Mirror.
 */
export const syncMirror = <ThrowOnError extends boolean = false>(
  options: Options<SyncMirrorData, ThrowOnError>,
) =>
  (options.client ?? client).post<
    SyncMirrorResponses,
    SyncMirrorErrors,
    ThrowOnError
  >({
    url: "/v1beta1/{name}:sync",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * ListMirrorPackages retrieves a list of packages available in the Mirror.
 */
export const listMirrorPackages = <ThrowOnError extends boolean = false>(
  options: Options<ListMirrorPackagesData, ThrowOnError>,
) =>
  (options.client ?? client).get<
    ListMirrorPackagesResponses,
    ListMirrorPackagesErrors,
    ThrowOnError
  >({ url: "/v1beta1/{parent_1}/packages", ...options });

/**
 * ListLocalPackages retrieves a list of packages available in the Local.
 */
export const listLocalPackages = <ThrowOnError extends boolean = false>(
  options: Options<ListLocalPackagesData, ThrowOnError>,
) =>
  (options.client ?? client).get<
    ListLocalPackagesResponses,
    ListLocalPackagesErrors,
    ThrowOnError
  >({ url: "/v1beta1/{parent}/packages", ...options });

/**
 * UpdatePublication updates the details of a Publication.
 */
export const updatePublication = <ThrowOnError extends boolean = false>(
  options: Options<UpdatePublicationData, ThrowOnError>,
) =>
  (options.client ?? client).patch<
    UpdatePublicationResponses,
    UpdatePublicationErrors,
    ThrowOnError
  >({
    responseTransformer: updatePublicationResponseTransformer,
    url: "/v1beta1/{publication.name}",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * UpdatePublicationTarget replaces (full replacement) an existing PublicationTarget.
 */
export const updatePublicationTarget = <ThrowOnError extends boolean = false>(
  options: Options<UpdatePublicationTargetData, ThrowOnError>,
) =>
  (options.client ?? client).patch<
    UpdatePublicationTargetResponses,
    UpdatePublicationTargetErrors,
    ThrowOnError
  >({
    url: "/v1beta1/{publicationTarget.name}",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });
