// @canonical/landscape-openapi@0.0.60/generated/transformers.gen.ts
// This file is auto-generated by @hey-api/openapi-ts

import type {
  BatchGetMirrorsResponse2,
  CreateMirrorResponse,
  CreatePublicationResponse,
  GetMirrorResponse,
  GetPublicationResponse,
  ListMirrorsResponse2,
  ListPublicationsResponse2,
  UpdateMirrorResponse,
  UpdatePublicationResponse,
} from "./types.gen";

const mirrorSchemaResponseTransformer = (data: any) => {
  if (data.lastDownloadDate) {
    data.lastDownloadDate = new Date(data.lastDownloadDate);
  }
  return data;
};

const listMirrorsResponseSchemaResponseTransformer = (data: any) => {
  if (data.mirrors) {
    data.mirrors = data.mirrors.map((item: any) =>
      mirrorSchemaResponseTransformer(item),
    );
  }
  return data;
};

export const listMirrorsResponseTransformer = async (
  data: any,
): Promise<ListMirrorsResponse2> => {
  data = listMirrorsResponseSchemaResponseTransformer(data);
  return data;
};

export const createMirrorResponseTransformer = async (
  data: any,
): Promise<CreateMirrorResponse> => {
  data = mirrorSchemaResponseTransformer(data);
  return data;
};

const batchGetMirrorsResponseSchemaResponseTransformer = (data: any) => {
  if (data.mirrors) {
    data.mirrors = data.mirrors.map((item: any) =>
      mirrorSchemaResponseTransformer(item),
    );
  }
  return data;
};

export const batchGetMirrorsResponseTransformer = async (
  data: any,
): Promise<BatchGetMirrorsResponse2> => {
  data = batchGetMirrorsResponseSchemaResponseTransformer(data);
  return data;
};

const publicationSchemaResponseTransformer = (data: any) => {
  if (data.publishTime) {
    data.publishTime = new Date(data.publishTime);
  }
  return data;
};

const listPublicationsResponseSchemaResponseTransformer = (data: any) => {
  if (data.publications) {
    data.publications = data.publications.map((item: any) =>
      publicationSchemaResponseTransformer(item),
    );
  }
  return data;
};

export const listPublicationsResponseTransformer = async (
  data: any,
): Promise<ListPublicationsResponse2> => {
  data = listPublicationsResponseSchemaResponseTransformer(data);
  return data;
};

export const createPublicationResponseTransformer = async (
  data: any,
): Promise<CreatePublicationResponse> => {
  data = publicationSchemaResponseTransformer(data);
  return data;
};

export const updateMirrorResponseTransformer = async (
  data: any,
): Promise<UpdateMirrorResponse> => {
  data = mirrorSchemaResponseTransformer(data);
  return data;
};

export const getMirrorResponseTransformer = async (
  data: any,
): Promise<GetMirrorResponse> => {
  data = mirrorSchemaResponseTransformer(data);
  return data;
};

export const getPublicationResponseTransformer = async (
  data: any,
): Promise<GetPublicationResponse> => {
  data = publicationSchemaResponseTransformer(data);
  return data;
};

export const updatePublicationResponseTransformer = async (
  data: any,
): Promise<UpdatePublicationResponse> => {
  data = publicationSchemaResponseTransformer(data);
  return data;
};
