// @canonical/react-components@3.11.1/dist/esm/components/ApplicationLayout/ApplicationLayout.js
var _excluded = ["aside", "children", "dark", "logo", "mainClassName", "menuCollapsed", "menuPinned", "navigationBarClassName", "navigationClassName", "navItems", "navLinkComponent", "onCollapseMenu", "onPinMenu", "sideNavigation", "status", "statusClassName", "mainId"];
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React from "react";
import classNames from "classnames";
import { useState } from "react";
import Panel from "../Panel";
import SideNavigation from "../SideNavigation";
import AppMain from "./AppMain";
import AppNavigation from "./AppNavigation";
import AppNavigationBar from "./AppNavigationBar";
import AppStatus from "./AppStatus";
import Application from "./Application";
import Button from "../Button";
import Icon from "../Icon";
import SkipLink from "../SkipLink";
/**
 * This is a [React](https://reactjs.org/) component for the Vanilla
 * [Application Layout](https://vanillaframework.io/docs/layouts/application).
 *
 * This component combines the various sub-components that make up the
 * Application Layout to make it easier to consume. For some applications this
 * component maybe not be flexible enough, in which case you may choose to use
 * the sub-components directly. Be aware that the application layout requires a
 * specific structure and states to function correctly so there will be a
 * trade-off when using the sub-components directly.
 */
var ApplicationLayout = _ref => {
  var {
      aside,
      children,
      dark = true,
      logo,
      mainClassName,
      menuCollapsed,
      menuPinned,
      navigationBarClassName,
      navigationClassName,
      navItems,
      navLinkComponent,
      onCollapseMenu,
      onPinMenu,
      sideNavigation,
      status,
      statusClassName,
      mainId = "main-content"
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  var [internalMenuPinned, setInternalMenuPinned] = useState(false);
  var [internalMenuCollapsed, setInternalMenuCollapsed] = useState(true);
  var menuIsPinned = menuPinned !== null && menuPinned !== void 0 ? menuPinned : internalMenuPinned;
  var setMenuPinned = onPinMenu !== null && onPinMenu !== void 0 ? onPinMenu : setInternalMenuPinned;
  var menuIsCollapsed = menuCollapsed !== null && menuCollapsed !== void 0 ? menuCollapsed : internalMenuCollapsed;
  var setMenuCollapsed = onCollapseMenu !== null && onCollapseMenu !== void 0 ? onCollapseMenu : setInternalMenuCollapsed;
  return /*#__PURE__*/React.createElement(Application, props, /*#__PURE__*/React.createElement(SkipLink, {
    mainId: mainId
  }), (navItems || sideNavigation) && /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(AppNavigationBar, {
    className: navigationBarClassName
  }, /*#__PURE__*/React.createElement(Panel, {
    dark: dark,
    logo: logo,
    toggle: {
      label: "Menu",
      onClick: () => setMenuCollapsed(!menuIsCollapsed)
    }
  })), /*#__PURE__*/React.createElement(AppNavigation, {
    className: navigationClassName,
    collapsed: menuIsCollapsed,
    pinned: menuIsPinned
  }, /*#__PURE__*/React.createElement(Panel, {
    dark: dark,
    controls: /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Button, {
      hasIcon: true,
      appearance: "base",
      className: classNames("u-no-margin u-hide--medium", {
        "is-dark": dark
      }),
      onClick: evt => {
        setMenuCollapsed(true);
        // The menu stays open while its content has focus, so the
        // close button must blur to actually close the menu.
        evt.currentTarget.blur();
      }
    }, /*#__PURE__*/React.createElement(Icon, {
      name: "close",
      className: classNames({
        "is-light": dark
      })
    }, "Close menu")), /*#__PURE__*/React.createElement(Button, {
      hasIcon: true,
      appearance: "base",
      className: classNames("u-no-margin u-hide--small", {
        "is-dark": dark
      }),
      onClick: () => {
        setMenuPinned(!menuIsPinned);
      }
    }, /*#__PURE__*/React.createElement(Icon, {
      name: menuIsPinned ? "close" : "pin",
      className: classNames({
        "is-light": dark
      })
    }, menuIsPinned ? "Unpin menu" : "Pin menu"))),
    controlsClassName: "u-hide--large",
    stickyHeader: true,
    logo: logo
  }, navItems ? /*#__PURE__*/React.createElement(SideNavigation, {
    dark: dark,
    items: navItems,
    linkComponent: navLinkComponent
  }) : sideNavigation))), /*#__PURE__*/React.createElement(AppMain, {
    id: mainId,
    className: mainClassName
  }, children), aside, status && /*#__PURE__*/React.createElement(AppStatus, {
    className: statusClassName
  }, status));
};
export default ApplicationLayout;
