// @canonical/react-components@4.5.0/dist/esm/components/PrefixedIpInput/utils.js
/**
 * Checks if a given IP address is a valid IPv4 address.
 * @param ip The IP address to check
 * @returns True if the IP is a valid IPv4 address, false otherwise
 */
export var isIPv4 = ip => {
  var ipv4Regex = /^(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}$/;
  return ipv4Regex.test(ip);
};

/**
 * Takes a subnet CIDR notation (IPv4) and returns the first and last IP of the subnet.
 * The network and host addresses are excluded.
 *
 * @param cidr The CIDR notation of the subnet
 * @returns The first and last valid IP addresses as two strings in a list.
 */
export var getIpRangeFromCidr = cidr => {
  // https://gist.github.com/binarymax/6114792

  // Get start IP and number of valid addresses
  var [unmaskedStartIp, mask] = cidr.split("/");
  var maskBits = parseInt(mask, 10);
  var subnetMask = maskBits === 0 ? 0 : 0xffffffff << 32 - maskBits >>> 0;
  var startIp = convertUint32ToIp(convertIpToUint32(unmaskedStartIp) & subnetMask);
  var numberOfAddresses = (1 << 32 - maskBits) - 1;

  // IPv4 can be represented by an unsigned 32-bit integer, so we can use a Uint32Array to store the IP
  var buffer = new ArrayBuffer(4); //4 octets
  var int32 = new Uint32Array(buffer);

  // Convert starting IP to Uint32 and add the number of addresses to get the end IP.
  // Subtract 1 from the number of addresses to exclude the broadcast address.
  int32[0] = convertIpToUint32(startIp) + numberOfAddresses - 1;

  // Convert the buffer to a Uint8Array to get the octets, then convert it to an array
  var arrayApplyBuffer = Array.from(new Uint8Array(buffer));

  // Reverse the octets and join them with "." to get the end IP
  var endIp = arrayApplyBuffer.reverse().join(".");
  var firstValidIp = getFirstValidIp(startIp);
  return [firstValidIp, endIp];
};
export var getFirstValidIp = ip => {
  var buffer = new ArrayBuffer(4); //4 octets
  var int32 = new Uint32Array(buffer);

  // add 1 because the first IP is the network address
  int32[0] = convertIpToUint32(ip) + 1;
  var arrayApplyBuffer = Array.from(new Uint8Array(buffer));
  return arrayApplyBuffer.reverse().join(".");
};
export var convertIpToUint32 = ip => {
  var octets = ip.split(".").map(a => parseInt(a));
  var buffer = new ArrayBuffer(4);
  var int32 = new Uint32Array(buffer);
  int32[0] = (octets[0] << 24) + (octets[1] << 16) + (octets[2] << 8) + octets[3];
  return int32[0];
};
export var convertUint32ToIp = ipAsUint32 => {
  var buffer = new ArrayBuffer(4); //4 octets
  var int32 = new Uint32Array(buffer);
  int32[0] = ipAsUint32;
  var octets = Array.from(new Uint8Array(buffer));
  return octets.reverse().join(".");
};

/**
 * Checks if an IPv4 address is valid for the given subnet.
 *
 * @param ip The IPv4 address to check, as a string
 * @param cidr The subnet's CIDR notation e.g. 192.168.0.0/24
 * @returns True if the IP is in the subnet, false otherwise
 */
export var isIpInSubnet = (ip, cidr) => {
  var [startIP, endIP] = getIpRangeFromCidr(cidr);
  var ipUint32 = convertIpToUint32(ip);
  var startIPUint32 = convertIpToUint32(startIP);
  var endIPUint32 = convertIpToUint32(endIP);
  return ipUint32 >= startIPUint32 && ipUint32 <= endIPUint32;
};

/**
 * Separates the immutable and editable octets of an IPv4 subnet range.
 *
 * @param startIp The start IP of the subnet
 * @param endIp The end IP of the subnet
 * @returns The immutable and editable octects as two strings in a list
 */
export var getImmutableAndEditableOctets = (startIp, endIp) => {
  var startIpOctetList = startIp.split(".");
  var endIpOctetList = endIp.split(".");
  var immutable = [];
  var editable = [];
  startIpOctetList.forEach((octet, index) => {
    if (octet === endIpOctetList[index]) {
      immutable.push(octet);
    } else {
      editable.push("[".concat(octet, "-").concat(endIpOctetList[index], "]"));
    }
  });
  return [immutable.join("."), editable.join(".")];
};

/**
 * Separates the immutable and editable parts of an IPv6 subnet range.
 * For simplcity, if the prefix is not on a group boundary, the entire last group is considered editable.
 *
 * @param cidr The CIDR notation of the subnet
 * @returns The immutable and editable parts as two strings in a list
 */
export var getImmutableAndEditableIPv6 = cidr => {
  var [address, prefix] = cidr.split("/");
  var prefixLength = parseInt(prefix, 10);
  var [left = "", right = ""] = address.split("::");
  var leftGroups = left ? left.split(":").filter(Boolean) : [];
  var rightGroups = right ? right.split(":").filter(Boolean) : [];
  var missingGroups = Math.max(0, 8 - (leftGroups.length + rightGroups.length));
  var expandedGroups = [...leftGroups, ...Array(missingGroups).fill("0"), ...Array(rightGroups.length).fill("0")];
  var immutableGroupCount = Math.floor(prefixLength / 16);
  var immutableIPV6 = immutableGroupCount > 0 ? "".concat(expandedGroups.slice(0, immutableGroupCount).join(":")) : "";
  if (immutableGroupCount < 8) {
    immutableIPV6 += immutableIPV6 ? ":" : "";
  }
  var editableIPV6 = "".concat(expandedGroups.slice(immutableGroupCount).join(":"));
  return [immutableIPV6, editableIPV6];
};

/**
 * Get the immutable and editable parts of an IPv4 or IPv6 subnet.
 *
 * @param cidr The CIDR notation of the subnet
 * @returns The immutable and editable as two strings in a list
 */
export var getImmutableAndEditable = cidr => {
  var isIPV4 = isIPv4(cidr.split("/")[0]);
  if (isIPV4) {
    var [startIp, endIp] = getIpRangeFromCidr(cidr);
    return getImmutableAndEditableOctets(startIp, endIp);
  }
  return getImmutableAndEditableIPv6(cidr);
};
