// @canonical/react-components@3.11.1/dist/components/PrefixedInput/PrefixedInput.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithLabel = exports.WithHelpText = exports.WithError = exports.Required = exports.Disabled = exports.Default = void 0;
var _PrefixedInput = _interopRequireDefault(require("./PrefixedInput"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const meta = {
  component: _PrefixedInput.default,
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  args: {
    immutableText: "https://",
    placeholder: "example.com"
  }
};
const WithLabel = exports.WithLabel = {
  args: {
    immutableText: "https://",
    label: "Website URL",
    placeholder: "example.com"
  }
};
const Disabled = exports.Disabled = {
  args: {
    immutableText: "@",
    label: "Username",
    placeholder: "username",
    disabled: true
  }
};
const WithError = exports.WithError = {
  args: {
    immutableText: "https://",
    label: "Website URL",
    placeholder: "example.com",
    error: "Invalid URL format"
  }
};
const WithHelpText = exports.WithHelpText = {
  args: {
    immutableText: "User ID:",
    label: "User Identifier",
    placeholder: " Enter user ID",
    help: "This will be used to identify your account"
  }
};
const Required = exports.Required = {
  args: {
    immutableText: "https://",
    label: "Website URL",
    placeholder: "example.com",
    required: true
  }
};
