// @canonical/react-components@3.8.2/dist/esm/components/Switch/Switch.js
var _excluded = ["label", "disabled", "help", "helpClassName"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React from "react";
import classNames from "classnames";
import { useId } from "react";
import "./Switch.scss";
/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Switch](https://vanillaframework.io/docs/patterns/switch) component.
 */
export var Switch = _ref => {
  var {
      label,
      disabled = false,
      help,
      helpClassName
    } = _ref,
    inputProps = _objectWithoutProperties(_ref, _excluded);
  var helpId = useId();
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("label", {
    className: "p-switch"
  }, /*#__PURE__*/React.createElement("input", _extends({
    type: "checkbox",
    className: "p-switch__input",
    role: "switch",
    disabled: disabled
  }, inputProps)), /*#__PURE__*/React.createElement("span", {
    className: "p-switch__slider"
  }), /*#__PURE__*/React.createElement("span", {
    className: "p-switch__label"
  }, label)), help && /*#__PURE__*/React.createElement("p", {
    className: classNames("p-form-help-text", "switch-help-text", helpClassName),
    id: helpId
  }, help));
};
export default Switch;
