// @canonical/react-components@3.11.1/dist/components/Slider/Slider.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WithInput = exports.Default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _Slider = _interopRequireDefault(require("./Slider"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
const Template = args => {
  const [selectedValue, setSelectedValue] = (0, _react.useState)(50);
  return /*#__PURE__*/_react.default.createElement(_Slider.default, _extends({}, args, {
    onChange: e => setSelectedValue(Number(e.target.value)),
    value: selectedValue
  }));
};
const meta = {
  component: _Slider.default,
  render: Template,
  tags: ["autodocs"],
  argTypes: {
    caution: {
      control: {
        type: "text"
      }
    },
    error: {
      control: {
        type: "text"
      }
    },
    help: {
      control: {
        type: "text"
      }
    },
    label: {
      control: {
        type: "text"
      }
    }
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  render: Template.bind({}),
  name: "Default",
  args: {
    disabled: false,
    label: "Volume",
    max: 100,
    min: 0,
    showInput: false
  }
};

/**
 * You can also optionally provide a numeric representation of the slider value.
 */
const WithInput = exports.WithInput = {
  render: Template.bind({}),
  name: "With input",
  args: {
    disabled: false,
    inputDisabled: false,
    label: "Volume",
    max: 100,
    min: 0,
    showInput: true
  }
};
