// @canonical/react-components@3.11.1/dist/esm/components/PasswordToggle/PasswordToggle.js
var _excluded = ["caution", "className", "error", "help", "id", "label", "readOnly", "required", "success", "type", "wrapperClassName"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import classNames from "classnames";
import React, { useId, useState } from "react";
import Button from "../Button";
import Field from "../Field";
import VanillaLabel from "../Label";
import "./PasswordToggle.scss";
export var Label = /*#__PURE__*/function (Label) {
  Label["Hide"] = "Hide";
  Label["Show"] = "Show";
  return Label;
}({});

/**
 * The props for the Password Toggle component.
 */

/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Password Toggle](https://docs.vanillaframework.io/base/forms/#password-toggle).
 *
 * It can be used when an input needs to obscure its value, whilst giving the user a way to reveal it if needed.
 */
var PasswordToggle = /*#__PURE__*/React.forwardRef((_ref, ref) => {
  var {
      caution,
      className,
      error,
      help,
      id,
      label,
      readOnly,
      required,
      success,
      type,
      wrapperClassName
    } = _ref,
    inputProps = _objectWithoutProperties(_ref, _excluded);
  var [isPassword, setIsPassword] = useState(true);
  var validationId = useId();
  var helpId = useId();
  var hasError = !!error;
  var defaultPasswordToggleId = useId();
  var passwordToggleId = id || defaultPasswordToggleId;
  var togglePassword = () => {
    if (isPassword) {
      setIsPassword(false);
    } else {
      setIsPassword(true);
    }
  };
  return /*#__PURE__*/React.createElement(Field, {
    caution: caution,
    className: wrapperClassName,
    error: error,
    help: help,
    helpId: helpId,
    required: required,
    success: success,
    validationId: validationId
  }, /*#__PURE__*/React.createElement("div", {
    className: "password-toggle"
  }, /*#__PURE__*/React.createElement("input", _extends({
    "aria-describedby": [help ? helpId : null, success ? validationId : null].filter(Boolean).join(" "),
    "aria-errormessage": hasError ? validationId : null,
    "aria-invalid": hasError,
    className: classNames("p-form-validation__input", className),
    id: passwordToggleId,
    readOnly: readOnly,
    ref: ref,
    type: isPassword ? "password" : "text"
  }, inputProps)), /*#__PURE__*/React.createElement("div", {
    className: "p-form-password-toggle"
  }, /*#__PURE__*/React.createElement(VanillaLabel, {
    forId: passwordToggleId,
    required: required
  }, label), /*#__PURE__*/React.createElement(Button, {
    appearance: "base",
    type: type ? type : "button",
    className: "u-no-margin--bottom",
    hasIcon: true,
    "aria-controls": id,
    "aria-live": "polite",
    onClick: () => togglePassword()
  }, /*#__PURE__*/React.createElement("span", {
    className: "p-form-password-toggle__label"
  }, isPassword ? Label.Show : Label.Hide), /*#__PURE__*/React.createElement("i", {
    className: isPassword ? "p-icon--show" : "p-icon--hide"
  })))));
});
export default PasswordToggle;
