// @canonical/react-components@3.11.1/dist/esm/components/Stepper/Step/Step.js
import _pt from "prop-types";
var _excluded = ["hasProgressLine", "index", "title", "label", "linkProps", "enabled", "selected", "iconName", "iconClassName", "handleClick"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import classNames from "classnames";
import React from "react";
import Icon from "../../Icon";
import Link from "../../Link";
import "./Step.scss";
var Step = _ref => {
  var {
      hasProgressLine,
      index,
      title,
      label,
      linkProps,
      enabled,
      selected = false,
      iconName,
      iconClassName,
      handleClick
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  var stepStatusClass = enabled ? "step-enabled" : "step-disabled";
  return /*#__PURE__*/React.createElement("div", _extends({
    className: classNames("step", {
      "progress-line": hasProgressLine,
      "step-selected": selected
    })
  }, props), iconName === "number" ? /*#__PURE__*/React.createElement("span", {
    className: classNames("step-number", {
      "step-number-disabled": !enabled
    })
  }, index) : /*#__PURE__*/React.createElement(Icon, {
    name: iconName,
    className: classNames("step-status-icon", iconClassName)
  }), /*#__PURE__*/React.createElement("div", {
    className: "step-content"
  }, /*#__PURE__*/React.createElement("span", {
    className: classNames(stepStatusClass),
    onClick: handleClick
  }, title), label && /*#__PURE__*/React.createElement("span", {
    className: classNames("step-optional-content", "u-no-margin--bottom", {
      "step-disabled": !enabled
    })
  }, label), linkProps && /*#__PURE__*/React.createElement(Link, _extends({
    className: "p-text--small u-no-margin--bottom step-optional-content"
  }, linkProps), linkProps.children)));
};
Step.propTypes = {
  hasProgressLine: _pt.bool.isRequired,
  index: _pt.number.isRequired,
  title: _pt.string.isRequired,
  label: _pt.string,
  enabled: _pt.bool.isRequired,
  selected: _pt.bool,
  iconName: _pt.string.isRequired,
  handleClick: _pt.func.isRequired
};
export default Step;
