// @canonical/react-components@3.8.2/dist/components/ApplicationLayout/AppAside/AppAside.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _Application = _interopRequireDefault(require("../Application"));
var _AppMain = _interopRequireDefault(require("../AppMain"));
var _Button = _interopRequireDefault(require("../../Button"));
var _Col = _interopRequireDefault(require("../../Col"));
var _Form = _interopRequireDefault(require("../../Form"));
var _Icon = _interopRequireDefault(require("../../Icon"));
var _Input = _interopRequireDefault(require("../../Input"));
var _Panel = _interopRequireDefault(require("../../Panel"));
var _Row = _interopRequireDefault(require("../../Row"));
var _AppAside = _interopRequireDefault(require("./AppAside"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); } /* eslint-disable react-hooks/rules-of-hooks */
const meta = {
  component: _AppAside.default,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: false
    }
  }
};
var _default = exports.default = meta;
/**
 * In most common cases an `AppAside` should contain a `<Panel>` to display the
 * content as intended in the application layout.
 *
 * `AppAside` should be a direct child of an `<Application>` or passed to the
 * application layout `<ApplicationLayout aside={<AppAside .../>}>`.
 */
const Default = exports.Default = {
  render: args => {
    const [pinned, setPinned] = (0, _react.useState)(false);
    const [width, setWidth] = (0, _react.useState)(null);
    const [collapsed, setCollapsed] = (0, _react.useState)(false);
    return /*#__PURE__*/_react.default.createElement(_Application.default, null, /*#__PURE__*/_react.default.createElement(_AppMain.default, null, /*#__PURE__*/_react.default.createElement("p", null, "Scroll to the right to see the panel."), /*#__PURE__*/_react.default.createElement(_Button.default, {
      onClick: () => setCollapsed(false)
    }, "Open"), /*#__PURE__*/_react.default.createElement(_Button.default, {
      onClick: () => setWidth("narrow")
    }, "Narrow"), /*#__PURE__*/_react.default.createElement(_Button.default, {
      onClick: () => setWidth(null)
    }, "Default"), /*#__PURE__*/_react.default.createElement(_Button.default, {
      onClick: () => setWidth("wide")
    }, "Wide")), /*#__PURE__*/_react.default.createElement(_AppAside.default, _extends({}, args, {
      pinned: pinned,
      wide: width === "wide",
      narrow: width === "narrow",
      collapsed: collapsed
    }), /*#__PURE__*/_react.default.createElement(_Panel.default, {
      controls: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_Button.default, {
        onClick: () => setPinned(!pinned),
        dense: true,
        className: "u-no-margin--bottom"
      }, "Pin"), /*#__PURE__*/_react.default.createElement(_Button.default, {
        appearance: "base",
        className: "u-no-margin--bottom",
        hasIcon: true,
        onClick: () => setCollapsed(!collapsed)
      }, /*#__PURE__*/_react.default.createElement(_Icon.default, {
        name: "close"
      }, "Close"))),
      title: "App aside"
    }, /*#__PURE__*/_react.default.createElement(_Form.default, {
      stacked: true
    }, /*#__PURE__*/_react.default.createElement(_Input.default, {
      label: "Full name",
      type: "text",
      name: "fullName",
      autoComplete: "name",
      stacked: true
    }), /*#__PURE__*/_react.default.createElement(_Input.default, {
      label: "Username",
      type: "text",
      name: "username-stacked",
      autoComplete: "username",
      "aria-describedby": "exampleHelpTextMessage",
      stacked: true,
      help: "30 characters or fewer."
    }), /*#__PURE__*/_react.default.createElement(_Input.default, {
      type: "text",
      label: "Email address",
      "aria-invalid": "true",
      name: "username-stackederror",
      autoComplete: "email",
      required: true,
      error: "This field is required.",
      stacked: true
    }), /*#__PURE__*/_react.default.createElement(_Input.default, {
      label: "Address line 1",
      type: "text",
      name: "address-optional-stacked",
      autoComplete: "address-line1",
      stacked: true
    }), /*#__PURE__*/_react.default.createElement(_Input.default, {
      label: "Address line 2",
      type: "text",
      name: "address-optional-stacked",
      autoComplete: "address-line3",
      stacked: true
    }), /*#__PURE__*/_react.default.createElement(_Row.default, null, /*#__PURE__*/_react.default.createElement(_Col.default, {
      size: 12
    }, /*#__PURE__*/_react.default.createElement(_Button.default, {
      appearance: "positive",
      className: "u-float-right",
      name: "add-details"
    }, "Add details")))))));
  }
};
