// @canonical/react-components@4.5.4/dist/components/OutputField/OutputField.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Required = exports.HelpText = exports.Default = void 0;
var _OutputField = _interopRequireDefault(require("./OutputField"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const meta = {
  component: _OutputField.default,
  tags: ["autodocs"],
  argTypes: {
    label: {
      control: {
        type: "text"
      }
    },
    id: {
      control: {
        type: "text"
      }
    },
    value: {
      control: {
        type: "text"
      }
    },
    help: {
      control: {
        type: "text"
      }
    },
    required: {
      control: {
        type: "boolean"
      }
    }
  },
  args: {
    label: "Label",
    id: "output-field",
    value: "Output value",
    help: "",
    required: false
  }
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default"
};
const HelpText = exports.HelpText = {
  name: "Help Text",
  args: {
    help: "This is an output field with text"
  }
};
const Required = exports.Required = {
  name: "Required",
  args: {
    required: true
  }
};
