// @canonical/react-components@4.5.4/dist/components/OutputField/OutputField.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.OutputField = void 0;
var _propTypes = _interopRequireDefault(require("prop-types"));
var _react = _interopRequireDefault(require("react"));
var _Field = _interopRequireDefault(require("../Field"));
require("./OutputField.scss");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
/**
 *
 * Use output fields to display read-only information with a label and optional help text. Output fields are ideal for displaying calculated values or results of user input in a consistent format.
 */
const OutputField = _ref => {
  let {
    id,
    label,
    value,
    help,
    required
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(_Field.default, {
    forId: id,
    label: label,
    help: help,
    labelClassName: "u-no-margin--bottom",
    className: "output-field",
    required: required
  }, /*#__PURE__*/_react.default.createElement("output", {
    id: id,
    className: "mono-font u-sv2"
  }, /*#__PURE__*/_react.default.createElement("b", null, value)));
};
exports.OutputField = OutputField;
OutputField.propTypes = {
  id: _propTypes.default.string.isRequired,
  label: _propTypes.default.string.isRequired,
  value: _propTypes.default.string.isRequired,
  help: _propTypes.default.node,
  required: _propTypes.default.bool
};
var _default = exports.default = OutputField;
