// @canonical/landscape-openapi@0.0.59/generated/types.gen.ts
// This file is auto-generated by @hey-api/openapi-ts

export type ClientOptions = {
  baseUrl: `${string}://openapi.json` | (string & {});
};

/**
 * `Any` contains an arbitrary serialized protocol buffer message along with a
 * URL that describes the type of the serialized message.
 *
 * Protobuf library provides support to pack/unpack Any values in the form
 * of utility functions or additional generated methods of the Any type.
 *
 * Example 1: Pack and unpack a message in C++.
 *
 * Foo foo = ...;
 * Any any;
 * any.PackFrom(foo);
 * ...
 * if (any.UnpackTo(&foo)) {
 * ...
 * }
 *
 * Example 2: Pack and unpack a message in Java.
 *
 * Foo foo = ...;
 * Any any = Any.pack(foo);
 * ...
 * if (any.is(Foo.class)) {
 * foo = any.unpack(Foo.class);
 * }
 * // or ...
 * if (any.isSameTypeAs(Foo.getDefaultInstance())) {
 * foo = any.unpack(Foo.getDefaultInstance());
 * }
 *
 * Example 3: Pack and unpack a message in Python.
 *
 * foo = Foo(...)
 * any = Any()
 * any.Pack(foo)
 * ...
 * if any.Is(Foo.DESCRIPTOR):
 * any.Unpack(foo)
 * ...
 *
 * Example 4: Pack and unpack a message in Go
 *
 * foo := &pb.Foo{...}
 * any, err := anypb.New(foo)
 * if err != nil {
 * ...
 * }
 * ...
 * foo := &pb.Foo{}
 * if err := any.UnmarshalTo(foo); err != nil {
 * ...
 * }
 *
 * The pack methods provided by protobuf library will by default use
 * 'type.googleapis.com/full.type.name' as the type URL and the unpack
 * methods only use the fully qualified type name after the last '/'
 * in the type URL, for example "foo.bar.com/x/y.z" will yield type
 * name "y.z".
 *
 * JSON
 * ====
 * The JSON representation of an `Any` value uses the regular
 * representation of the deserialized, embedded message, with an
 * additional field `@type` which contains the type URL. Example:
 *
 * package google.profile;
 * message Person {
 * string first_name = 1;
 * string last_name = 2;
 * }
 *
 * {
 * "@type": "type.googleapis.com/google.profile.Person",
 * "firstName": <string>,
 * "lastName": <string>
 * }
 *
 * If the embedded message type is well-known and has a custom JSON
 * representation, that representation will be embedded adding a field
 * `value` which holds the custom JSON in addition to the `@type`
 * field. Example (for message [google.protobuf.Duration][]):
 *
 * {
 * "@type": "type.googleapis.com/google.protobuf.Duration",
 * "value": "1.212s"
 * }
 */
export type Any = {
  /**
   * A URL/resource name that uniquely identifies the type of the serialized
   * protocol buffer message. This string must contain at least
   * one "/" character. The last segment of the URL's path must represent
   * the fully qualified name of the type (as in
   * `path/google.protobuf.Duration`). The name should be in a canonical form
   * (e.g., leading "." is not accepted).
   *
   * In practice, teams usually precompile into the binary all types that they
   * expect it to use in the context of Any. However, for URLs which use the
   * scheme `http`, `https`, or no scheme, one can optionally set up a type
   * server that maps type URLs to message definitions as follows:
   *
   * * If no scheme is provided, `https` is assumed.
   * * An HTTP GET on the URL must yield a [google.protobuf.Type][]
   * value in binary format, or produce an error.
   * * Applications are allowed to cache lookup results based on the
   * URL, or have them precompiled into a binary to avoid any
   * lookup. Therefore, binary compatibility needs to be preserved
   * on changes to types. (Use versioned type names to manage
   * breaking changes.)
   *
   * Note: this functionality is not currently available in the official
   * protobuf release, and it is not used for type URLs beginning with
   * type.googleapis.com. As of May 2023, there are no widely used type server
   * implementations and no plans to implement one.
   *
   * Schemes other than `http`, `https` (or the empty scheme) might be
   * used with implementation specific semantics.
   */
  "@type"?: string;
  [key: string]: unknown;
};

/**
 * BatchGetLocalsRequest is the request message for BatchGetLocals
 */
export type BatchGetLocalsRequest = {
  /**
   * The names of the locals to retrieve.
   * A maximum of 100 locals can be retrieved in a batch.
   * Format: locals/{local}
   */
  names: Array<string>;
};

/**
 * BatchGetLocalsResponse is the response message for BatchGetLocals
 */
export type BatchGetLocalsResponse = {
  /**
   * Locals requested.
   */
  locals?: Array<Local>;
};

/**
 * BatchGetMirrorsRequest is the request message for BatchGetMirrors
 */
export type BatchGetMirrorsRequest = {
  /**
   * The names of the mirrors to retrieve.
   * A maximum of 100 mirrors can be retrieved in a batch.
   * Format: mirrors/{mirror}
   */
  names: Array<string>;
};

/**
 * BatchGetMirrorsResponse is the response message for BatchGetMirrors
 */
export type BatchGetMirrorsResponse = {
  /**
   * Mirrors requested.
   */
  mirrors?: Array<Mirror>;
};

/**
 * BatchGetPublicationTargetsRequest is the request message for BatchGetPublicationTargets
 */
export type BatchGetPublicationTargetsRequest = {
  /**
   * The names of the publication targets to retrieve.
   * A maximum of 100 publication targets can be retrieved in a batch.
   * Format: publicationTargets/{publication_target}
   */
  names: Array<string>;
};

/**
 * BatchGetPublicationTargetsResponse is the response message for BatchGetPublicationTargets
 */
export type BatchGetPublicationTargetsResponse = {
  /**
   * Publication targets requested.
   */
  publicationTargets?: Array<PublicationTarget>;
};

/**
 * FilesystemTarget represents a local filesystem storage destination.
 */
export type FilesystemTarget = {
  /**
   * The absolute path to the directory where files should be published.
   */
  path: string;
  linkMethod?: FilesystemTargetLinkMethod;
};

/**
 * Possible link methods for published files. Defaults to HARDLINK.
 *
 * - HARDLINK: Create hardlinks for published files.
 * - SYMLINK: Create symlinks for published files.
 * - COPY: Copy published files.
 */
export const FilesystemTargetLinkMethod = {
  HARDLINK: "HARDLINK",
  SYMLINK: "SYMLINK",
  COPY: "COPY",
} as const;

/**
 * Possible link methods for published files. Defaults to HARDLINK.
 *
 * - HARDLINK: Create hardlinks for published files.
 * - SYMLINK: Create symlinks for published files.
 * - COPY: Copy published files.
 */
export type FilesystemTargetLinkMethod =
  (typeof FilesystemTargetLinkMethod)[keyof typeof FilesystemTargetLinkMethod];

/**
 * GpgKey represents a gpg key.
 */
export type GpgKey = {
  /**
   * The GPG key in ASCII armor format.
   */
  armor: string;
  /**
   * Indicates if the key holds a private component.
   */
  readonly isPrivate?: boolean;
  /**
   * The fingerprint of the key.
   */
  readonly fingerprint?: string;
};

/**
 * ListLocalPackagesResponse is the response message for ListLocalPackages.
 */
export type ListLocalPackagesResponse = {
  /**
   * The list of package strings.
   * Example: ["snap-http_1.4.0_source", ...]
   */
  localPackages?: Array<string>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * ListLocalsResponse is the response message for ListLocals.
 */
export type ListLocalsResponse = {
  /**
   * The list of local repositories.
   */
  locals?: Array<Local>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * ListMirrorPackagesResponse is the response message for ListMirrorPackages.
 */
export type ListMirrorPackagesResponse = {
  /**
   * The list of package strings.
   * Example: ["snap-http_1.4.0_source", ...]
   */
  mirrorPackages?: Array<string>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * ListMirrorsResponse is the response message for ListMirrors.
 */
export type ListMirrorsResponse = {
  /**
   * The list of mirrors.
   */
  mirrors?: Array<Mirror>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * ListPublicationTargetsResponse is the response message for ListPublicationTargets.
 */
export type ListPublicationTargetsResponse = {
  /**
   * The publication targets.
   */
  publicationTargets?: Array<PublicationTarget>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * ListPublicationsResponse is the response message for ListPublications.
 */
export type ListPublicationsResponse = {
  /**
   * A list of publications.
   */
  publications?: Array<Publication>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * Local represents a local repository configuration.
 */
export type Local = {
  /**
   * The resource name of the local repository.
   * Format: "locals/{local}"
   */
  name?: string;
  /**
   * The local repository ID is exposed separately for convenience.
   */
  readonly localId?: string;
  /**
   * The human-readable name used to reference this local repository.
   */
  displayName: string;
  /**
   * A user-facing description of the local repository.
   */
  comment?: string;
  /**
   * The default distribution to publish.
   */
  defaultDistribution?: string;
  /**
   * The default component to publish.
   */
  defaultComponent: string;
  /**
   * The last associated LRO with the local
   */
  readonly lastOperation?: string;
};

/**
 * ImportLocalPackagesRequest is the request message for ImportLocalPackages.
 */
export type LocalServiceImportLocalPackagesBody = {
  /**
   * The URL of the package list or index to import from.
   */
  url: string;
  /**
   * If set to true, the request is validated without actually performing the import.
   * The set of packages that would be imported is returned as the result of the task.
   * See: https://google.aip.dev/163
   */
  validateOnly?: boolean;
};

/**
 * Mirror represents a repository mirror configuration and status.
 */
export type Mirror = {
  /**
   * The resource name of the mirror.
   * Format: "mirrors/{mirror}"
   */
  name?: string;
  /**
   * The mirror ID is exposed separately for convenience.
   */
  readonly mirrorId?: string;
  /**
   * The status of the mirror.
   */
  readonly status?: string;
  /**
   * The timestamp of the last successful download.
   */
  readonly lastDownloadDate?: Date;
  /**
   * The human-readable name used to reference this mirror.
   */
  displayName: string;
  /**
   * Root of the debian archive (URL).
   */
  archiveRoot: string;
  /**
   * The name of the distribution (subdirectory of `dists`).
   */
  distribution?: string;
  /**
   * List of architectures to fetch.
   */
  architectures?: Array<string>;
  /**
   * List of components to fetch.
   */
  components: Array<string>;
  /**
   * Indicates if installer files should be downloaded.
   */
  downloadInstaller?: boolean;
  /**
   * Indicates if sources files should be downloaded.
   */
  downloadSources?: boolean;
  /**
   * Indicates if `.udebs` files should be downloaded.
   */
  downloadUdebs?: boolean;
  /**
   * Package query applied to the mirror.
   * Follows the [Reprepro/Aptly language](https://www.aptly.info/doc/feature/query/).
   */
  filter?: string;
  /**
   * Indicates if dependencies of matching packages should be included.
   */
  filterWithDeps?: boolean;
  /**
   * Indicates if the architecture list verification should be skipped.
   */
  skipArchitectureCheck?: boolean;
  /**
   * Indicates if the component list verification should be skipped.
   */
  skipComponentCheck?: boolean;
  /**
   * Indicates if the mirror is a signature-preserving mirror.
   */
  preserveSignatures?: boolean;
  gpgKey?: GpgKey;
  /**
   * The last associated LRO with the mirror
   */
  readonly lastOperation?: string;
};

/**
 * SyncMirrorRequest is the request message for SyncMirror.
 */
export type MirrorServiceSyncMirrorBody = {
  /**
   * Optional keyring files to use for validation.
   */
  keyrings?: Array<string>;
  /**
   * Ignore checksum mismatches.
   */
  ignoreChecksums?: boolean;
  /**
   * Ignore signature verification failures.
   */
  ignoreSignatures?: boolean;
  /**
   * Force a full update.
   */
  forceUpdate?: boolean;
  /**
   * Skip downloading packages that already exist.
   */
  skipExistingPackages?: boolean;
};

/**
 * This resource represents a long-running operation that is the result of a
 * network API call.
 */
export type Operation = {
  /**
   * The server-assigned name, which is only unique within the same service that
   * originally returns it. If you use the default HTTP mapping, the
   * `name` should be a resource name ending with `operations/{unique_id}`.
   */
  name?: string;
  metadata?: Any;
  /**
   * If the value is `false`, it means the operation is still in progress.
   * If `true`, the operation is completed, and either `error` or `response` is
   * available.
   */
  done?: boolean;
  error?: Status;
  response?: Any;
};

/**
 * Publication represents the configuration for publishing Debian artifacts
 * from a Source (Mirror or Local) to a Target.
 */
export type Publication = {
  /**
   * The resource name of the publication.
   * Format: "publications/{uuid}"
   */
  name?: string;
  /**
   * The human-readable name used to reference this publication.
   */
  displayName: string;
  /**
   * The publication ID (UUID) is exposed separately for convenience.
   */
  readonly publicationId?: string;
  /**
   * The resource name of the publication target.
   * Format: "publicationTargets/{publication_target}"
   */
  publicationTarget: string;
  /**
   * The resource name of the source mirror or local.
   * Format: "mirrors/{mirror}" or "locals/{local}"
   */
  source: string;
  /**
   * The name of the distribution.
   * Inferred from the source if omitted.
   */
  distribution?: string;
  /**
   * Value for the `Label:` field in the Release file.
   */
  label?: string;
  /**
   * Value for the `Origin:` field in the Release file.
   * Inferred from the source if omitted.
   */
  origin?: string;
  /**
   * List of architectures to publish.
   * Defaults to all source architectures if omitted.
   */
  architectures?: Array<string>;
  /**
   * Provides index file by hash if unique.
   */
  acquireByHash?: boolean;
  /**
   * Sets `ButAutomaticUpgrade: yes|no` in the Release file if provided.
   */
  butAutomaticUpgrades?: boolean;
  /**
   * Sets `NotAutomatic: yes|no` in the Release file if provided.
   */
  notAutomatic?: boolean;
  /**
   * Indicates if multiple distributions are supported.
   */
  multiDist?: boolean;
  /**
   * Indicates if bz2 compression should be skipped for index files.
   */
  skipBz2?: boolean;
  /**
   * Indicates if content indexes should not be generated.
   */
  skipContents?: boolean;
  gpgKey?: GpgKey;
  /**
   * The timestamp of the last successful publication.
   */
  readonly publishTime?: Date;
  /**
   * The last associated LRO with the publication
   */
  readonly lastOperation?: string;
};

/**
 * PublishPublicationRequest is the request message for PublishPublication.
 */
export type PublicationServicePublishPublicationBody = {
  /**
   * Force overwrite of existing artifacts.
   */
  forceOverwrite?: boolean;
  /**
   * Force cleanup of old artifacts.
   */
  forceCleanup?: boolean;
};

/**
 * PublicationTarget represents a storage destination (currently supports S3 or Swift).
 */
export type PublicationTarget = {
  /**
   * The resource name of the publication target.
   * Format: "publicationTargets/{publication_target}"
   */
  name?: string;
  /**
   * The user-defined display name.
   */
  displayName: string;
  /**
   * The publication_id (uuid) is part of the `name`, and is exposed separately for convenience.
   */
  readonly publicationTargetId?: string;
  s3?: S3Target;
  swift?: SwiftTarget;
  filesystem?: FilesystemTarget;
};

/**
 * S3Target represents an S3 storage destination.
 */
export type S3Target = {
  /**
   * The AWS region for the S3 bucket.
   */
  region: string;
  /**
   * The name of the S3 bucket.
   */
  bucket: string;
  /**
   * Hostname used for S3 compatible storage.
   * If set, the region is ignored.
   */
  endpoint?: string;
  /**
   * AWS credentials (Access Key ID) to access the S3 bucket.
   */
  awsAccessKeyId: string;
  /**
   * AWS credentials (Secret Access Key) to access the S3 bucket.
   */
  awsSecretAccessKey: string;
  /**
   * Prefix used in the bucket.
   * Defaults to the bucket root if not specified.
   */
  prefix?: string;
  /**
   * ACL (Access Control List) used for published files.
   */
  acl?: string;
  /**
   * The AWS S3 storage class to use.
   */
  storageClass?: string;
  /**
   * Server-side encryption method. Defaults to none.
   * Currently, the only available encryption method is `AES256`.
   */
  encryptionMethod?: string;
  /**
   * Workaround for specific S3-compatible issues (e.g. "+" in filenames).
   */
  plusWorkaround?: boolean;
  /**
   * Disables the `MultiDel` S3 API.
   * Useful for S3-compatible cloud storage providers that do not support it.
   */
  disableMultiDel?: boolean;
  /**
   * Disables Signature V4 support.
   * Useful for S3-compatible cloud storage providers that do not support SigV4.
   */
  forceSigV2?: boolean;
  /**
   * Enables debug logging for S3 operations.
   */
  debug?: boolean;
};

/**
 * The `Status` type defines a logical error model that is suitable for
 * different programming environments, including REST APIs and RPC APIs. It is
 * used by [gRPC](https://github.com/grpc). Each `Status` message contains
 * three pieces of data: error code, error message, and error details.
 *
 * You can find out more about this error model and how to work with it in the
 * [API Design Guide](https://cloud.google.com/apis/design/errors).
 */
export type Status = {
  /**
   * The status code, which should be an enum value of
   * [google.rpc.Code][google.rpc.Code].
   */
  code?: number;
  /**
   * A developer-facing error message, which should be in English. Any
   * user-facing error message should be localized and sent in the
   * [google.rpc.Status.details][google.rpc.Status.details] field, or localized
   * by the client.
   */
  message?: string;
  /**
   * A list of messages that carry the error details.  There is a common set of
   * message types for APIs to use.
   */
  details?: Array<Any>;
};

/**
 * SwiftTarget represents a Swift (OpenStack) storage destination.
 */
export type SwiftTarget = {
  /**
   * The name of the container.
   */
  container: string;
  /**
   * OpenStack credentials (Username) to access Keystone.
   */
  username: string;
  /**
   * OpenStack credentials (Password) to access Keystone.
   */
  password: string;
  /**
   * Prefix used in the container.
   * Defaults to the container root if not specified.
   */
  prefix?: string;
  /**
   * The full URL of the Keystone server (including port and version).
   */
  authUrl: string;
  /**
   * The OpenStack tenant name.
   */
  tenant?: string;
  /**
   * The OpenStack tenant ID.
   */
  tenantId?: string;
  /**
   * The OpenStack domain name.
   */
  domain?: string;
  /**
   * The OpenStack domain ID.
   */
  domainId?: string;
  /**
   * The OpenStack domain name the tenant/project belongs to.
   */
  tenantDomain?: string;
  /**
   * The OpenStack domain ID the tenant/project belongs to.
   */
  tenantDomainId?: string;
};

/**
 * BatchGetLocalsResponse is the response message for BatchGetLocals
 */
export type BatchGetLocalsResponseWritable = {
  /**
   * Locals requested.
   */
  locals?: Array<LocalWritable>;
};

/**
 * BatchGetMirrorsResponse is the response message for BatchGetMirrors
 */
export type BatchGetMirrorsResponseWritable = {
  /**
   * Mirrors requested.
   */
  mirrors?: Array<MirrorWritable>;
};

/**
 * BatchGetPublicationTargetsResponse is the response message for BatchGetPublicationTargets
 */
export type BatchGetPublicationTargetsResponseWritable = {
  /**
   * Publication targets requested.
   */
  publicationTargets?: Array<PublicationTargetWritable>;
};

/**
 * GpgKey represents a gpg key.
 */
export type GpgKeyWritable = {
  /**
   * The GPG key in ASCII armor format.
   */
  armor: string;
};

/**
 * ListLocalsResponse is the response message for ListLocals.
 */
export type ListLocalsResponseWritable = {
  /**
   * The list of local repositories.
   */
  locals?: Array<LocalWritable>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * ListMirrorsResponse is the response message for ListMirrors.
 */
export type ListMirrorsResponseWritable = {
  /**
   * The list of mirrors.
   */
  mirrors?: Array<MirrorWritable>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * ListPublicationTargetsResponse is the response message for ListPublicationTargets.
 */
export type ListPublicationTargetsResponseWritable = {
  /**
   * The publication targets.
   */
  publicationTargets?: Array<PublicationTargetWritable>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * ListPublicationsResponse is the response message for ListPublications.
 */
export type ListPublicationsResponseWritable = {
  /**
   * A list of publications.
   */
  publications?: Array<PublicationWritable>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * Local represents a local repository configuration.
 */
export type LocalWritable = {
  /**
   * The resource name of the local repository.
   * Format: "locals/{local}"
   */
  name?: string;
  /**
   * The human-readable name used to reference this local repository.
   */
  displayName: string;
  /**
   * A user-facing description of the local repository.
   */
  comment?: string;
  /**
   * The default distribution to publish.
   */
  defaultDistribution?: string;
  /**
   * The default component to publish.
   */
  defaultComponent: string;
};

/**
 * Mirror represents a repository mirror configuration and status.
 */
export type MirrorWritable = {
  /**
   * The resource name of the mirror.
   * Format: "mirrors/{mirror}"
   */
  name?: string;
  /**
   * The human-readable name used to reference this mirror.
   */
  displayName: string;
  /**
   * Root of the debian archive (URL).
   */
  archiveRoot: string;
  /**
   * The name of the distribution (subdirectory of `dists`).
   */
  distribution?: string;
  /**
   * List of architectures to fetch.
   */
  architectures?: Array<string>;
  /**
   * List of components to fetch.
   */
  components: Array<string>;
  /**
   * Indicates if installer files should be downloaded.
   */
  downloadInstaller?: boolean;
  /**
   * Indicates if sources files should be downloaded.
   */
  downloadSources?: boolean;
  /**
   * Indicates if `.udebs` files should be downloaded.
   */
  downloadUdebs?: boolean;
  /**
   * Package query applied to the mirror.
   * Follows the [Reprepro/Aptly language](https://www.aptly.info/doc/feature/query/).
   */
  filter?: string;
  /**
   * Indicates if dependencies of matching packages should be included.
   */
  filterWithDeps?: boolean;
  /**
   * Indicates if the architecture list verification should be skipped.
   */
  skipArchitectureCheck?: boolean;
  /**
   * Indicates if the component list verification should be skipped.
   */
  skipComponentCheck?: boolean;
  /**
   * Indicates if the mirror is a signature-preserving mirror.
   */
  preserveSignatures?: boolean;
  gpgKey?: GpgKeyWritable;
};

/**
 * Publication represents the configuration for publishing Debian artifacts
 * from a Source (Mirror or Local) to a Target.
 */
export type PublicationWritable = {
  /**
   * The resource name of the publication.
   * Format: "publications/{uuid}"
   */
  name?: string;
  /**
   * The human-readable name used to reference this publication.
   */
  displayName: string;
  /**
   * The resource name of the publication target.
   * Format: "publicationTargets/{publication_target}"
   */
  publicationTarget: string;
  /**
   * The resource name of the source mirror or local.
   * Format: "mirrors/{mirror}" or "locals/{local}"
   */
  source: string;
  /**
   * The name of the distribution.
   * Inferred from the source if omitted.
   */
  distribution?: string;
  /**
   * Value for the `Label:` field in the Release file.
   */
  label?: string;
  /**
   * Value for the `Origin:` field in the Release file.
   * Inferred from the source if omitted.
   */
  origin?: string;
  /**
   * List of architectures to publish.
   * Defaults to all source architectures if omitted.
   */
  architectures?: Array<string>;
  /**
   * Provides index file by hash if unique.
   */
  acquireByHash?: boolean;
  /**
   * Sets `ButAutomaticUpgrade: yes|no` in the Release file if provided.
   */
  butAutomaticUpgrades?: boolean;
  /**
   * Sets `NotAutomatic: yes|no` in the Release file if provided.
   */
  notAutomatic?: boolean;
  /**
   * Indicates if multiple distributions are supported.
   */
  multiDist?: boolean;
  /**
   * Indicates if bz2 compression should be skipped for index files.
   */
  skipBz2?: boolean;
  /**
   * Indicates if content indexes should not be generated.
   */
  skipContents?: boolean;
  gpgKey?: GpgKeyWritable;
};

/**
 * PublicationTarget represents a storage destination (currently supports S3 or Swift).
 */
export type PublicationTargetWritable = {
  /**
   * The resource name of the publication target.
   * Format: "publicationTargets/{publication_target}"
   */
  name?: string;
  /**
   * The user-defined display name.
   */
  displayName: string;
  s3?: S3Target;
  swift?: SwiftTarget;
  filesystem?: FilesystemTarget;
};

export type ListLocalsData = {
  body?: never;
  path?: never;
  query?: {
    /**
     * The maximum number of local repositories to return.
     * The service may return fewer than this value.
     * If unspecified, at most 100 local repositories will be returned.
     * The maximum value is 1000; values above 1000 will be coerced to 1000.
     */
    pageSize?: number;
    /**
     * A page token, received from a previous `ListLocals` call.
     * Provide this to retrieve the subsequent page.
     *
     * When paginating, all other parameters provided to `ListLocals` must match
     * the call that provided the page token.
     */
    pageToken?: string;
    /**
     * An expression for filtering the results of the request.
     * The fields eligible for filtering are:
     *
     * * `display_name`
     *
     * Some examples of using filters are:
     *
     * * `display_name="local"`  --> The local has the display name "local".
     * * `display_name="local*"` --> The local has the prefix "local" in its display name.
     */
    filter?: string;
  };
  url: "/v1beta1/locals";
};

export type ListLocalsErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type ListLocalsError = ListLocalsErrors[keyof ListLocalsErrors];

export type ListLocalsResponses = {
  /**
   * A successful response.
   */
  200: ListLocalsResponse;
};

export type ListLocalsResponse2 =
  ListLocalsResponses[keyof ListLocalsResponses];

export type CreateLocalData = {
  /**
   * The local repository to create.
   * The `name` field in the local is ignored.
   */
  body: LocalWritable;
  path?: never;
  query?: {
    /**
     * The ID to use for the local repository (optional).
     * If provided, the full resource name will be `locals/{local_id}`.
     */
    localId?: string;
  };
  url: "/v1beta1/locals";
};

export type CreateLocalErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type CreateLocalError = CreateLocalErrors[keyof CreateLocalErrors];

export type CreateLocalResponses = {
  /**
   * A successful response.
   */
  200: Local;
};

export type CreateLocalResponse =
  CreateLocalResponses[keyof CreateLocalResponses];

export type BatchGetLocalsData = {
  body: BatchGetLocalsRequest;
  path?: never;
  query?: never;
  url: "/v1beta1/locals:batchGet";
};

export type BatchGetLocalsErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type BatchGetLocalsError =
  BatchGetLocalsErrors[keyof BatchGetLocalsErrors];

export type BatchGetLocalsResponses = {
  /**
   * A successful response.
   */
  200: BatchGetLocalsResponse;
};

export type BatchGetLocalsResponse2 =
  BatchGetLocalsResponses[keyof BatchGetLocalsResponses];

export type ListMirrorsData = {
  body?: never;
  path?: never;
  query?: {
    /**
     * The maximum number of mirrors to return.
     * The service may return fewer than this value.
     * If unspecified, at most 100 mirrors will be returned.
     * The maximum value is 1000; values above 1000 will be coerced to 1000.
     */
    pageSize?: number;
    /**
     * A page token, received from a previous `ListMirrors` call.
     * Provide this to retrieve the subsequent page.
     *
     * When paginating, all other parameters provided to `ListMirrors` must match
     * the call that provided the page token.
     */
    pageToken?: string;
    /**
     * An expression for filtering the results of the request.
     * The fields eligible for filtering are:
     *
     * * `display_name`
     *
     * Some examples of using filters are:
     *
     * * `display_name="mirror"`  --> The mirror has the display name "mirror".
     * * `display_name="mirror*"` --> The mirror has the prefix "mirror" in its display name.
     */
    filter?: string;
  };
  url: "/v1beta1/mirrors";
};

export type ListMirrorsErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type ListMirrorsError = ListMirrorsErrors[keyof ListMirrorsErrors];

export type ListMirrorsResponses = {
  /**
   * A successful response.
   */
  200: ListMirrorsResponse;
};

export type ListMirrorsResponse2 =
  ListMirrorsResponses[keyof ListMirrorsResponses];

export type CreateMirrorData = {
  /**
   * The mirror to create.
   * The `name` field in the mirror is ignored.
   */
  body: MirrorWritable;
  path?: never;
  query?: {
    /**
     * The ID to use for the mirror (optional).
     * If provided, the full resource name will be `mirrors/{mirror_id}`.
     */
    mirrorId?: string;
  };
  url: "/v1beta1/mirrors";
};

export type CreateMirrorErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type CreateMirrorError = CreateMirrorErrors[keyof CreateMirrorErrors];

export type CreateMirrorResponses = {
  /**
   * A successful response.
   */
  200: Mirror;
};

export type CreateMirrorResponse =
  CreateMirrorResponses[keyof CreateMirrorResponses];

export type BatchGetMirrorsData = {
  body: BatchGetMirrorsRequest;
  path?: never;
  query?: never;
  url: "/v1beta1/mirrors:batchGet";
};

export type BatchGetMirrorsErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type BatchGetMirrorsError =
  BatchGetMirrorsErrors[keyof BatchGetMirrorsErrors];

export type BatchGetMirrorsResponses = {
  /**
   * A successful response.
   */
  200: BatchGetMirrorsResponse;
};

export type BatchGetMirrorsResponse2 =
  BatchGetMirrorsResponses[keyof BatchGetMirrorsResponses];

export type ListPublicationTargetsData = {
  body?: never;
  path?: never;
  query?: {
    /**
     * The maximum number of publication targets to return.
     * The service may return fewer than this value.
     * If unspecified, at most 100 publication targets will be returned.
     * The maximum value is 1000; values above 1000 will be coerced to 1000.
     */
    pageSize?: number;
    /**
     * A page token, received from a previous `ListPublicationTargets` call.
     * Provide this to retrieve the subsequent page.
     *
     * When paginating, all other parameters provided to `ListPublicationTargets` must match
     * the call that provided the page token.
     */
    pageToken?: string;
    /**
     * An expression for filtering the results of the request.
     * The fields eligible for filtering are:
     *
     * * `display_name`
     *
     * Some examples of using filters are:
     *
     * * `display_name="target"`  --> The publication target has the display name "target".
     * * `display_name="target*"` --> The publication target has the prefix "target" in its display name.
     */
    filter?: string;
  };
  url: "/v1beta1/publicationTargets";
};

export type ListPublicationTargetsErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type ListPublicationTargetsError =
  ListPublicationTargetsErrors[keyof ListPublicationTargetsErrors];

export type ListPublicationTargetsResponses = {
  /**
   * A successful response.
   */
  200: ListPublicationTargetsResponse;
};

export type ListPublicationTargetsResponse2 =
  ListPublicationTargetsResponses[keyof ListPublicationTargetsResponses];

export type CreatePublicationTargetData = {
  /**
   * The resource to create.
   */
  body: PublicationTargetWritable;
  path?: never;
  query?: {
    /**
     * The ID to use for the publication target (optional).
     * If not provided, the server will generate one.
     */
    publicationTargetId?: string;
  };
  url: "/v1beta1/publicationTargets";
};

export type CreatePublicationTargetErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type CreatePublicationTargetError =
  CreatePublicationTargetErrors[keyof CreatePublicationTargetErrors];

export type CreatePublicationTargetResponses = {
  /**
   * A successful response.
   */
  200: PublicationTarget;
};

export type CreatePublicationTargetResponse =
  CreatePublicationTargetResponses[keyof CreatePublicationTargetResponses];

export type BatchGetPublicationTargetsData = {
  body: BatchGetPublicationTargetsRequest;
  path?: never;
  query?: never;
  url: "/v1beta1/publicationTargets:batchGet";
};

export type BatchGetPublicationTargetsErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type BatchGetPublicationTargetsError =
  BatchGetPublicationTargetsErrors[keyof BatchGetPublicationTargetsErrors];

export type BatchGetPublicationTargetsResponses = {
  /**
   * A successful response.
   */
  200: BatchGetPublicationTargetsResponse;
};

export type BatchGetPublicationTargetsResponse2 =
  BatchGetPublicationTargetsResponses[keyof BatchGetPublicationTargetsResponses];

export type ListPublicationsData = {
  body?: never;
  path?: never;
  query?: {
    /**
     * The maximum number of publications to return.
     * The service may return fewer than this value.
     * If unspecified, at most 100 publications will be returned.
     * The maximum value is 1000; values above 1000 will be coerced to 1000.
     */
    pageSize?: number;
    /**
     * A page token, received from a previous `ListPublication` call.
     * Provide this to retrieve the subsequent page.
     *
     * When paginating, all other parameters provided to `ListPublication` must match
     * the call that provided the page token.
     */
    pageToken?: string;
    /**
     * An expression for filtering the results of the request.
     * The fields eligible for filtering are:
     *
     * * `source`
     * * `publication_target`
     *
     * Some examples of using filters are:
     *
     * * `source="locals/123"`  --> The publication has source "locals/123".
     * * `publication_target="publicationTargets/456"` --> The publication has target "publicationTargets/456".
     */
    filter?: string;
  };
  url: "/v1beta1/publications";
};

export type ListPublicationsErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type ListPublicationsError =
  ListPublicationsErrors[keyof ListPublicationsErrors];

export type ListPublicationsResponses = {
  /**
   * A successful response.
   */
  200: ListPublicationsResponse;
};

export type ListPublicationsResponse2 =
  ListPublicationsResponses[keyof ListPublicationsResponses];

export type CreatePublicationData = {
  /**
   * The publication to create.
   */
  body: PublicationWritable;
  path?: never;
  query?: {
    /**
     * The ID to use for the publication (optional).
     */
    publicationId?: string;
  };
  url: "/v1beta1/publications";
};

export type CreatePublicationErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type CreatePublicationError =
  CreatePublicationErrors[keyof CreatePublicationErrors];

export type CreatePublicationResponses = {
  /**
   * A successful response.
   */
  200: Publication;
};

export type CreatePublicationResponse =
  CreatePublicationResponses[keyof CreatePublicationResponses];

export type UpdateLocalData = {
  /**
   * The local repository to update.
   *
   * The local repository to update.
   */
  body: {
    /**
     * The local repository ID is exposed separately for convenience.
     */
    readonly localId?: string;
    /**
     * The human-readable name used to reference this local repository.
     */
    displayName: string;
    /**
     * A user-facing description of the local repository.
     */
    comment?: string;
    /**
     * The default distribution to publish.
     */
    defaultDistribution?: string;
    /**
     * The default component to publish.
     */
    defaultComponent: string;
    /**
     * The last associated LRO with the local
     */
    readonly lastOperation?: string;
  };
  path: {
    /**
     * The resource name of the local repository.
     * Format: "locals/{local}"
     */
    "local.name": string;
  };
  query?: never;
  url: "/v1beta1/{local.name}";
};

export type UpdateLocalErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type UpdateLocalError = UpdateLocalErrors[keyof UpdateLocalErrors];

export type UpdateLocalResponses = {
  /**
   * A successful response.
   */
  200: Local;
};

export type UpdateLocalResponse =
  UpdateLocalResponses[keyof UpdateLocalResponses];

export type UpdateMirrorData = {
  /**
   * The mirror to update.
   *
   * The mirror to update.
   */
  body: {
    /**
     * The mirror ID is exposed separately for convenience.
     */
    readonly mirrorId?: string;
    /**
     * The status of the mirror.
     */
    readonly status?: string;
    /**
     * The timestamp of the last successful download.
     */
    readonly lastDownloadDate?: Date;
    /**
     * The human-readable name used to reference this mirror.
     */
    displayName: string;
    /**
     * Root of the debian archive (URL).
     */
    archiveRoot: string;
    /**
     * The name of the distribution (subdirectory of `dists`).
     */
    distribution?: string;
    /**
     * List of architectures to fetch.
     */
    architectures?: Array<string>;
    /**
     * List of components to fetch.
     */
    components: Array<string>;
    /**
     * Indicates if installer files should be downloaded.
     */
    downloadInstaller?: boolean;
    /**
     * Indicates if sources files should be downloaded.
     */
    downloadSources?: boolean;
    /**
     * Indicates if `.udebs` files should be downloaded.
     */
    downloadUdebs?: boolean;
    /**
     * Package query applied to the mirror.
     * Follows the [Reprepro/Aptly language](https://www.aptly.info/doc/feature/query/).
     */
    filter?: string;
    /**
     * Indicates if dependencies of matching packages should be included.
     */
    filterWithDeps?: boolean;
    /**
     * Indicates if the architecture list verification should be skipped.
     */
    skipArchitectureCheck?: boolean;
    /**
     * Indicates if the component list verification should be skipped.
     */
    skipComponentCheck?: boolean;
    /**
     * Indicates if the mirror is a signature-preserving mirror.
     */
    preserveSignatures?: boolean;
    gpgKey?: GpgKeyWritable;
    /**
     * The last associated LRO with the mirror
     */
    readonly lastOperation?: string;
  };
  path: {
    /**
     * The resource name of the mirror.
     * Format: "mirrors/{mirror}"
     */
    "mirror.name": string;
  };
  query?: never;
  url: "/v1beta1/{mirror.name}";
};

export type UpdateMirrorErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type UpdateMirrorError = UpdateMirrorErrors[keyof UpdateMirrorErrors];

export type UpdateMirrorResponses = {
  /**
   * A successful response.
   */
  200: Mirror;
};

export type UpdateMirrorResponse =
  UpdateMirrorResponses[keyof UpdateMirrorResponses];

export type DeleteMirrorData = {
  body?: never;
  path: {
    /**
     * The resource name of the mirror to delete.
     * Format: "mirrors/{mirror}"
     */
    name_1: string;
  };
  query?: {
    /**
     * Flag for if you want to force delete the mirror
     */
    force?: boolean;
  };
  url: "/v1beta1/{name_1}";
};

export type DeleteMirrorErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type DeleteMirrorError = DeleteMirrorErrors[keyof DeleteMirrorErrors];

export type DeleteMirrorResponses = {
  /**
   * A successful response.
   */
  200: {
    [key: string]: unknown;
  };
};

export type DeleteMirrorResponse =
  DeleteMirrorResponses[keyof DeleteMirrorResponses];

export type GetMirrorData = {
  body?: never;
  path: {
    /**
     * The resource name of the mirror to retrieve.
     * Format: "mirrors/{mirror}"
     */
    name_1: string;
  };
  query?: never;
  url: "/v1beta1/{name_1}";
};

export type GetMirrorErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type GetMirrorError = GetMirrorErrors[keyof GetMirrorErrors];

export type GetMirrorResponses = {
  /**
   * A successful response.
   */
  200: Mirror;
};

export type GetMirrorResponse = GetMirrorResponses[keyof GetMirrorResponses];

export type DeletePublicationData = {
  body?: never;
  path: {
    /**
     * The resource name of the publication to delete.
     * Format: "publications/{publication}"
     */
    name_2: string;
  };
  query?: never;
  url: "/v1beta1/{name_2}";
};

export type DeletePublicationErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type DeletePublicationError =
  DeletePublicationErrors[keyof DeletePublicationErrors];

export type DeletePublicationResponses = {
  /**
   * A successful response.
   */
  200: {
    [key: string]: unknown;
  };
};

export type DeletePublicationResponse =
  DeletePublicationResponses[keyof DeletePublicationResponses];

export type GetOperationData = {
  body?: never;
  path: {
    /**
     * The resource name of the mirror to retrieve.
     * Format: "operations/{operation}"
     */
    name_2: string;
  };
  query?: never;
  url: "/v1beta1/{name_2}";
};

export type GetOperationErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type GetOperationError = GetOperationErrors[keyof GetOperationErrors];

export type GetOperationResponses = {
  /**
   * A successful response.
   */
  200: Operation;
};

export type GetOperationResponse =
  GetOperationResponses[keyof GetOperationResponses];

export type DeletePublicationTargetData = {
  body?: never;
  path: {
    /**
     * The name of the publication target to delete.
     * Format: "publicationTargets/{publication_target}"
     */
    name_3: string;
  };
  query?: never;
  url: "/v1beta1/{name_3}";
};

export type DeletePublicationTargetErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type DeletePublicationTargetError =
  DeletePublicationTargetErrors[keyof DeletePublicationTargetErrors];

export type DeletePublicationTargetResponses = {
  /**
   * A successful response.
   */
  200: {
    [key: string]: unknown;
  };
};

export type DeletePublicationTargetResponse =
  DeletePublicationTargetResponses[keyof DeletePublicationTargetResponses];

export type GetPublicationData = {
  body?: never;
  path: {
    /**
     * The resource name of the publication to retrieve.
     * Format: "publications/{publication}"
     */
    name_3: string;
  };
  query?: never;
  url: "/v1beta1/{name_3}";
};

export type GetPublicationErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type GetPublicationError =
  GetPublicationErrors[keyof GetPublicationErrors];

export type GetPublicationResponses = {
  /**
   * A successful response.
   */
  200: Publication;
};

export type GetPublicationResponse =
  GetPublicationResponses[keyof GetPublicationResponses];

export type GetPublicationTargetData = {
  body?: never;
  path: {
    /**
     * The name of the publication target to retrieve.
     * Format: "publicationTargets/{publication_target}"
     */
    name_4: string;
  };
  query?: never;
  url: "/v1beta1/{name_4}";
};

export type GetPublicationTargetErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type GetPublicationTargetError =
  GetPublicationTargetErrors[keyof GetPublicationTargetErrors];

export type GetPublicationTargetResponses = {
  /**
   * A successful response.
   */
  200: PublicationTarget;
};

export type GetPublicationTargetResponse =
  GetPublicationTargetResponses[keyof GetPublicationTargetResponses];

export type DeleteLocalData = {
  body?: never;
  path: {
    /**
     * The resource name of the local repository to delete.
     * Format: "locals/{local}"
     */
    name: string;
  };
  query?: never;
  url: "/v1beta1/{name}";
};

export type DeleteLocalErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type DeleteLocalError = DeleteLocalErrors[keyof DeleteLocalErrors];

export type DeleteLocalResponses = {
  /**
   * A successful response.
   */
  200: {
    [key: string]: unknown;
  };
};

export type DeleteLocalResponse =
  DeleteLocalResponses[keyof DeleteLocalResponses];

export type GetLocalData = {
  body?: never;
  path: {
    /**
     * The resource name of the local repository to retrieve.
     * Format: "locals/{local}"
     */
    name: string;
  };
  query?: never;
  url: "/v1beta1/{name}";
};

export type GetLocalErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type GetLocalError = GetLocalErrors[keyof GetLocalErrors];

export type GetLocalResponses = {
  /**
   * A successful response.
   */
  200: Local;
};

export type GetLocalResponse = GetLocalResponses[keyof GetLocalResponses];

export type ImportLocalPackagesData = {
  body: LocalServiceImportLocalPackagesBody;
  path: {
    /**
     * The resource name of the local repository to import packages into.
     * Format: "locals/{local}"
     */
    name: string;
  };
  query?: never;
  url: "/v1beta1/{name}:importPackages";
};

export type ImportLocalPackagesErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type ImportLocalPackagesError =
  ImportLocalPackagesErrors[keyof ImportLocalPackagesErrors];

export type ImportLocalPackagesResponses = {
  /**
   * A successful response.
   */
  200: Operation;
};

export type ImportLocalPackagesResponse =
  ImportLocalPackagesResponses[keyof ImportLocalPackagesResponses];

export type PublishPublicationData = {
  body: PublicationServicePublishPublicationBody;
  path: {
    /**
     * The resource name of the publication to publish.
     * Format: "publications/{publication}"
     */
    name: string;
  };
  query?: never;
  url: "/v1beta1/{name}:publish";
};

export type PublishPublicationErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type PublishPublicationError =
  PublishPublicationErrors[keyof PublishPublicationErrors];

export type PublishPublicationResponses = {
  /**
   * A successful response.
   */
  200: Operation;
};

export type PublishPublicationResponse =
  PublishPublicationResponses[keyof PublishPublicationResponses];

export type SyncMirrorData = {
  body: MirrorServiceSyncMirrorBody;
  path: {
    /**
     * The resource name of the mirror to sync.
     * Format: "mirrors/{mirror}"
     */
    name: string;
  };
  query?: never;
  url: "/v1beta1/{name}:sync";
};

export type SyncMirrorErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type SyncMirrorError = SyncMirrorErrors[keyof SyncMirrorErrors];

export type SyncMirrorResponses = {
  /**
   * A successful response.
   */
  200: Operation;
};

export type SyncMirrorResponse = SyncMirrorResponses[keyof SyncMirrorResponses];

export type ListMirrorPackagesData = {
  body?: never;
  path: {
    /**
     * The parent, which owns this collection of packages.
     * Format: "mirrors/{mirror}"
     */
    parent_1: string;
  };
  query?: {
    /**
     * The maximum number of packages to return.
     * The service may return fewer than this value.
     * If unspecified, at most 100 packages will be returned.
     * The maximum value is 1000; values above 1000 will be coerced to 1000.
     */
    pageSize?: number;
    /**
     * A page token, received from a previous `ListMirrorPackages` call.
     * Provide this to retrieve the subsequent page.
     *
     * When paginating, all other parameters provided to `ListMirrorPackages` must match
     * the call that provided the page token.
     */
    pageToken?: string;
  };
  url: "/v1beta1/{parent_1}/packages";
};

export type ListMirrorPackagesErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type ListMirrorPackagesError =
  ListMirrorPackagesErrors[keyof ListMirrorPackagesErrors];

export type ListMirrorPackagesResponses = {
  /**
   * A successful response.
   */
  200: ListMirrorPackagesResponse;
};

export type ListMirrorPackagesResponse2 =
  ListMirrorPackagesResponses[keyof ListMirrorPackagesResponses];

export type ListLocalPackagesData = {
  body?: never;
  path: {
    /**
     * The parent, which owns this collection of packages.
     * Format: "locals/{local}"
     */
    parent: string;
  };
  query?: {
    /**
     * The maximum number of packages to return.
     * The service may return fewer than this value.
     * If unspecified, at most 100 packages will be returned.
     * The maximum value is 1000; values above 1000 will be coerced to 1000.
     */
    pageSize?: number;
    /**
     * A page token, received from a previous `ListLocalPackages` call.
     * Provide this to retrieve the subsequent page.
     *
     * When paginating, all other parameters provided to `ListLocalPackages` must match
     * the call that provided the page token.
     */
    pageToken?: string;
  };
  url: "/v1beta1/{parent}/packages";
};

export type ListLocalPackagesErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type ListLocalPackagesError =
  ListLocalPackagesErrors[keyof ListLocalPackagesErrors];

export type ListLocalPackagesResponses = {
  /**
   * A successful response.
   */
  200: ListLocalPackagesResponse;
};

export type ListLocalPackagesResponse2 =
  ListLocalPackagesResponses[keyof ListLocalPackagesResponses];

export type UpdatePublicationData = {
  /**
   * The publication to update.
   *
   * The publication to update.
   */
  body: {
    /**
     * The human-readable name used to reference this publication.
     */
    displayName: string;
    /**
     * The publication ID (UUID) is exposed separately for convenience.
     */
    readonly publicationId?: string;
    /**
     * The resource name of the publication target.
     * Format: "publicationTargets/{publication_target}"
     */
    publicationTarget: string;
    /**
     * The resource name of the source mirror or local.
     * Format: "mirrors/{mirror}" or "locals/{local}"
     */
    source: string;
    /**
     * The name of the distribution.
     * Inferred from the source if omitted.
     */
    distribution?: string;
    /**
     * Value for the `Label:` field in the Release file.
     */
    label?: string;
    /**
     * Value for the `Origin:` field in the Release file.
     * Inferred from the source if omitted.
     */
    origin?: string;
    /**
     * List of architectures to publish.
     * Defaults to all source architectures if omitted.
     */
    architectures?: Array<string>;
    /**
     * Provides index file by hash if unique.
     */
    acquireByHash?: boolean;
    /**
     * Sets `ButAutomaticUpgrade: yes|no` in the Release file if provided.
     */
    butAutomaticUpgrades?: boolean;
    /**
     * Sets `NotAutomatic: yes|no` in the Release file if provided.
     */
    notAutomatic?: boolean;
    /**
     * Indicates if multiple distributions are supported.
     */
    multiDist?: boolean;
    /**
     * Indicates if bz2 compression should be skipped for index files.
     */
    skipBz2?: boolean;
    /**
     * Indicates if content indexes should not be generated.
     */
    skipContents?: boolean;
    gpgKey?: GpgKeyWritable;
    /**
     * The timestamp of the last successful publication.
     */
    readonly publishTime?: Date;
    /**
     * The last associated LRO with the publication
     */
    readonly lastOperation?: string;
  };
  path: {
    /**
     * The resource name of the publication.
     * Format: "publications/{uuid}"
     */
    "publication.name": string;
  };
  query?: never;
  url: "/v1beta1/{publication.name}";
};

export type UpdatePublicationErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type UpdatePublicationError =
  UpdatePublicationErrors[keyof UpdatePublicationErrors];

export type UpdatePublicationResponses = {
  /**
   * A successful response.
   */
  200: Publication;
};

export type UpdatePublicationResponse =
  UpdatePublicationResponses[keyof UpdatePublicationResponses];

export type UpdatePublicationTargetData = {
  /**
   * The resource to update.
   *
   * The resource to update.
   */
  body: {
    /**
     * The user-defined display name.
     */
    displayName: string;
    /**
     * The publication_id (uuid) is part of the `name`, and is exposed separately for convenience.
     */
    readonly publicationTargetId?: string;
    s3?: S3Target;
    swift?: SwiftTarget;
    filesystem?: FilesystemTarget;
  };
  path: {
    /**
     * The resource name of the publication target.
     * Format: "publicationTargets/{publication_target}"
     */
    "publicationTarget.name": string;
  };
  query?: never;
  url: "/v1beta1/{publicationTarget.name}";
};

export type UpdatePublicationTargetErrors = {
  /**
   * An unexpected error response.
   */
  default: Status;
};

export type UpdatePublicationTargetError =
  UpdatePublicationTargetErrors[keyof UpdatePublicationTargetErrors];

export type UpdatePublicationTargetResponses = {
  /**
   * A successful response.
   */
  200: PublicationTarget;
};

export type UpdatePublicationTargetResponse =
  UpdatePublicationTargetResponses[keyof UpdatePublicationTargetResponses];
