// @canonical/react-components@3.11.1/dist/components/Stepper/Step/Step.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Default = void 0;
var _react = _interopRequireDefault(require("react"));
var _Step = _interopRequireDefault(require("./Step"));
var _Stepper = _interopRequireDefault(require("../Stepper"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
const meta = {
  component: _Step.default,
  render: args => /*#__PURE__*/_react.default.createElement(_Stepper.default, {
    variant: "horizontal",
    steps: [/*#__PURE__*/_react.default.createElement(_Step.default, _extends({
      key: "step"
    }, args))]
  }),
  tags: ["autodocs"]
};
var _default = exports.default = meta;
const Default = exports.Default = {
  name: "Default",
  args: {
    title: "Step 1",
    index: 1,
    enabled: false,
    hasProgressLine: false,
    iconName: "number",
    handleClick: () => {}
  }
};
