// @canonical/landscape-openapi@0.0.70/generated/sdk.gen.ts
// This file is auto-generated by @hey-api/openapi-ts

import type {
  Client,
  ClientMeta,
  Options as Options2,
  RequestResult,
  TDataShape,
} from "./client";
import { client } from "./client.gen";
import {
  localServiceBatchGetLocalsResponseTransformer,
  localServiceCreateLocalResponseTransformer,
  localServiceGetLocalResponseTransformer,
  localServiceListLocalsResponseTransformer,
  localServiceUpdateLocalResponseTransformer,
  mirrorServiceBatchGetMirrorsResponseTransformer,
  mirrorServiceCreateMirrorResponseTransformer,
  mirrorServiceGetMirrorResponseTransformer,
  mirrorServiceListMirrorsResponseTransformer,
  mirrorServiceUpdateMirrorResponseTransformer,
  publicationServiceCreatePublicationResponseTransformer,
  publicationServiceGetPublicationResponseTransformer,
  publicationServiceListPublicationsResponseTransformer,
  publicationServiceUpdatePublicationResponseTransformer,
} from "./transformers.gen";
import type {
  LocalServiceBatchGetLocalsData,
  LocalServiceBatchGetLocalsErrors,
  LocalServiceBatchGetLocalsResponses,
  LocalServiceCreateLocalData,
  LocalServiceCreateLocalErrors,
  LocalServiceCreateLocalResponses,
  LocalServiceDeleteLocalData,
  LocalServiceDeleteLocalErrors,
  LocalServiceDeleteLocalResponses,
  LocalServiceGetLocalData,
  LocalServiceGetLocalErrors,
  LocalServiceGetLocalResponses,
  LocalServiceImportLocalPackagesData,
  LocalServiceImportLocalPackagesErrors,
  LocalServiceImportLocalPackagesResponses,
  LocalServiceListLocalPackagesData,
  LocalServiceListLocalPackagesErrors,
  LocalServiceListLocalPackagesResponses,
  LocalServiceListLocalsData,
  LocalServiceListLocalsErrors,
  LocalServiceListLocalsResponses,
  LocalServiceUpdateLocalData,
  LocalServiceUpdateLocalErrors,
  LocalServiceUpdateLocalResponses,
  MirrorServiceBatchGetMirrorsData,
  MirrorServiceBatchGetMirrorsErrors,
  MirrorServiceBatchGetMirrorsResponses,
  MirrorServiceCreateMirrorData,
  MirrorServiceCreateMirrorErrors,
  MirrorServiceCreateMirrorResponses,
  MirrorServiceDeleteMirrorData,
  MirrorServiceDeleteMirrorErrors,
  MirrorServiceDeleteMirrorResponses,
  MirrorServiceGetMirrorData,
  MirrorServiceGetMirrorErrors,
  MirrorServiceGetMirrorResponses,
  MirrorServiceListMirrorPackagesData,
  MirrorServiceListMirrorPackagesErrors,
  MirrorServiceListMirrorPackagesResponses,
  MirrorServiceListMirrorsData,
  MirrorServiceListMirrorsErrors,
  MirrorServiceListMirrorsResponses,
  MirrorServiceSyncMirrorData,
  MirrorServiceSyncMirrorErrors,
  MirrorServiceSyncMirrorResponses,
  MirrorServiceUpdateMirrorData,
  MirrorServiceUpdateMirrorErrors,
  MirrorServiceUpdateMirrorResponses,
  OperationServiceGetOperationData,
  OperationServiceGetOperationErrors,
  OperationServiceGetOperationResponses,
  PublicationServiceCreatePublicationData,
  PublicationServiceCreatePublicationErrors,
  PublicationServiceCreatePublicationResponses,
  PublicationServiceDeletePublicationData,
  PublicationServiceDeletePublicationErrors,
  PublicationServiceDeletePublicationResponses,
  PublicationServiceGetPublicationData,
  PublicationServiceGetPublicationErrors,
  PublicationServiceGetPublicationResponses,
  PublicationServiceListPublicationsData,
  PublicationServiceListPublicationsErrors,
  PublicationServiceListPublicationsResponses,
  PublicationServicePublishPublicationData,
  PublicationServicePublishPublicationErrors,
  PublicationServicePublishPublicationResponses,
  PublicationServiceUpdatePublicationData,
  PublicationServiceUpdatePublicationErrors,
  PublicationServiceUpdatePublicationResponses,
  PublicationTargetServiceBatchGetPublicationTargetsData,
  PublicationTargetServiceBatchGetPublicationTargetsErrors,
  PublicationTargetServiceBatchGetPublicationTargetsResponses,
  PublicationTargetServiceCreatePublicationTargetData,
  PublicationTargetServiceCreatePublicationTargetErrors,
  PublicationTargetServiceCreatePublicationTargetResponses,
  PublicationTargetServiceDeletePublicationTargetData,
  PublicationTargetServiceDeletePublicationTargetErrors,
  PublicationTargetServiceDeletePublicationTargetResponses,
  PublicationTargetServiceGetPublicationTargetData,
  PublicationTargetServiceGetPublicationTargetErrors,
  PublicationTargetServiceGetPublicationTargetResponses,
  PublicationTargetServiceListPublicationTargetsData,
  PublicationTargetServiceListPublicationTargetsErrors,
  PublicationTargetServiceListPublicationTargetsResponses,
  PublicationTargetServiceUpdatePublicationTargetData,
  PublicationTargetServiceUpdatePublicationTargetErrors,
  PublicationTargetServiceUpdatePublicationTargetResponses,
} from "./types.gen";

export type Options<
  TData extends TDataShape = TDataShape,
  ThrowOnError extends boolean = boolean,
  TResponse = unknown,
> = Options2<TData, ThrowOnError, TResponse> & {
  /**
   * You can provide a client instance returned by `createClient()` instead of
   * individual options. This might be also useful if you want to implement a
   * custom client.
   */
  client?: Client;
  /**
   * You can pass arbitrary values through the `meta` object. This can be
   * used to access values that aren't defined as part of the SDK function.
   */
  meta?: keyof ClientMeta extends never ? Record<string, unknown> : ClientMeta;
};

/**
 * ListLocals returns a list of Locals.
 */
export const localServiceListLocals = <ThrowOnError extends boolean = false>(
  options?: Options<LocalServiceListLocalsData, ThrowOnError>,
): RequestResult<
  LocalServiceListLocalsResponses,
  LocalServiceListLocalsErrors,
  ThrowOnError
> =>
  (options?.client ?? client).get<
    LocalServiceListLocalsResponses,
    LocalServiceListLocalsErrors,
    ThrowOnError
  >({
    responseTransformer: localServiceListLocalsResponseTransformer,
    responseType: "json",
    url: "/v1beta1/locals",
    ...options,
  });

/**
 * CreateLocal creates a new Local repository.
 */
export const localServiceCreateLocal = <ThrowOnError extends boolean = false>(
  options: Options<LocalServiceCreateLocalData, ThrowOnError>,
): RequestResult<
  LocalServiceCreateLocalResponses,
  LocalServiceCreateLocalErrors,
  ThrowOnError
> =>
  (options.client ?? client).post<
    LocalServiceCreateLocalResponses,
    LocalServiceCreateLocalErrors,
    ThrowOnError
  >({
    responseTransformer: localServiceCreateLocalResponseTransformer,
    responseType: "json",
    url: "/v1beta1/locals",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * DeleteLocal deletes a Local.
 */
export const localServiceDeleteLocal = <ThrowOnError extends boolean = false>(
  options: Options<LocalServiceDeleteLocalData, ThrowOnError>,
): RequestResult<
  LocalServiceDeleteLocalResponses,
  LocalServiceDeleteLocalErrors,
  ThrowOnError
> =>
  (options.client ?? client).delete<
    LocalServiceDeleteLocalResponses,
    LocalServiceDeleteLocalErrors,
    ThrowOnError
  >({ url: "/v1beta1/locals/{local}", ...options });

/**
 * GetLocal retrieves a Local by its resource name.
 */
export const localServiceGetLocal = <ThrowOnError extends boolean = false>(
  options: Options<LocalServiceGetLocalData, ThrowOnError>,
): RequestResult<
  LocalServiceGetLocalResponses,
  LocalServiceGetLocalErrors,
  ThrowOnError
> =>
  (options.client ?? client).get<
    LocalServiceGetLocalResponses,
    LocalServiceGetLocalErrors,
    ThrowOnError
  >({
    responseTransformer: localServiceGetLocalResponseTransformer,
    responseType: "json",
    url: "/v1beta1/locals/{local}",
    ...options,
  });

/**
 * UpdateLocal updates the details of a Local.
 */
export const localServiceUpdateLocal = <ThrowOnError extends boolean = false>(
  options: Options<LocalServiceUpdateLocalData, ThrowOnError>,
): RequestResult<
  LocalServiceUpdateLocalResponses,
  LocalServiceUpdateLocalErrors,
  ThrowOnError
> =>
  (options.client ?? client).patch<
    LocalServiceUpdateLocalResponses,
    LocalServiceUpdateLocalErrors,
    ThrowOnError
  >({
    responseTransformer: localServiceUpdateLocalResponseTransformer,
    responseType: "json",
    url: "/v1beta1/locals/{local}",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * ListLocalPackages retrieves a list of packages available in the Local.
 */
export const localServiceListLocalPackages = <
  ThrowOnError extends boolean = false,
>(
  options: Options<LocalServiceListLocalPackagesData, ThrowOnError>,
): RequestResult<
  LocalServiceListLocalPackagesResponses,
  LocalServiceListLocalPackagesErrors,
  ThrowOnError
> =>
  (options.client ?? client).get<
    LocalServiceListLocalPackagesResponses,
    LocalServiceListLocalPackagesErrors,
    ThrowOnError
  >({
    responseType: "json",
    url: "/v1beta1/locals/{local}/packages",
    ...options,
  });

/**
 * ImportLocalPackages imports packages from a URL pointing to a package list or index
 * into the Local. This is an async operation; poll the returned Operation for status.
 */
export const localServiceImportLocalPackages = <
  ThrowOnError extends boolean = false,
>(
  options: Options<LocalServiceImportLocalPackagesData, ThrowOnError>,
): RequestResult<
  LocalServiceImportLocalPackagesResponses,
  LocalServiceImportLocalPackagesErrors,
  ThrowOnError
> =>
  (options.client ?? client).post<
    LocalServiceImportLocalPackagesResponses,
    LocalServiceImportLocalPackagesErrors,
    ThrowOnError
  >({
    responseType: "json",
    url: "/v1beta1/locals/{local}:importPackages",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * BatchGetLocals retrieves a list of locals based on their name.
 */
export const localServiceBatchGetLocals = <
  ThrowOnError extends boolean = false,
>(
  options: Options<LocalServiceBatchGetLocalsData, ThrowOnError>,
): RequestResult<
  LocalServiceBatchGetLocalsResponses,
  LocalServiceBatchGetLocalsErrors,
  ThrowOnError
> =>
  (options.client ?? client).post<
    LocalServiceBatchGetLocalsResponses,
    LocalServiceBatchGetLocalsErrors,
    ThrowOnError
  >({
    responseTransformer: localServiceBatchGetLocalsResponseTransformer,
    responseType: "json",
    url: "/v1beta1/locals:batchGet",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * ListMirrors returns a list of Mirrors.
 */
export const mirrorServiceListMirrors = <ThrowOnError extends boolean = false>(
  options?: Options<MirrorServiceListMirrorsData, ThrowOnError>,
): RequestResult<
  MirrorServiceListMirrorsResponses,
  MirrorServiceListMirrorsErrors,
  ThrowOnError
> =>
  (options?.client ?? client).get<
    MirrorServiceListMirrorsResponses,
    MirrorServiceListMirrorsErrors,
    ThrowOnError
  >({
    responseTransformer: mirrorServiceListMirrorsResponseTransformer,
    responseType: "json",
    url: "/v1beta1/mirrors",
    ...options,
  });

/**
 * CreateMirror creates a new Mirror.
 * It validates the upstream but does not initiate synchronization.
 */
export const mirrorServiceCreateMirror = <ThrowOnError extends boolean = false>(
  options: Options<MirrorServiceCreateMirrorData, ThrowOnError>,
): RequestResult<
  MirrorServiceCreateMirrorResponses,
  MirrorServiceCreateMirrorErrors,
  ThrowOnError
> =>
  (options.client ?? client).post<
    MirrorServiceCreateMirrorResponses,
    MirrorServiceCreateMirrorErrors,
    ThrowOnError
  >({
    responseTransformer: mirrorServiceCreateMirrorResponseTransformer,
    responseType: "json",
    url: "/v1beta1/mirrors",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * DeleteMirror deletes a Mirror.
 */
export const mirrorServiceDeleteMirror = <ThrowOnError extends boolean = false>(
  options: Options<MirrorServiceDeleteMirrorData, ThrowOnError>,
): RequestResult<
  MirrorServiceDeleteMirrorResponses,
  MirrorServiceDeleteMirrorErrors,
  ThrowOnError
> =>
  (options.client ?? client).delete<
    MirrorServiceDeleteMirrorResponses,
    MirrorServiceDeleteMirrorErrors,
    ThrowOnError
  >({ url: "/v1beta1/mirrors/{mirror}", ...options });

/**
 * GetMirror retrieves a Mirror by its resource name.
 */
export const mirrorServiceGetMirror = <ThrowOnError extends boolean = false>(
  options: Options<MirrorServiceGetMirrorData, ThrowOnError>,
): RequestResult<
  MirrorServiceGetMirrorResponses,
  MirrorServiceGetMirrorErrors,
  ThrowOnError
> =>
  (options.client ?? client).get<
    MirrorServiceGetMirrorResponses,
    MirrorServiceGetMirrorErrors,
    ThrowOnError
  >({
    responseTransformer: mirrorServiceGetMirrorResponseTransformer,
    responseType: "json",
    url: "/v1beta1/mirrors/{mirror}",
    ...options,
  });

/**
 * UpdateMirror updates the details of a Mirror.
 */
export const mirrorServiceUpdateMirror = <ThrowOnError extends boolean = false>(
  options: Options<MirrorServiceUpdateMirrorData, ThrowOnError>,
): RequestResult<
  MirrorServiceUpdateMirrorResponses,
  MirrorServiceUpdateMirrorErrors,
  ThrowOnError
> =>
  (options.client ?? client).patch<
    MirrorServiceUpdateMirrorResponses,
    MirrorServiceUpdateMirrorErrors,
    ThrowOnError
  >({
    responseTransformer: mirrorServiceUpdateMirrorResponseTransformer,
    responseType: "json",
    url: "/v1beta1/mirrors/{mirror}",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * ListMirrorPackages retrieves a list of packages available in the Mirror.
 */
export const mirrorServiceListMirrorPackages = <
  ThrowOnError extends boolean = false,
>(
  options: Options<MirrorServiceListMirrorPackagesData, ThrowOnError>,
): RequestResult<
  MirrorServiceListMirrorPackagesResponses,
  MirrorServiceListMirrorPackagesErrors,
  ThrowOnError
> =>
  (options.client ?? client).get<
    MirrorServiceListMirrorPackagesResponses,
    MirrorServiceListMirrorPackagesErrors,
    ThrowOnError
  >({
    responseType: "json",
    url: "/v1beta1/mirrors/{mirror}/packages",
    ...options,
  });

/**
 * SyncMirror synchronizes a Mirror.
 */
export const mirrorServiceSyncMirror = <ThrowOnError extends boolean = false>(
  options: Options<MirrorServiceSyncMirrorData, ThrowOnError>,
): RequestResult<
  MirrorServiceSyncMirrorResponses,
  MirrorServiceSyncMirrorErrors,
  ThrowOnError
> =>
  (options.client ?? client).post<
    MirrorServiceSyncMirrorResponses,
    MirrorServiceSyncMirrorErrors,
    ThrowOnError
  >({
    responseType: "json",
    url: "/v1beta1/mirrors/{mirror}:sync",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * BatchGetMirrors retrieves a list of mirrors based on their name.
 */
export const mirrorServiceBatchGetMirrors = <
  ThrowOnError extends boolean = false,
>(
  options: Options<MirrorServiceBatchGetMirrorsData, ThrowOnError>,
): RequestResult<
  MirrorServiceBatchGetMirrorsResponses,
  MirrorServiceBatchGetMirrorsErrors,
  ThrowOnError
> =>
  (options.client ?? client).post<
    MirrorServiceBatchGetMirrorsResponses,
    MirrorServiceBatchGetMirrorsErrors,
    ThrowOnError
  >({
    responseTransformer: mirrorServiceBatchGetMirrorsResponseTransformer,
    responseType: "json",
    url: "/v1beta1/mirrors:batchGet",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * Gets the latest state of a long-running operation.  Clients can use this
 * method to poll the operation result at intervals as recommended by the API
 * service.
 */
export const operationServiceGetOperation = <
  ThrowOnError extends boolean = false,
>(
  options: Options<OperationServiceGetOperationData, ThrowOnError>,
): RequestResult<
  OperationServiceGetOperationResponses,
  OperationServiceGetOperationErrors,
  ThrowOnError
> =>
  (options.client ?? client).get<
    OperationServiceGetOperationResponses,
    OperationServiceGetOperationErrors,
    ThrowOnError
  >({
    responseType: "json",
    url: "/v1beta1/operations/{operation}",
    ...options,
  });

/**
 * ListPublicationTargets lists all PublicationTargets.
 */
export const publicationTargetServiceListPublicationTargets = <
  ThrowOnError extends boolean = false,
>(
  options?: Options<
    PublicationTargetServiceListPublicationTargetsData,
    ThrowOnError
  >,
): RequestResult<
  PublicationTargetServiceListPublicationTargetsResponses,
  PublicationTargetServiceListPublicationTargetsErrors,
  ThrowOnError
> =>
  (options?.client ?? client).get<
    PublicationTargetServiceListPublicationTargetsResponses,
    PublicationTargetServiceListPublicationTargetsErrors,
    ThrowOnError
  >({
    responseType: "json",
    url: "/v1beta1/publicationTargets",
    ...options,
  });

/**
 * CreatePublicationTarget creates a new PublicationTarget.
 */
export const publicationTargetServiceCreatePublicationTarget = <
  ThrowOnError extends boolean = false,
>(
  options: Options<
    PublicationTargetServiceCreatePublicationTargetData,
    ThrowOnError
  >,
): RequestResult<
  PublicationTargetServiceCreatePublicationTargetResponses,
  PublicationTargetServiceCreatePublicationTargetErrors,
  ThrowOnError
> =>
  (options.client ?? client).post<
    PublicationTargetServiceCreatePublicationTargetResponses,
    PublicationTargetServiceCreatePublicationTargetErrors,
    ThrowOnError
  >({
    responseType: "json",
    url: "/v1beta1/publicationTargets",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * DeletePublicationTarget deletes a PublicationTarget.
 */
export const publicationTargetServiceDeletePublicationTarget = <
  ThrowOnError extends boolean = false,
>(
  options: Options<
    PublicationTargetServiceDeletePublicationTargetData,
    ThrowOnError
  >,
): RequestResult<
  PublicationTargetServiceDeletePublicationTargetResponses,
  PublicationTargetServiceDeletePublicationTargetErrors,
  ThrowOnError
> =>
  (options.client ?? client).delete<
    PublicationTargetServiceDeletePublicationTargetResponses,
    PublicationTargetServiceDeletePublicationTargetErrors,
    ThrowOnError
  >({ url: "/v1beta1/publicationTargets/{publicationTarget}", ...options });

/**
 * GetPublicationTarget gets details of a single PublicationTarget.
 */
export const publicationTargetServiceGetPublicationTarget = <
  ThrowOnError extends boolean = false,
>(
  options: Options<
    PublicationTargetServiceGetPublicationTargetData,
    ThrowOnError
  >,
): RequestResult<
  PublicationTargetServiceGetPublicationTargetResponses,
  PublicationTargetServiceGetPublicationTargetErrors,
  ThrowOnError
> =>
  (options.client ?? client).get<
    PublicationTargetServiceGetPublicationTargetResponses,
    PublicationTargetServiceGetPublicationTargetErrors,
    ThrowOnError
  >({
    responseType: "json",
    url: "/v1beta1/publicationTargets/{publicationTarget}",
    ...options,
  });

/**
 * UpdatePublicationTarget replaces (full replacement) an existing PublicationTarget.
 */
export const publicationTargetServiceUpdatePublicationTarget = <
  ThrowOnError extends boolean = false,
>(
  options: Options<
    PublicationTargetServiceUpdatePublicationTargetData,
    ThrowOnError
  >,
): RequestResult<
  PublicationTargetServiceUpdatePublicationTargetResponses,
  PublicationTargetServiceUpdatePublicationTargetErrors,
  ThrowOnError
> =>
  (options.client ?? client).patch<
    PublicationTargetServiceUpdatePublicationTargetResponses,
    PublicationTargetServiceUpdatePublicationTargetErrors,
    ThrowOnError
  >({
    responseType: "json",
    url: "/v1beta1/publicationTargets/{publicationTarget}",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * BatchGetPublicationTargets retrieves a list of publication targets based on their name.
 */
export const publicationTargetServiceBatchGetPublicationTargets = <
  ThrowOnError extends boolean = false,
>(
  options: Options<
    PublicationTargetServiceBatchGetPublicationTargetsData,
    ThrowOnError
  >,
): RequestResult<
  PublicationTargetServiceBatchGetPublicationTargetsResponses,
  PublicationTargetServiceBatchGetPublicationTargetsErrors,
  ThrowOnError
> =>
  (options.client ?? client).post<
    PublicationTargetServiceBatchGetPublicationTargetsResponses,
    PublicationTargetServiceBatchGetPublicationTargetsErrors,
    ThrowOnError
  >({
    responseType: "json",
    url: "/v1beta1/publicationTargets:batchGet",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * ListPublications returns a list of Publications.
 */
export const publicationServiceListPublications = <
  ThrowOnError extends boolean = false,
>(
  options?: Options<PublicationServiceListPublicationsData, ThrowOnError>,
): RequestResult<
  PublicationServiceListPublicationsResponses,
  PublicationServiceListPublicationsErrors,
  ThrowOnError
> =>
  (options?.client ?? client).get<
    PublicationServiceListPublicationsResponses,
    PublicationServiceListPublicationsErrors,
    ThrowOnError
  >({
    responseTransformer: publicationServiceListPublicationsResponseTransformer,
    responseType: "json",
    url: "/v1beta1/publications",
    ...options,
  });

/**
 * CreatePublication creates a new Publication.
 */
export const publicationServiceCreatePublication = <
  ThrowOnError extends boolean = false,
>(
  options: Options<PublicationServiceCreatePublicationData, ThrowOnError>,
): RequestResult<
  PublicationServiceCreatePublicationResponses,
  PublicationServiceCreatePublicationErrors,
  ThrowOnError
> =>
  (options.client ?? client).post<
    PublicationServiceCreatePublicationResponses,
    PublicationServiceCreatePublicationErrors,
    ThrowOnError
  >({
    responseTransformer: publicationServiceCreatePublicationResponseTransformer,
    responseType: "json",
    url: "/v1beta1/publications",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * DeletePublication deletes a Publication.
 * Note: This operation may trigger a background Task to clean up published artifacts.
 */
export const publicationServiceDeletePublication = <
  ThrowOnError extends boolean = false,
>(
  options: Options<PublicationServiceDeletePublicationData, ThrowOnError>,
): RequestResult<
  PublicationServiceDeletePublicationResponses,
  PublicationServiceDeletePublicationErrors,
  ThrowOnError
> =>
  (options.client ?? client).delete<
    PublicationServiceDeletePublicationResponses,
    PublicationServiceDeletePublicationErrors,
    ThrowOnError
  >({ url: "/v1beta1/publications/{publication}", ...options });

/**
 * GetPublication retrieves a Publication by its resource name.
 */
export const publicationServiceGetPublication = <
  ThrowOnError extends boolean = false,
>(
  options: Options<PublicationServiceGetPublicationData, ThrowOnError>,
): RequestResult<
  PublicationServiceGetPublicationResponses,
  PublicationServiceGetPublicationErrors,
  ThrowOnError
> =>
  (options.client ?? client).get<
    PublicationServiceGetPublicationResponses,
    PublicationServiceGetPublicationErrors,
    ThrowOnError
  >({
    responseTransformer: publicationServiceGetPublicationResponseTransformer,
    responseType: "json",
    url: "/v1beta1/publications/{publication}",
    ...options,
  });

/**
 * UpdatePublication updates the details of a Publication.
 */
export const publicationServiceUpdatePublication = <
  ThrowOnError extends boolean = false,
>(
  options: Options<PublicationServiceUpdatePublicationData, ThrowOnError>,
): RequestResult<
  PublicationServiceUpdatePublicationResponses,
  PublicationServiceUpdatePublicationErrors,
  ThrowOnError
> =>
  (options.client ?? client).patch<
    PublicationServiceUpdatePublicationResponses,
    PublicationServiceUpdatePublicationErrors,
    ThrowOnError
  >({
    responseTransformer: publicationServiceUpdatePublicationResponseTransformer,
    responseType: "json",
    url: "/v1beta1/publications/{publication}",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });

/**
 * PublishPublication triggers the actual publication process.
 * This is a custom method that returns a background Task.
 */
export const publicationServicePublishPublication = <
  ThrowOnError extends boolean = false,
>(
  options: Options<PublicationServicePublishPublicationData, ThrowOnError>,
): RequestResult<
  PublicationServicePublishPublicationResponses,
  PublicationServicePublishPublicationErrors,
  ThrowOnError
> =>
  (options.client ?? client).post<
    PublicationServicePublishPublicationResponses,
    PublicationServicePublishPublicationErrors,
    ThrowOnError
  >({
    responseType: "json",
    url: "/v1beta1/publications/{publication}:publish",
    ...options,
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
  });
