// @canonical/landscape-openapi@0.0.70/generated/transformers.gen.ts
// This file is auto-generated by @hey-api/openapi-ts

import type {
  LocalServiceBatchGetLocalsResponse,
  LocalServiceCreateLocalResponse,
  LocalServiceGetLocalResponse,
  LocalServiceListLocalsResponse,
  LocalServiceUpdateLocalResponse,
  MirrorServiceBatchGetMirrorsResponse,
  MirrorServiceCreateMirrorResponse,
  MirrorServiceGetMirrorResponse,
  MirrorServiceListMirrorsResponse,
  MirrorServiceUpdateMirrorResponse,
  PublicationServiceCreatePublicationResponse,
  PublicationServiceGetPublicationResponse,
  PublicationServiceListPublicationsResponse,
  PublicationServiceUpdatePublicationResponse,
} from "./types.gen";

const localSchemaResponseTransformer = (data: any) => {
  if (data.lastImportTime) {
    data.lastImportTime = new Date(data.lastImportTime);
  }
  return data;
};

const listLocalsResponseSchemaResponseTransformer = (data: any) => {
  if (data.locals) {
    data.locals = data.locals.map((item: any) =>
      localSchemaResponseTransformer(item),
    );
  }
  return data;
};

export const localServiceListLocalsResponseTransformer = async (
  data: any,
): Promise<LocalServiceListLocalsResponse> => {
  data = listLocalsResponseSchemaResponseTransformer(data);
  return data;
};

export const localServiceCreateLocalResponseTransformer = async (
  data: any,
): Promise<LocalServiceCreateLocalResponse> => {
  data = localSchemaResponseTransformer(data);
  return data;
};

export const localServiceGetLocalResponseTransformer = async (
  data: any,
): Promise<LocalServiceGetLocalResponse> => {
  data = localSchemaResponseTransformer(data);
  return data;
};

export const localServiceUpdateLocalResponseTransformer = async (
  data: any,
): Promise<LocalServiceUpdateLocalResponse> => {
  data = localSchemaResponseTransformer(data);
  return data;
};

const batchGetLocalsResponseSchemaResponseTransformer = (data: any) => {
  if (data.locals) {
    data.locals = data.locals.map((item: any) =>
      localSchemaResponseTransformer(item),
    );
  }
  return data;
};

export const localServiceBatchGetLocalsResponseTransformer = async (
  data: any,
): Promise<LocalServiceBatchGetLocalsResponse> => {
  data = batchGetLocalsResponseSchemaResponseTransformer(data);
  return data;
};

const mirrorSchemaResponseTransformer = (data: any) => {
  if (data.lastDownloadDate) {
    data.lastDownloadDate = new Date(data.lastDownloadDate);
  }
  return data;
};

const listMirrorsResponseSchemaResponseTransformer = (data: any) => {
  if (data.mirrors) {
    data.mirrors = data.mirrors.map((item: any) =>
      mirrorSchemaResponseTransformer(item),
    );
  }
  return data;
};

export const mirrorServiceListMirrorsResponseTransformer = async (
  data: any,
): Promise<MirrorServiceListMirrorsResponse> => {
  data = listMirrorsResponseSchemaResponseTransformer(data);
  return data;
};

export const mirrorServiceCreateMirrorResponseTransformer = async (
  data: any,
): Promise<MirrorServiceCreateMirrorResponse> => {
  data = mirrorSchemaResponseTransformer(data);
  return data;
};

export const mirrorServiceGetMirrorResponseTransformer = async (
  data: any,
): Promise<MirrorServiceGetMirrorResponse> => {
  data = mirrorSchemaResponseTransformer(data);
  return data;
};

export const mirrorServiceUpdateMirrorResponseTransformer = async (
  data: any,
): Promise<MirrorServiceUpdateMirrorResponse> => {
  data = mirrorSchemaResponseTransformer(data);
  return data;
};

const batchGetMirrorsResponseSchemaResponseTransformer = (data: any) => {
  if (data.mirrors) {
    data.mirrors = data.mirrors.map((item: any) =>
      mirrorSchemaResponseTransformer(item),
    );
  }
  return data;
};

export const mirrorServiceBatchGetMirrorsResponseTransformer = async (
  data: any,
): Promise<MirrorServiceBatchGetMirrorsResponse> => {
  data = batchGetMirrorsResponseSchemaResponseTransformer(data);
  return data;
};

const publicationSchemaResponseTransformer = (data: any) => {
  if (data.publishTime) {
    data.publishTime = new Date(data.publishTime);
  }
  return data;
};

const listPublicationsResponseSchemaResponseTransformer = (data: any) => {
  if (data.publications) {
    data.publications = data.publications.map((item: any) =>
      publicationSchemaResponseTransformer(item),
    );
  }
  return data;
};

export const publicationServiceListPublicationsResponseTransformer = async (
  data: any,
): Promise<PublicationServiceListPublicationsResponse> => {
  data = listPublicationsResponseSchemaResponseTransformer(data);
  return data;
};

export const publicationServiceCreatePublicationResponseTransformer = async (
  data: any,
): Promise<PublicationServiceCreatePublicationResponse> => {
  data = publicationSchemaResponseTransformer(data);
  return data;
};

export const publicationServiceGetPublicationResponseTransformer = async (
  data: any,
): Promise<PublicationServiceGetPublicationResponse> => {
  data = publicationSchemaResponseTransformer(data);
  return data;
};

export const publicationServiceUpdatePublicationResponseTransformer = async (
  data: any,
): Promise<PublicationServiceUpdatePublicationResponse> => {
  data = publicationSchemaResponseTransformer(data);
  return data;
};
