// @canonical/landscape-openapi@0.0.70/generated/types.gen.ts
// This file is auto-generated by @hey-api/openapi-ts

export type ClientOptions = {
  baseURL: `${string}://openapi.yaml` | (string & {});
};

/**
 * BatchGetLocalsRequest is the request message for BatchGetLocals
 */
export type BatchGetLocalsRequest = {
  /**
   * The names of the locals to retrieve.
   * A maximum of 100 locals can be retrieved in a batch.
   * Format: locals/{local}
   */
  names: Array<string>;
};

/**
 * BatchGetLocalsResponse is the response message for BatchGetLocals
 */
export type BatchGetLocalsResponse = {
  /**
   * Locals requested.
   */
  locals?: Array<Local>;
};

/**
 * BatchGetMirrorsRequest is the request message for BatchGetMirrors
 */
export type BatchGetMirrorsRequest = {
  /**
   * The names of the mirrors to retrieve.
   * A maximum of 100 mirrors can be retrieved in a batch.
   * Format: mirrors/{mirror}
   */
  names: Array<string>;
};

/**
 * BatchGetMirrorsResponse is the response message for BatchGetMirrors
 */
export type BatchGetMirrorsResponse = {
  /**
   * Mirrors requested.
   */
  mirrors?: Array<Mirror>;
};

/**
 * BatchGetPublicationTargetsRequest is the request message for BatchGetPublicationTargets
 */
export type BatchGetPublicationTargetsRequest = {
  /**
   * The names of the publication targets to retrieve.
   * A maximum of 100 publication targets can be retrieved in a batch.
   * Format: publicationTargets/{publication_target}
   */
  names: Array<string>;
};

/**
 * BatchGetPublicationTargetsResponse is the response message for BatchGetPublicationTargets
 */
export type BatchGetPublicationTargetsResponse = {
  /**
   * Publication targets requested.
   */
  publicationTargets?: Array<PublicationTarget>;
};

/**
 * FilesystemTarget represents a local filesystem storage destination.
 */
export type FilesystemTarget = {
  /**
   * The absolute path to the directory where files should be published.
   */
  path: string;
  /**
   * The link method to use for published files.
   */
  linkMethod?: "HARDLINK" | "SYMLINK" | "COPY";
};

/**
 * Contains an arbitrary serialized message along with a @type that describes the type of the serialized message.
 */
export type GoogleProtobufAny = {
  /**
   * The type of the serialized message.
   */
  "@type"?: string;
  [key: string]: unknown;
};

/**
 * GpgKey represents a gpg key.
 */
export type GpgKey = {
  /**
   * The GPG key in ASCII armor format.
   */
  armor: string;
  /**
   * Indicates if the key holds a private component.
   */
  readonly isPrivate?: boolean;
  /**
   * The fingerprint of the key.
   */
  readonly fingerprint?: string;
};

/**
 * ImportLocalPackagesRequest is the request message for ImportLocalPackages.
 */
export type ImportLocalPackagesRequest = {
  /**
   * The resource name of the local repository to import packages into.
   * Format: "locals/{local}"
   */
  name: string;
  /**
   * The URL of the package list or index to import from.
   */
  url: string;
  /**
   * If set to true, the request is validated without actually performing the import.
   * The set of packages that would be imported is returned as the result of the task.
   * See: https://google.aip.dev/163
   */
  validateOnly?: boolean;
};

/**
 * ListLocalPackagesResponse is the response message for ListLocalPackages.
 */
export type ListLocalPackagesResponse = {
  /**
   * The list of package strings.
   * Example: ["snap-http_1.4.0_source", ...]
   */
  localPackages?: Array<string>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * ListLocalsResponse is the response message for ListLocals.
 */
export type ListLocalsResponse = {
  /**
   * The list of local repositories.
   */
  locals?: Array<Local>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * ListMirrorPackagesResponse is the response message for ListMirrorPackages.
 */
export type ListMirrorPackagesResponse = {
  /**
   * The list of package strings.
   * Example: ["snap-http_1.4.0_source", ...]
   */
  mirrorPackages?: Array<string>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * ListMirrorsResponse is the response message for ListMirrors.
 */
export type ListMirrorsResponse = {
  /**
   * The list of mirrors.
   */
  mirrors?: Array<Mirror>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * ListPublicationTargetsResponse is the response message for ListPublicationTargets.
 */
export type ListPublicationTargetsResponse = {
  /**
   * The publication targets.
   */
  publicationTargets?: Array<PublicationTarget>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * ListPublicationsResponse is the response message for ListPublications.
 */
export type ListPublicationsResponse = {
  /**
   * A list of publications.
   */
  publications?: Array<Publication>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * Local represents a local repository configuration.
 */
export type Local = {
  /**
   * The resource name of the local repository.
   * Format: "locals/{local}"
   */
  name?: string;
  /**
   * The local repository ID is exposed separately for convenience.
   */
  readonly localId?: string;
  /**
   * The human-readable name used to reference this local repository.
   */
  displayName: string;
  /**
   * A user-facing description of the local repository.
   */
  comment?: string;
  /**
   * The default distribution to publish.
   */
  defaultDistribution?: string;
  /**
   * The default component to publish.
   */
  defaultComponent: string;
  /**
   * The last associated LRO with the local
   */
  readonly lastOperation?: string;
  /**
   * The timestamp of the last successful package import.
   */
  readonly lastImportTime?: Date;
};

/**
 * Mirror represents a repository mirror configuration and status.
 */
export type Mirror = {
  /**
   * The resource name of the mirror.
   * Format: "mirrors/{mirror}"
   */
  name?: string;
  /**
   * The mirror ID is exposed separately for convenience.
   */
  readonly mirrorId?: string;
  /**
   * The status of the mirror.
   */
  readonly status?: string;
  /**
   * The timestamp of the last successful download.
   */
  readonly lastDownloadDate?: Date;
  /**
   * The human-readable name used to reference this mirror.
   */
  displayName: string;
  /**
   * Root of the debian archive (URL).
   */
  archiveRoot: string;
  /**
   * The name of the distribution (subdirectory of `dists`).
   */
  distribution?: string;
  /**
   * List of architectures to fetch.
   */
  architectures?: Array<string>;
  /**
   * List of components to fetch.
   */
  components: Array<string>;
  /**
   * Indicates if installer files should be downloaded.
   */
  downloadInstaller?: boolean;
  /**
   * Indicates if sources files should be downloaded.
   */
  downloadSources?: boolean;
  /**
   * Indicates if `.udebs` files should be downloaded.
   */
  downloadUdebs?: boolean;
  /**
   * Package query applied to the mirror.
   * Follows the [Reprepro/Aptly language](https://www.aptly.info/doc/feature/query/).
   */
  filter?: string;
  /**
   * Indicates if dependencies of matching packages should be included.
   */
  filterWithDeps?: boolean;
  /**
   * Indicates if the architecture list verification should be skipped.
   */
  skipArchitectureCheck?: boolean;
  /**
   * Indicates if the component list verification should be skipped.
   */
  skipComponentCheck?: boolean;
  /**
   * Indicates if the mirror is a signature-preserving mirror.
   */
  preserveSignatures?: boolean;
  /**
   * Optional GPG key to validate the upstream signature.
   */
  gpgKey?: GpgKey;
  /**
   * The last associated LRO with the mirror
   */
  readonly lastOperation?: string;
  /**
   * The type of upstream repository being mirrored.
   */
  mirrorType?:
    | "MIRROR_TYPE_UNSPECIFIED"
    | "UBUNTU_ARCHIVE"
    | "UBUNTU_SNAPSHOTS"
    | "UBUNTU_PRO"
    | "THIRD_PARTY";
};

/**
 * This resource represents a long-running operation that is the result of a
 * network API call.
 */
export type Operation = {
  /**
   * The server-assigned name, which is only unique within the same service that
   * originally returns it. If you use the default HTTP mapping, the
   * `name` should be a resource name ending with `operations/{unique_id}`.
   */
  name?: string;
  /**
   * Service-specific metadata associated with the operation.  It typically
   * contains progress information and common metadata such as create time.
   * Some services might not provide such metadata.  Any method that returns a
   * long-running operation should document the metadata type, if any.
   */
  metadata?: GoogleProtobufAny;
  /**
   * If the value is `false`, it means the operation is still in progress.
   * If `true`, the operation is completed, and either `error` or `response` is
   * available.
   */
  done?: boolean;
  /**
   * The error result of the operation in case of failure or cancellation.
   */
  error?: Status;
  /**
   * The normal, successful response of the operation.  If the original
   * method returns no data on success, such as `Delete`, the response is
   * `google.protobuf.Empty`.  If the original method is standard
   * `Get`/`Create`/`Update`, the response should be the resource.  For other
   * methods, the response should have the type `XxxResponse`, where `Xxx`
   * is the original method name.  For example, if the original method name
   * is `TakeSnapshot()`, the inferred response type is
   * `TakeSnapshotResponse`.
   */
  response?: GoogleProtobufAny;
};

/**
 * Publication represents the configuration for publishing Debian artifacts
 * from a Source (Mirror or Local) to a Target.
 */
export type Publication = {
  /**
   * The resource name of the publication.
   * Format: "publications/{uuid}"
   */
  name?: string;
  /**
   * The human-readable name used to reference this publication.
   */
  displayName: string;
  /**
   * The publication ID (UUID) is exposed separately for convenience.
   */
  readonly publicationId?: string;
  /**
   * The resource name of the publication target.
   * Format: "publicationTargets/{publication_target}"
   */
  publicationTarget: string;
  /**
   * The resource name of the source mirror or local.
   * Format: "mirrors/{mirror}" or "locals/{local}"
   */
  source: string;
  /**
   * The name of the distribution.
   * Inferred from the source if omitted.
   */
  distribution?: string;
  /**
   * Value for the `Label:` field in the Release file.
   */
  label?: string;
  /**
   * Value for the `Origin:` field in the Release file.
   * Inferred from the source if omitted.
   */
  origin?: string;
  /**
   * List of architectures to publish.
   * Defaults to all source architectures if omitted.
   */
  architectures?: Array<string>;
  /**
   * Provides index file by hash if unique.
   */
  acquireByHash?: boolean;
  /**
   * Sets `ButAutomaticUpgrade: yes|no` in the Release file if provided.
   */
  butAutomaticUpgrades?: boolean;
  /**
   * Sets `NotAutomatic: yes|no` in the Release file if provided.
   */
  notAutomatic?: boolean;
  /**
   * Indicates if multiple distributions are supported.
   */
  multiDist?: boolean;
  /**
   * Indicates if bz2 compression should be skipped for index files.
   */
  skipBz2?: boolean;
  /**
   * Indicates if content indexes should not be generated.
   */
  skipContents?: boolean;
  /**
   * Optional GPG key to sign the release file(s).
   */
  gpgKey?: GpgKey;
  /**
   * The timestamp of the last successful publication.
   */
  readonly publishTime?: Date;
  /**
   * The last associated LRO with the publication
   */
  readonly lastOperation?: string;
};

/**
 * PublicationTarget represents a storage destination (currently supports S3 or Swift).
 */
export type PublicationTarget = {
  /**
   * The resource name of the publication target.
   * Format: "publicationTargets/{publication_target}"
   */
  name?: string;
  /**
   * The user-defined display name.
   */
  displayName: string;
  /**
   * The publication_id (uuid) is part of the `name`, and is exposed separately for convenience.
   */
  readonly publicationTargetId?: string;
  /**
   * The S3 storage.
   */
  s3?: S3Target;
  /**
   * The Swift storage.
   */
  swift?: SwiftTarget;
  /**
   * The Filesystem storage.
   */
  filesystem?: FilesystemTarget;
};

/**
 * PublishPublicationRequest is the request message for PublishPublication.
 */
export type PublishPublicationRequest = {
  /**
   * The resource name of the publication to publish.
   * Format: "publications/{publication}"
   */
  name: string;
  /**
   * Force overwrite of existing artifacts.
   */
  forceOverwrite?: boolean;
  /**
   * Force cleanup of old artifacts.
   */
  forceCleanup?: boolean;
};

/**
 * S3Target represents an S3 storage destination.
 */
export type S3Target = {
  /**
   * The AWS region for the S3 bucket.
   */
  region: string;
  /**
   * The name of the S3 bucket.
   */
  bucket: string;
  /**
   * Hostname used for S3 compatible storage.
   * If set, the region is ignored.
   */
  endpoint?: string;
  /**
   * Prefix used in the bucket.
   * Defaults to the bucket root if not specified.
   */
  prefix?: string;
  /**
   * ACL (Access Control List) used for published files.
   */
  acl?: string;
  /**
   * The AWS S3 storage class to use.
   */
  storageClass?: string;
  /**
   * Server-side encryption method. Defaults to none.
   * Currently, the only available encryption method is `AES256`.
   */
  encryptionMethod?: string;
  /**
   * Workaround for specific S3-compatible issues (e.g. "+" in filenames).
   */
  plusWorkaround?: boolean;
  /**
   * Disables the `MultiDel` S3 API.
   * Useful for S3-compatible cloud storage providers that do not support it.
   */
  disableMultiDel?: boolean;
  /**
   * Disables Signature V4 support.
   * Useful for S3-compatible cloud storage providers that do not support SigV4.
   */
  forceSigV2?: boolean;
  /**
   * Enables debug logging for S3 operations.
   */
  debug?: boolean;
};

/**
 * The `Status` type defines a logical error model that is suitable for different programming environments, including REST APIs and RPC APIs. It is used by [gRPC](https://github.com/grpc). Each `Status` message contains three pieces of data: error code, error message, and error details. You can find out more about this error model and how to work with it in the [API Design Guide](https://cloud.google.com/apis/design/errors).
 */
export type Status = {
  /**
   * The status code, which should be an enum value of [google.rpc.Code][google.rpc.Code].
   */
  code?: number;
  /**
   * A developer-facing error message, which should be in English. Any user-facing error message should be localized and sent in the [google.rpc.Status.details][google.rpc.Status.details] field, or localized by the client.
   */
  message?: string;
  /**
   * A list of messages that carry the error details.  There is a common set of message types for APIs to use.
   */
  details?: Array<GoogleProtobufAny>;
};

/**
 * SwiftTarget represents a Swift (OpenStack) storage destination.
 */
export type SwiftTarget = {
  /**
   * The name of the container.
   */
  container: string;
  /**
   * Prefix used in the container.
   * Defaults to the container root if not specified.
   */
  prefix?: string;
  /**
   * The full URL of the Keystone server (including port and version).
   */
  authUrl: string;
  /**
   * The OpenStack tenant name.
   */
  tenant?: string;
  /**
   * The OpenStack tenant ID.
   */
  tenantId?: string;
  /**
   * The OpenStack domain name.
   */
  domain?: string;
  /**
   * The OpenStack domain ID.
   */
  domainId?: string;
  /**
   * The OpenStack domain name the tenant/project belongs to.
   */
  tenantDomain?: string;
  /**
   * The OpenStack domain ID the tenant/project belongs to.
   */
  tenantDomainId?: string;
};

/**
 * SyncMirrorRequest is the request message for SyncMirror.
 */
export type SyncMirrorRequest = {
  /**
   * The resource name of the mirror to sync.
   * Format: "mirrors/{mirror}"
   */
  name: string;
  /**
   * Optional keyring files to use for validation.
   */
  keyrings?: Array<string>;
  /**
   * Ignore checksum mismatches.
   */
  ignoreChecksums?: boolean;
  /**
   * Ignore signature verification failures.
   */
  ignoreSignatures?: boolean;
  /**
   * Force a full update.
   */
  forceUpdate?: boolean;
  /**
   * Skip downloading packages that already exist.
   */
  skipExistingPackages?: boolean;
};

/**
 * BatchGetLocalsResponse is the response message for BatchGetLocals
 */
export type BatchGetLocalsResponseWritable = {
  /**
   * Locals requested.
   */
  locals?: Array<LocalWritable>;
};

/**
 * BatchGetMirrorsResponse is the response message for BatchGetMirrors
 */
export type BatchGetMirrorsResponseWritable = {
  /**
   * Mirrors requested.
   */
  mirrors?: Array<MirrorWritable>;
};

/**
 * BatchGetPublicationTargetsResponse is the response message for BatchGetPublicationTargets
 */
export type BatchGetPublicationTargetsResponseWritable = {
  /**
   * Publication targets requested.
   */
  publicationTargets?: Array<PublicationTargetWritable>;
};

/**
 * ListLocalsResponse is the response message for ListLocals.
 */
export type ListLocalsResponseWritable = {
  /**
   * The list of local repositories.
   */
  locals?: Array<LocalWritable>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * ListMirrorsResponse is the response message for ListMirrors.
 */
export type ListMirrorsResponseWritable = {
  /**
   * The list of mirrors.
   */
  mirrors?: Array<MirrorWritable>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * ListPublicationTargetsResponse is the response message for ListPublicationTargets.
 */
export type ListPublicationTargetsResponseWritable = {
  /**
   * The publication targets.
   */
  publicationTargets?: Array<PublicationTargetWritable>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * ListPublicationsResponse is the response message for ListPublications.
 */
export type ListPublicationsResponseWritable = {
  /**
   * A list of publications.
   */
  publications?: Array<PublicationWritable>;
  /**
   * A token, which can be sent as `page_token` to retrieve the next page.
   * If this field is omitted, there are no subsequent pages.
   */
  nextPageToken?: string;
};

/**
 * Local represents a local repository configuration.
 */
export type LocalWritable = {
  /**
   * The resource name of the local repository.
   * Format: "locals/{local}"
   */
  name?: string;
  /**
   * The human-readable name used to reference this local repository.
   */
  displayName: string;
  /**
   * A user-facing description of the local repository.
   */
  comment?: string;
  /**
   * The default distribution to publish.
   */
  defaultDistribution?: string;
  /**
   * The default component to publish.
   */
  defaultComponent: string;
};

/**
 * Mirror represents a repository mirror configuration and status.
 */
export type MirrorWritable = {
  /**
   * The resource name of the mirror.
   * Format: "mirrors/{mirror}"
   */
  name?: string;
  /**
   * The human-readable name used to reference this mirror.
   */
  displayName: string;
  /**
   * Root of the debian archive (URL).
   */
  archiveRoot: string;
  /**
   * The name of the distribution (subdirectory of `dists`).
   */
  distribution?: string;
  /**
   * List of architectures to fetch.
   */
  architectures?: Array<string>;
  /**
   * List of components to fetch.
   */
  components: Array<string>;
  /**
   * Indicates if installer files should be downloaded.
   */
  downloadInstaller?: boolean;
  /**
   * Indicates if sources files should be downloaded.
   */
  downloadSources?: boolean;
  /**
   * Indicates if `.udebs` files should be downloaded.
   */
  downloadUdebs?: boolean;
  /**
   * Package query applied to the mirror.
   * Follows the [Reprepro/Aptly language](https://www.aptly.info/doc/feature/query/).
   */
  filter?: string;
  /**
   * Indicates if dependencies of matching packages should be included.
   */
  filterWithDeps?: boolean;
  /**
   * Indicates if the architecture list verification should be skipped.
   */
  skipArchitectureCheck?: boolean;
  /**
   * Indicates if the component list verification should be skipped.
   */
  skipComponentCheck?: boolean;
  /**
   * Indicates if the mirror is a signature-preserving mirror.
   */
  preserveSignatures?: boolean;
  /**
   * Optional GPG key to validate the upstream signature.
   */
  gpgKey?: GpgKey;
  /**
   * The type of upstream repository being mirrored.
   */
  mirrorType?:
    | "MIRROR_TYPE_UNSPECIFIED"
    | "UBUNTU_ARCHIVE"
    | "UBUNTU_SNAPSHOTS"
    | "UBUNTU_PRO"
    | "THIRD_PARTY";
};

/**
 * Publication represents the configuration for publishing Debian artifacts
 * from a Source (Mirror or Local) to a Target.
 */
export type PublicationWritable = {
  /**
   * The resource name of the publication.
   * Format: "publications/{uuid}"
   */
  name?: string;
  /**
   * The human-readable name used to reference this publication.
   */
  displayName: string;
  /**
   * The resource name of the publication target.
   * Format: "publicationTargets/{publication_target}"
   */
  publicationTarget: string;
  /**
   * The resource name of the source mirror or local.
   * Format: "mirrors/{mirror}" or "locals/{local}"
   */
  source: string;
  /**
   * The name of the distribution.
   * Inferred from the source if omitted.
   */
  distribution?: string;
  /**
   * Value for the `Label:` field in the Release file.
   */
  label?: string;
  /**
   * Value for the `Origin:` field in the Release file.
   * Inferred from the source if omitted.
   */
  origin?: string;
  /**
   * List of architectures to publish.
   * Defaults to all source architectures if omitted.
   */
  architectures?: Array<string>;
  /**
   * Provides index file by hash if unique.
   */
  acquireByHash?: boolean;
  /**
   * Sets `ButAutomaticUpgrade: yes|no` in the Release file if provided.
   */
  butAutomaticUpgrades?: boolean;
  /**
   * Sets `NotAutomatic: yes|no` in the Release file if provided.
   */
  notAutomatic?: boolean;
  /**
   * Indicates if multiple distributions are supported.
   */
  multiDist?: boolean;
  /**
   * Indicates if bz2 compression should be skipped for index files.
   */
  skipBz2?: boolean;
  /**
   * Indicates if content indexes should not be generated.
   */
  skipContents?: boolean;
  /**
   * Optional GPG key to sign the release file(s).
   */
  gpgKey?: GpgKey;
};

/**
 * PublicationTarget represents a storage destination (currently supports S3 or Swift).
 */
export type PublicationTargetWritable = {
  /**
   * The resource name of the publication target.
   * Format: "publicationTargets/{publication_target}"
   */
  name?: string;
  /**
   * The user-defined display name.
   */
  displayName: string;
  /**
   * The S3 storage.
   */
  s3?: S3TargetWritable;
  /**
   * The Swift storage.
   */
  swift?: SwiftTargetWritable;
  /**
   * The Filesystem storage.
   */
  filesystem?: FilesystemTarget;
};

/**
 * S3Target represents an S3 storage destination.
 */
export type S3TargetWritable = {
  /**
   * The AWS region for the S3 bucket.
   */
  region: string;
  /**
   * The name of the S3 bucket.
   */
  bucket: string;
  /**
   * Hostname used for S3 compatible storage.
   * If set, the region is ignored.
   */
  endpoint?: string;
  /**
   * AWS credentials (Access Key ID) to access the S3 bucket.
   */
  awsAccessKeyId: string;
  /**
   * AWS credentials (Secret Access Key) to access the S3 bucket.
   */
  awsSecretAccessKey: string;
  /**
   * Prefix used in the bucket.
   * Defaults to the bucket root if not specified.
   */
  prefix?: string;
  /**
   * ACL (Access Control List) used for published files.
   */
  acl?: string;
  /**
   * The AWS S3 storage class to use.
   */
  storageClass?: string;
  /**
   * Server-side encryption method. Defaults to none.
   * Currently, the only available encryption method is `AES256`.
   */
  encryptionMethod?: string;
  /**
   * Workaround for specific S3-compatible issues (e.g. "+" in filenames).
   */
  plusWorkaround?: boolean;
  /**
   * Disables the `MultiDel` S3 API.
   * Useful for S3-compatible cloud storage providers that do not support it.
   */
  disableMultiDel?: boolean;
  /**
   * Disables Signature V4 support.
   * Useful for S3-compatible cloud storage providers that do not support SigV4.
   */
  forceSigV2?: boolean;
  /**
   * Enables debug logging for S3 operations.
   */
  debug?: boolean;
};

/**
 * SwiftTarget represents a Swift (OpenStack) storage destination.
 */
export type SwiftTargetWritable = {
  /**
   * The name of the container.
   */
  container: string;
  /**
   * OpenStack credentials (Username) to access Keystone.
   */
  username: string;
  /**
   * OpenStack credentials (Password) to access Keystone.
   */
  password: string;
  /**
   * Prefix used in the container.
   * Defaults to the container root if not specified.
   */
  prefix?: string;
  /**
   * The full URL of the Keystone server (including port and version).
   */
  authUrl: string;
  /**
   * The OpenStack tenant name.
   */
  tenant?: string;
  /**
   * The OpenStack tenant ID.
   */
  tenantId?: string;
  /**
   * The OpenStack domain name.
   */
  domain?: string;
  /**
   * The OpenStack domain ID.
   */
  domainId?: string;
  /**
   * The OpenStack domain name the tenant/project belongs to.
   */
  tenantDomain?: string;
  /**
   * The OpenStack domain ID the tenant/project belongs to.
   */
  tenantDomainId?: string;
};

export type LocalServiceListLocalsData = {
  body?: never;
  path?: never;
  query?: {
    /**
     * The maximum number of local repositories to return.
     * The service may return fewer than this value.
     * If unspecified, at most 100 local repositories will be returned.
     * The maximum value is 1000; values above 1000 will be coerced to 1000.
     */
    pageSize?: number;
    /**
     * A page token, received from a previous `ListLocals` call.
     * Provide this to retrieve the subsequent page.
     *
     * When paginating, all other parameters provided to `ListLocals` must match
     * the call that provided the page token.
     */
    pageToken?: string;
    /**
     * An expression for filtering the results of the request.
     * The fields eligible for filtering are:
     *
     * * `display_name`
     *
     * Some examples of using filters are:
     *
     * * `display_name="local"`  --> The local has the display name "local".
     * * `display_name="local*"` --> The local has the prefix "local" in its display name.
     */
    filter?: string;
  };
  url: "/v1beta1/locals";
};

export type LocalServiceListLocalsErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type LocalServiceListLocalsError =
  LocalServiceListLocalsErrors[keyof LocalServiceListLocalsErrors];

export type LocalServiceListLocalsResponses = {
  /**
   * OK
   */
  200: ListLocalsResponse;
};

export type LocalServiceListLocalsResponse =
  LocalServiceListLocalsResponses[keyof LocalServiceListLocalsResponses];

export type LocalServiceCreateLocalData = {
  body: LocalWritable;
  path?: never;
  query?: {
    /**
     * The ID to use for the local repository (optional).
     * If provided, the full resource name will be `locals/{local_id}`.
     */
    localId?: string;
  };
  url: "/v1beta1/locals";
};

export type LocalServiceCreateLocalErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type LocalServiceCreateLocalError =
  LocalServiceCreateLocalErrors[keyof LocalServiceCreateLocalErrors];

export type LocalServiceCreateLocalResponses = {
  /**
   * OK
   */
  200: Local;
};

export type LocalServiceCreateLocalResponse =
  LocalServiceCreateLocalResponses[keyof LocalServiceCreateLocalResponses];

export type LocalServiceDeleteLocalData = {
  body?: never;
  path: {
    /**
     * The local id.
     */
    local: string;
  };
  query?: never;
  url: "/v1beta1/locals/{local}";
};

export type LocalServiceDeleteLocalErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type LocalServiceDeleteLocalError =
  LocalServiceDeleteLocalErrors[keyof LocalServiceDeleteLocalErrors];

export type LocalServiceDeleteLocalResponses = {
  /**
   * OK
   */
  200: unknown;
};

export type LocalServiceGetLocalData = {
  body?: never;
  path: {
    /**
     * The local id.
     */
    local: string;
  };
  query?: never;
  url: "/v1beta1/locals/{local}";
};

export type LocalServiceGetLocalErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type LocalServiceGetLocalError =
  LocalServiceGetLocalErrors[keyof LocalServiceGetLocalErrors];

export type LocalServiceGetLocalResponses = {
  /**
   * OK
   */
  200: Local;
};

export type LocalServiceGetLocalResponse =
  LocalServiceGetLocalResponses[keyof LocalServiceGetLocalResponses];

export type LocalServiceUpdateLocalData = {
  body: LocalWritable;
  path: {
    /**
     * The local id.
     */
    local: string;
  };
  query?: {
    /**
     * The list of fields to update.
     */
    updateMask?: string;
  };
  url: "/v1beta1/locals/{local}";
};

export type LocalServiceUpdateLocalErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type LocalServiceUpdateLocalError =
  LocalServiceUpdateLocalErrors[keyof LocalServiceUpdateLocalErrors];

export type LocalServiceUpdateLocalResponses = {
  /**
   * OK
   */
  200: Local;
};

export type LocalServiceUpdateLocalResponse =
  LocalServiceUpdateLocalResponses[keyof LocalServiceUpdateLocalResponses];

export type LocalServiceListLocalPackagesData = {
  body?: never;
  path: {
    /**
     * The local id.
     */
    local: string;
  };
  query?: {
    /**
     * The maximum number of packages to return.
     * The service may return fewer than this value.
     * If unspecified, at most 100 packages will be returned.
     * The maximum value is 1000; values above 1000 will be coerced to 1000.
     */
    pageSize?: number;
    /**
     * A page token, received from a previous `ListLocalPackages` call.
     * Provide this to retrieve the subsequent page.
     *
     * When paginating, all other parameters provided to `ListLocalPackages` must match
     * the call that provided the page token.
     */
    pageToken?: string;
  };
  url: "/v1beta1/locals/{local}/packages";
};

export type LocalServiceListLocalPackagesErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type LocalServiceListLocalPackagesError =
  LocalServiceListLocalPackagesErrors[keyof LocalServiceListLocalPackagesErrors];

export type LocalServiceListLocalPackagesResponses = {
  /**
   * OK
   */
  200: ListLocalPackagesResponse;
};

export type LocalServiceListLocalPackagesResponse =
  LocalServiceListLocalPackagesResponses[keyof LocalServiceListLocalPackagesResponses];

export type LocalServiceImportLocalPackagesData = {
  body: ImportLocalPackagesRequest;
  path: {
    /**
     * The local id.
     */
    local: string;
  };
  query?: never;
  url: "/v1beta1/locals/{local}:importPackages";
};

export type LocalServiceImportLocalPackagesErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type LocalServiceImportLocalPackagesError =
  LocalServiceImportLocalPackagesErrors[keyof LocalServiceImportLocalPackagesErrors];

export type LocalServiceImportLocalPackagesResponses = {
  /**
   * OK
   */
  200: Operation;
};

export type LocalServiceImportLocalPackagesResponse =
  LocalServiceImportLocalPackagesResponses[keyof LocalServiceImportLocalPackagesResponses];

export type LocalServiceBatchGetLocalsData = {
  body: BatchGetLocalsRequest;
  path?: never;
  query?: never;
  url: "/v1beta1/locals:batchGet";
};

export type LocalServiceBatchGetLocalsErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type LocalServiceBatchGetLocalsError =
  LocalServiceBatchGetLocalsErrors[keyof LocalServiceBatchGetLocalsErrors];

export type LocalServiceBatchGetLocalsResponses = {
  /**
   * OK
   */
  200: BatchGetLocalsResponse;
};

export type LocalServiceBatchGetLocalsResponse =
  LocalServiceBatchGetLocalsResponses[keyof LocalServiceBatchGetLocalsResponses];

export type MirrorServiceListMirrorsData = {
  body?: never;
  path?: never;
  query?: {
    /**
     * The maximum number of mirrors to return.
     * The service may return fewer than this value.
     * If unspecified, at most 100 mirrors will be returned.
     * The maximum value is 1000; values above 1000 will be coerced to 1000.
     */
    pageSize?: number;
    /**
     * A page token, received from a previous `ListMirrors` call.
     * Provide this to retrieve the subsequent page.
     *
     * When paginating, all other parameters provided to `ListMirrors` must match
     * the call that provided the page token.
     */
    pageToken?: string;
    /**
     * An expression for filtering the results of the request.
     * The fields eligible for filtering are:
     *
     * * `display_name`
     *
     * Some examples of using filters are:
     *
     * * `display_name="mirror"`  --> The mirror has the display name "mirror".
     * * `display_name="mirror*"` --> The mirror has the prefix "mirror" in its display name.
     */
    filter?: string;
  };
  url: "/v1beta1/mirrors";
};

export type MirrorServiceListMirrorsErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type MirrorServiceListMirrorsError =
  MirrorServiceListMirrorsErrors[keyof MirrorServiceListMirrorsErrors];

export type MirrorServiceListMirrorsResponses = {
  /**
   * OK
   */
  200: ListMirrorsResponse;
};

export type MirrorServiceListMirrorsResponse =
  MirrorServiceListMirrorsResponses[keyof MirrorServiceListMirrorsResponses];

export type MirrorServiceCreateMirrorData = {
  body: MirrorWritable;
  path?: never;
  query?: {
    /**
     * The ID to use for the mirror (optional).
     * If provided, the full resource name will be `mirrors/{mirror_id}`.
     */
    mirrorId?: string;
  };
  url: "/v1beta1/mirrors";
};

export type MirrorServiceCreateMirrorErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type MirrorServiceCreateMirrorError =
  MirrorServiceCreateMirrorErrors[keyof MirrorServiceCreateMirrorErrors];

export type MirrorServiceCreateMirrorResponses = {
  /**
   * OK
   */
  200: Mirror;
};

export type MirrorServiceCreateMirrorResponse =
  MirrorServiceCreateMirrorResponses[keyof MirrorServiceCreateMirrorResponses];

export type MirrorServiceDeleteMirrorData = {
  body?: never;
  path: {
    /**
     * The mirror id.
     */
    mirror: string;
  };
  query?: {
    /**
     * Flag for if you want to force delete the mirror
     */
    force?: boolean;
  };
  url: "/v1beta1/mirrors/{mirror}";
};

export type MirrorServiceDeleteMirrorErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type MirrorServiceDeleteMirrorError =
  MirrorServiceDeleteMirrorErrors[keyof MirrorServiceDeleteMirrorErrors];

export type MirrorServiceDeleteMirrorResponses = {
  /**
   * OK
   */
  200: unknown;
};

export type MirrorServiceGetMirrorData = {
  body?: never;
  path: {
    /**
     * The mirror id.
     */
    mirror: string;
  };
  query?: never;
  url: "/v1beta1/mirrors/{mirror}";
};

export type MirrorServiceGetMirrorErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type MirrorServiceGetMirrorError =
  MirrorServiceGetMirrorErrors[keyof MirrorServiceGetMirrorErrors];

export type MirrorServiceGetMirrorResponses = {
  /**
   * OK
   */
  200: Mirror;
};

export type MirrorServiceGetMirrorResponse =
  MirrorServiceGetMirrorResponses[keyof MirrorServiceGetMirrorResponses];

export type MirrorServiceUpdateMirrorData = {
  body: MirrorWritable;
  path: {
    /**
     * The mirror id.
     */
    mirror: string;
  };
  query?: {
    /**
     * The list of fields to update.
     */
    updateMask?: string;
  };
  url: "/v1beta1/mirrors/{mirror}";
};

export type MirrorServiceUpdateMirrorErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type MirrorServiceUpdateMirrorError =
  MirrorServiceUpdateMirrorErrors[keyof MirrorServiceUpdateMirrorErrors];

export type MirrorServiceUpdateMirrorResponses = {
  /**
   * OK
   */
  200: Mirror;
};

export type MirrorServiceUpdateMirrorResponse =
  MirrorServiceUpdateMirrorResponses[keyof MirrorServiceUpdateMirrorResponses];

export type MirrorServiceListMirrorPackagesData = {
  body?: never;
  path: {
    /**
     * The mirror id.
     */
    mirror: string;
  };
  query?: {
    /**
     * The maximum number of packages to return.
     * The service may return fewer than this value.
     * If unspecified, at most 100 packages will be returned.
     * The maximum value is 1000; values above 1000 will be coerced to 1000.
     */
    pageSize?: number;
    /**
     * A page token, received from a previous `ListMirrorPackages` call.
     * Provide this to retrieve the subsequent page.
     *
     * When paginating, all other parameters provided to `ListMirrorPackages` must match
     * the call that provided the page token.
     */
    pageToken?: string;
  };
  url: "/v1beta1/mirrors/{mirror}/packages";
};

export type MirrorServiceListMirrorPackagesErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type MirrorServiceListMirrorPackagesError =
  MirrorServiceListMirrorPackagesErrors[keyof MirrorServiceListMirrorPackagesErrors];

export type MirrorServiceListMirrorPackagesResponses = {
  /**
   * OK
   */
  200: ListMirrorPackagesResponse;
};

export type MirrorServiceListMirrorPackagesResponse =
  MirrorServiceListMirrorPackagesResponses[keyof MirrorServiceListMirrorPackagesResponses];

export type MirrorServiceSyncMirrorData = {
  body: SyncMirrorRequest;
  path: {
    /**
     * The mirror id.
     */
    mirror: string;
  };
  query?: never;
  url: "/v1beta1/mirrors/{mirror}:sync";
};

export type MirrorServiceSyncMirrorErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type MirrorServiceSyncMirrorError =
  MirrorServiceSyncMirrorErrors[keyof MirrorServiceSyncMirrorErrors];

export type MirrorServiceSyncMirrorResponses = {
  /**
   * OK
   */
  200: Operation;
};

export type MirrorServiceSyncMirrorResponse =
  MirrorServiceSyncMirrorResponses[keyof MirrorServiceSyncMirrorResponses];

export type MirrorServiceBatchGetMirrorsData = {
  body: BatchGetMirrorsRequest;
  path?: never;
  query?: never;
  url: "/v1beta1/mirrors:batchGet";
};

export type MirrorServiceBatchGetMirrorsErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type MirrorServiceBatchGetMirrorsError =
  MirrorServiceBatchGetMirrorsErrors[keyof MirrorServiceBatchGetMirrorsErrors];

export type MirrorServiceBatchGetMirrorsResponses = {
  /**
   * OK
   */
  200: BatchGetMirrorsResponse;
};

export type MirrorServiceBatchGetMirrorsResponse =
  MirrorServiceBatchGetMirrorsResponses[keyof MirrorServiceBatchGetMirrorsResponses];

export type OperationServiceGetOperationData = {
  body?: never;
  path: {
    /**
     * The operation id.
     */
    operation: string;
  };
  query?: never;
  url: "/v1beta1/operations/{operation}";
};

export type OperationServiceGetOperationErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type OperationServiceGetOperationError =
  OperationServiceGetOperationErrors[keyof OperationServiceGetOperationErrors];

export type OperationServiceGetOperationResponses = {
  /**
   * OK
   */
  200: Operation;
};

export type OperationServiceGetOperationResponse =
  OperationServiceGetOperationResponses[keyof OperationServiceGetOperationResponses];

export type PublicationTargetServiceListPublicationTargetsData = {
  body?: never;
  path?: never;
  query?: {
    /**
     * The maximum number of publication targets to return.
     * The service may return fewer than this value.
     * If unspecified, at most 100 publication targets will be returned.
     * The maximum value is 1000; values above 1000 will be coerced to 1000.
     */
    pageSize?: number;
    /**
     * A page token, received from a previous `ListPublicationTargets` call.
     * Provide this to retrieve the subsequent page.
     *
     * When paginating, all other parameters provided to `ListPublicationTargets` must match
     * the call that provided the page token.
     */
    pageToken?: string;
    /**
     * An expression for filtering the results of the request.
     * The fields eligible for filtering are:
     *
     * * `display_name`
     *
     * Some examples of using filters are:
     *
     * * `display_name="target"`  --> The publication target has the display name "target".
     * * `display_name="target*"` --> The publication target has the prefix "target" in its display name.
     */
    filter?: string;
  };
  url: "/v1beta1/publicationTargets";
};

export type PublicationTargetServiceListPublicationTargetsErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type PublicationTargetServiceListPublicationTargetsError =
  PublicationTargetServiceListPublicationTargetsErrors[keyof PublicationTargetServiceListPublicationTargetsErrors];

export type PublicationTargetServiceListPublicationTargetsResponses = {
  /**
   * OK
   */
  200: ListPublicationTargetsResponse;
};

export type PublicationTargetServiceListPublicationTargetsResponse =
  PublicationTargetServiceListPublicationTargetsResponses[keyof PublicationTargetServiceListPublicationTargetsResponses];

export type PublicationTargetServiceCreatePublicationTargetData = {
  body: PublicationTargetWritable;
  path?: never;
  query?: {
    /**
     * The ID to use for the publication target (optional).
     * If not provided, the server will generate one.
     */
    publicationTargetId?: string;
  };
  url: "/v1beta1/publicationTargets";
};

export type PublicationTargetServiceCreatePublicationTargetErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type PublicationTargetServiceCreatePublicationTargetError =
  PublicationTargetServiceCreatePublicationTargetErrors[keyof PublicationTargetServiceCreatePublicationTargetErrors];

export type PublicationTargetServiceCreatePublicationTargetResponses = {
  /**
   * OK
   */
  200: PublicationTarget;
};

export type PublicationTargetServiceCreatePublicationTargetResponse =
  PublicationTargetServiceCreatePublicationTargetResponses[keyof PublicationTargetServiceCreatePublicationTargetResponses];

export type PublicationTargetServiceDeletePublicationTargetData = {
  body?: never;
  path: {
    /**
     * The publicationTarget id.
     */
    publicationTarget: string;
  };
  query?: never;
  url: "/v1beta1/publicationTargets/{publicationTarget}";
};

export type PublicationTargetServiceDeletePublicationTargetErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type PublicationTargetServiceDeletePublicationTargetError =
  PublicationTargetServiceDeletePublicationTargetErrors[keyof PublicationTargetServiceDeletePublicationTargetErrors];

export type PublicationTargetServiceDeletePublicationTargetResponses = {
  /**
   * OK
   */
  200: unknown;
};

export type PublicationTargetServiceGetPublicationTargetData = {
  body?: never;
  path: {
    /**
     * The publicationTarget id.
     */
    publicationTarget: string;
  };
  query?: never;
  url: "/v1beta1/publicationTargets/{publicationTarget}";
};

export type PublicationTargetServiceGetPublicationTargetErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type PublicationTargetServiceGetPublicationTargetError =
  PublicationTargetServiceGetPublicationTargetErrors[keyof PublicationTargetServiceGetPublicationTargetErrors];

export type PublicationTargetServiceGetPublicationTargetResponses = {
  /**
   * OK
   */
  200: PublicationTarget;
};

export type PublicationTargetServiceGetPublicationTargetResponse =
  PublicationTargetServiceGetPublicationTargetResponses[keyof PublicationTargetServiceGetPublicationTargetResponses];

export type PublicationTargetServiceUpdatePublicationTargetData = {
  body: PublicationTargetWritable;
  path: {
    /**
     * The publicationTarget id.
     */
    publicationTarget: string;
  };
  query?: {
    /**
     * The list of fields to update.
     */
    updateMask?: string;
  };
  url: "/v1beta1/publicationTargets/{publicationTarget}";
};

export type PublicationTargetServiceUpdatePublicationTargetErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type PublicationTargetServiceUpdatePublicationTargetError =
  PublicationTargetServiceUpdatePublicationTargetErrors[keyof PublicationTargetServiceUpdatePublicationTargetErrors];

export type PublicationTargetServiceUpdatePublicationTargetResponses = {
  /**
   * OK
   */
  200: PublicationTarget;
};

export type PublicationTargetServiceUpdatePublicationTargetResponse =
  PublicationTargetServiceUpdatePublicationTargetResponses[keyof PublicationTargetServiceUpdatePublicationTargetResponses];

export type PublicationTargetServiceBatchGetPublicationTargetsData = {
  body: BatchGetPublicationTargetsRequest;
  path?: never;
  query?: never;
  url: "/v1beta1/publicationTargets:batchGet";
};

export type PublicationTargetServiceBatchGetPublicationTargetsErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type PublicationTargetServiceBatchGetPublicationTargetsError =
  PublicationTargetServiceBatchGetPublicationTargetsErrors[keyof PublicationTargetServiceBatchGetPublicationTargetsErrors];

export type PublicationTargetServiceBatchGetPublicationTargetsResponses = {
  /**
   * OK
   */
  200: BatchGetPublicationTargetsResponse;
};

export type PublicationTargetServiceBatchGetPublicationTargetsResponse =
  PublicationTargetServiceBatchGetPublicationTargetsResponses[keyof PublicationTargetServiceBatchGetPublicationTargetsResponses];

export type PublicationServiceListPublicationsData = {
  body?: never;
  path?: never;
  query?: {
    /**
     * The maximum number of publications to return.
     * The service may return fewer than this value.
     * If unspecified, at most 100 publications will be returned.
     * The maximum value is 1000; values above 1000 will be coerced to 1000.
     */
    pageSize?: number;
    /**
     * A page token, received from a previous `ListPublication` call.
     * Provide this to retrieve the subsequent page.
     *
     * When paginating, all other parameters provided to `ListPublication` must match
     * the call that provided the page token.
     */
    pageToken?: string;
    /**
     * An expression for filtering the results of the request.
     * The fields eligible for filtering are:
     *
     * * `source`
     * * `publication_target`
     *
     * Some examples of using filters are:
     *
     * * `source="locals/123"`  --> The publication has source "locals/123".
     * * `publication_target="publicationTargets/456"` --> The publication has target "publicationTargets/456".
     */
    filter?: string;
  };
  url: "/v1beta1/publications";
};

export type PublicationServiceListPublicationsErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type PublicationServiceListPublicationsError =
  PublicationServiceListPublicationsErrors[keyof PublicationServiceListPublicationsErrors];

export type PublicationServiceListPublicationsResponses = {
  /**
   * OK
   */
  200: ListPublicationsResponse;
};

export type PublicationServiceListPublicationsResponse =
  PublicationServiceListPublicationsResponses[keyof PublicationServiceListPublicationsResponses];

export type PublicationServiceCreatePublicationData = {
  body: PublicationWritable;
  path?: never;
  query?: {
    /**
     * The ID to use for the publication (optional).
     */
    publicationId?: string;
  };
  url: "/v1beta1/publications";
};

export type PublicationServiceCreatePublicationErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type PublicationServiceCreatePublicationError =
  PublicationServiceCreatePublicationErrors[keyof PublicationServiceCreatePublicationErrors];

export type PublicationServiceCreatePublicationResponses = {
  /**
   * OK
   */
  200: Publication;
};

export type PublicationServiceCreatePublicationResponse =
  PublicationServiceCreatePublicationResponses[keyof PublicationServiceCreatePublicationResponses];

export type PublicationServiceDeletePublicationData = {
  body?: never;
  path: {
    /**
     * The publication id.
     */
    publication: string;
  };
  query?: never;
  url: "/v1beta1/publications/{publication}";
};

export type PublicationServiceDeletePublicationErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type PublicationServiceDeletePublicationError =
  PublicationServiceDeletePublicationErrors[keyof PublicationServiceDeletePublicationErrors];

export type PublicationServiceDeletePublicationResponses = {
  /**
   * OK
   */
  200: unknown;
};

export type PublicationServiceGetPublicationData = {
  body?: never;
  path: {
    /**
     * The publication id.
     */
    publication: string;
  };
  query?: never;
  url: "/v1beta1/publications/{publication}";
};

export type PublicationServiceGetPublicationErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type PublicationServiceGetPublicationError =
  PublicationServiceGetPublicationErrors[keyof PublicationServiceGetPublicationErrors];

export type PublicationServiceGetPublicationResponses = {
  /**
   * OK
   */
  200: Publication;
};

export type PublicationServiceGetPublicationResponse =
  PublicationServiceGetPublicationResponses[keyof PublicationServiceGetPublicationResponses];

export type PublicationServiceUpdatePublicationData = {
  body: PublicationWritable;
  path: {
    /**
     * The publication id.
     */
    publication: string;
  };
  query?: {
    /**
     * The list of fields to update.
     */
    updateMask?: string;
  };
  url: "/v1beta1/publications/{publication}";
};

export type PublicationServiceUpdatePublicationErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type PublicationServiceUpdatePublicationError =
  PublicationServiceUpdatePublicationErrors[keyof PublicationServiceUpdatePublicationErrors];

export type PublicationServiceUpdatePublicationResponses = {
  /**
   * OK
   */
  200: Publication;
};

export type PublicationServiceUpdatePublicationResponse =
  PublicationServiceUpdatePublicationResponses[keyof PublicationServiceUpdatePublicationResponses];

export type PublicationServicePublishPublicationData = {
  body: PublishPublicationRequest;
  path: {
    /**
     * The publication id.
     */
    publication: string;
  };
  query?: never;
  url: "/v1beta1/publications/{publication}:publish";
};

export type PublicationServicePublishPublicationErrors = {
  /**
   * Default error response
   */
  default: Status;
};

export type PublicationServicePublishPublicationError =
  PublicationServicePublishPublicationErrors[keyof PublicationServicePublishPublicationErrors];

export type PublicationServicePublishPublicationResponses = {
  /**
   * OK
   */
  200: Operation;
};

export type PublicationServicePublishPublicationResponse =
  PublicationServicePublishPublicationResponses[keyof PublicationServicePublishPublicationResponses];
