// @canonical/react-components@3.11.1/dist/esm/components/Textarea/Textarea.js
var _excluded = ["caution", "className", "error", "grow", "help", "id", "label", "labelClassName", "onKeyUp", "onControlEnter", "required", "stacked", "style", "success", "takeFocus", "wrapperClassName"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import classNames from "classnames";
import React, { useCallback, useEffect, useId, useLayoutEffect, useRef, useState } from "react";
import Field from "../Field";

/**
 * The props for the Textarea component.
 */

/**
 * This is a [React](https://reactjs.org/) component for the Vanilla [Textarea](https://docs.vanillaframework.io/base/forms/#textarea).
 *
 * The Textarea component defines a multi-line text input control.
 */
var Textarea = _ref => {
  var _props$value;
  var {
      caution,
      className,
      error,
      grow = false,
      help,
      id,
      label,
      labelClassName,
      onKeyUp: _onKeyUp,
      onControlEnter,
      required,
      stacked,
      style,
      success,
      takeFocus = false,
      wrapperClassName
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  var textareaRef = useRef(null);
  var validationId = useId();
  var helpId = useId();
  var hasError = !!error;
  var [innerValue, setInnervalue] = useState(props.defaultValue);
  var defaultTextAreaId = useId();
  var textAreaId = id || defaultTextAreaId;
  var handleKeyDown = useCallback(event => {
    if (event.key === "Enter" && (event.ctrlKey || event.metaKey) && document.activeElement === textareaRef.current) {
      onControlEnter();
    }
  }, [onControlEnter]);
  useEffect(() => {
    document.addEventListener("keydown", handleKeyDown);
    return () => {
      document.removeEventListener("keydown", handleKeyDown);
    };
  }, [handleKeyDown]);
  useEffect(() => {
    if (takeFocus) {
      textareaRef.current.focus();
    }
  }, [takeFocus]);
  useLayoutEffect(() => {
    if (grow) {
      var textArea = textareaRef.current;
      if (textArea) {
        textArea.style.height = "0px";
        var scrollHeight = textArea.scrollHeight;
        textArea.style.height = "".concat(scrollHeight, "px");
      }
    }
  }, [textareaRef, grow, innerValue, props.value]);
  return /*#__PURE__*/React.createElement(Field, {
    caution: caution,
    className: wrapperClassName,
    error: error,
    forId: textAreaId,
    help: help,
    helpId: helpId,
    label: label,
    labelClassName: labelClassName,
    required: required,
    stacked: stacked,
    success: success,
    validationId: validationId
  }, /*#__PURE__*/React.createElement("textarea", _extends({
    "aria-describedby": [help ? helpId : null, success ? validationId : null].filter(Boolean).join(" "),
    "aria-errormessage": hasError ? validationId : null,
    "aria-invalid": hasError,
    className: classNames("p-form-validation__input", className),
    id: textAreaId,
    onKeyUp: evt => {
      _onKeyUp && _onKeyUp(evt);
    },
    onChange: evt => {
      if (!props.value) {
        setInnervalue(evt.target.value);
      }
    },
    ref: textareaRef,
    style: grow && _objectSpread({
      minHeight: "5rem",
      resize: "none",
      overflow: "hidden",
      boxSizing: "border-box"
    }, style) || style,
    required: required
  }, props, {
    value: (_props$value = props.value) !== null && _props$value !== void 0 ? _props$value : innerValue
  })));
};
export default Textarea;
