// @canonical/react-components@3.11.1/dist/components/ThemeSwitcher/ThemeSwitcher.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.loadTheme = exports.isDarkTheme = exports.default = exports.applyTheme = void 0;
var _react = _interopRequireWildcard(require("react"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const LOCAL_STORAGE_KEY = "theme";
const THEME_SYSTEM = "system";
const THEME_DARK = "dark";
const THEME_LIGHT = "light";
const loadTheme = () => {
  const saved = localStorage.getItem(LOCAL_STORAGE_KEY);
  return saved || THEME_SYSTEM;
};
exports.loadTheme = loadTheme;
const saveTheme = theme => {
  localStorage.setItem(LOCAL_STORAGE_KEY, theme);
};
const isDarkTheme = theme => {
  if (theme === THEME_SYSTEM) {
    return window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)").matches;
  }
  return theme === THEME_DARK;
};
exports.isDarkTheme = isDarkTheme;
const applyTheme = theme => {
  if (isDarkTheme(theme)) {
    document.body.classList.add("is-dark");
  } else {
    document.body.classList.remove("is-dark");
  }
};

/**
 * This is a [React](https://reactjs.org/) component for the [Vanilla framework](https://docs.vanillaframework.io).
 *
 * The ThemeSwitcher component allows users to switch between different themes: dark, light, and system. It saves the selected theme in local storage and applies it to the document body. You can use it in user settings.
 *
 * In your root component, call the exported functions `loadTheme` and `applyTheme`, such as in the example below:
 * ```
 * useEffect(() => {
 *   const theme = loadTheme();
 *   applyTheme(theme);
 * }, []);
 * ```
 */
exports.applyTheme = applyTheme;
const ThemeSwitcher = () => {
  const [activeTheme, setActiveTheme] = (0, _react.useState)(loadTheme());
  const themeButton = theme => {
    return /*#__PURE__*/_react.default.createElement("button", {
      className: "p-segmented-control__button",
      type: "button",
      "aria-selected": activeTheme === theme ? "true" : "false",
      onClick: () => {
        saveTheme(theme);
        setActiveTheme(theme);
        applyTheme(theme);
      }
    }, theme);
  };
  return /*#__PURE__*/_react.default.createElement("div", {
    className: "p-segmented-control"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "p-segmented-control__list",
    "aria-label": "Theme switcher"
  }, themeButton(THEME_DARK), themeButton(THEME_LIGHT), themeButton(THEME_SYSTEM)));
};
var _default = exports.default = ThemeSwitcher;
